<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta http-equiv="x-ua-compatible" content="ie=edge">
<meta name="keywords" content="Mark Moll" />
<title>Mark Moll | Research</title>
<script defer src="https://use.fontawesome.com/releases/v5.5.0/js/all.js" integrity="sha384-GqVMZRt5Gn7tB9D9q7ONtcp4gtHIUEW/yG7h98J7IpE3kpi+srfFyyB/04OV6pG0" crossorigin="anonymous"></script>
<link rel="stylesheet" href="/assets/css/layout.css" media="screen" type="text/css">

</head>
<body>
  <nav class="navbar navbar-expand-md fixed-top navbar-dark bg-dark">
  <a class="navbar-brand" href="/index.html">Mark Moll</a>
  <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbar">
    <span class="navbar-toggler-icon"></span>
  </button>
  <div class="collapse navbar-collapse" id="navbar">
    <ul class="navbar-nav mr-auto"><li class="nav-item"><a class="nav-link" href="/index.html"><i class="fas fa-home"></i> Home</a></li><li class="nav-item"><a class="nav-link" href="/publications.html"><i class="fas fa-book"></i> Publications</a></li><li class="nav-item active"><a class="nav-link" href="/research.html"><i class="fas fa-lightbulb"></i> Research</a></li><li class="nav-item"><a class="nav-link" href="/software.html"><i class="fas fa-wrench"></i> Software</a></li><li class="nav-item"><a class="nav-link" href="/teaching.html"><i class="fas fa-users"></i> Teaching</a></li></ul>
  </div>
</nav>


  <div class="container defaultfullwidth">
    <div class="row"><div class="col">
      
        <h1>Research</h1>
      
      <h2 id="tracing-conformational-changes-in-proteins">Tracing conformational changes in proteins</h2>

<div class="research">
  <div>
    <p>Many large protein complexes undergo extensive conformational changes as part of their functionality. Tracing these changes is important for understanding the way these proteins function. We have developed a novel computational methodology to efficiently trace the conformational changes in biological macromolecules. The methodology borrows from robot motion planning techniques. A key component of the approach is the parametrization of the degrees of freedom in such a way that energetically favorable conformational changes are favored by the motion planning-inspired conformational sampling algorithm.</p>
  </div>
  <p><img src="assets/pics/groel_path.png" alt="" title="Path for a GroEL monomer." /></p>
</div>

<h2 id="functional-annotation-of-proteins">Functional annotation of proteins</h2>

<div class="research">
  <div>
    <p>Broad and extensive knowledge of the biological function of proteins would have immense practical impact on the identification of novel drug targets, the reduction of potential side effects, and on finding the molecular causes of disease. Unfortunately, the experimental determination of protein function is an expensive and time consuming process. In an effort to accelerate and guide the experimental process, computational techniques have been developed to annotate functional information about well-studied proteins onto predictably similar but less-studied proteins.</p>

    <p>We are currently investigating the extent that patterns in the structure and physicochemical properties of small sets of residues can be used to predict function. Specifically, we have developed algorithms for the following problems:</p>

    <p><em>The prediction of function as characterized by Enzyme Commission classification or Gene Ontology molecular function classification.</em> Given a small spatial arrangement of residues characteristic of some function (a <em>motif</em>), we are able to quickly identify with high sensitivity and specificity all proteins in the entire Protein Data Bank with that function.</p>

    <p><em>The identification of small variations within the structures available for a protein family.</em> These variations are often correlated with phylogeny or binding mode. Such variations can be exploited to further improve the predictions in the previous problem.</p>

    <p><em>The prediction of inhibitor affinity for the entire human kinome.</em> Kinases are all very similar to each other and are involved in many different biological processes. Being able to predict affinity of putative inhibitors is useful in computational drug design. Here, we do not assume a motif is known. Instead, we exhaustively explore all combinations of binding site residues whose structure and physicochemical properties are predictive of binding some inhibitor. There is not just one motif; the specificity-determining residues are inhibitor dependent.</p>
  </div>
  <p><img src="assets/pics/labelhash-match.png" alt="" title="Query residues (in white) and matching residues (in green) superimposed. The rest of the matching protein is shown in ribbon representation." /></p>
</div>

<h2 id="self-reconfigurable-robots"><a href="http://www.isi.edu/robots">Self-reconfigurable robots</a></h2>

<div class="research">
  <div>
    <p>Over the years many different robots have been developed for specific tasks. Self-reconfigurable robots are the only truly general purpose robots. For a given task, they can re-arrange themselves to perform that task. For instance, a number of modules can form an arm to help astronauts perform tasks during space walks, they can be reconfigured into planetary rover, form a network of communication beacons, and collect samples. The <a href="http://www.isi.edu/robots">Polymorphic Robotics Laboratory</a> is working on the next generation of self-reconfigurable robots called SuperBot. This system is currently being built. I was in charge of the software development, both on the modules and the simulator code.</p>

    <p>I also worked on computing mass properties of an ensemble of modules in a distributed fashion. The next step is to control these mass properties and synthesize local behaviors that give rise to stable locomotion on uneven terrain.</p>
  </div>
  <p><img src="assets/pics/SuperBotStand-sm.png" alt="" title="A homunculus composed of 6 SuperBot modules." /></p>
</div>

<div class="researchtext">
  <p><a href="http://www.isi.edu/robots" class="btn btn-primary btn-sm">More »</a></p>
</div>

<h2 id="protein-folding-analysis">Protein folding analysis</h2>

<div class="research">
  <div>
    <p>The definition of reaction coordinates for the characterization of a protein folding reaction has long been a controversial issue, even for the “simple” case where one single free energy barrier separates the folded and unfolded ensemble. Together with <a href="http://www.cs.rice.edu/~kavraki">Lydia Kavraki</a>’s group and <a href="http://leonardo.rice.edu/~cecilia/research">Cecilia Clementi</a>’s group we have developed a general approach to this problem to obtain a few collective coordinates by using nonlinear dimensionality reduction. We validated the usefulness of this method by characterizing the folding landscape associated with a coarse-grained protein model of src homology 3 as sampled by molecular dynamics simulations. The folding free-energy landscape projected on the few relevant coordinates emerging from the dimensionality reduction can correctly identify the transition-state ensemble of the reaction. The first embedding dimension efficiently captures the evolution of the folding process along the main folding route. These results clearly show that the proposed method can efficiently find a low-dimensional representation of a complex process such as protein folding.</p>

    <p>Although the work so far has focused on protein folding, the <a href="http://isomap.stanford.edu">isomap</a>-based method we have developed is applicable to other domains with large high-dimensional data sets. Especially if the data is non-uniformly sampled, our approach can offer a significant speed-up over plain landmark isomap.</p>
  </div>
  <p><img src="assets/pics/scimap-sm.png" alt="" title="Visualization of the free-energy landscape of src-SH3." /></p>
</div>

<h2 id="motion-planning-for-flexible-objects">Motion planning for flexible objects</h2>

<div class="research">
  <div>
    <p>We have been working on path planning for deformable linear objects such as sutures. The main difficulty in finding a solution for this problem is that the configuration space of a deformable linear object is infinite-dimensional. A subset of these configurations consists of stable poses, or equilibrium configurations. A pose is stable if for given endpoint constraints the potential energy of the wire is at a minimum.</p>

    <p>We came up with an adaptive algorithm based on subdivision that efficiently represents equilibrium configurations of deformable linear objects. Our path planner uses this algorithm to find paths from one stable configuration to another, such that all intermediate configurations are also stable. An example path is shown on the right. The little cylinders denote the positions and tangents of the endpoints, which are constrained by manipulators. This path planner can be used as a local in a sampling-based planner to build up a roadmap of “shape space.”</p>
  </div>
  <p><img src="assets/pics/curvepath-sm.png" alt="" title="A path from one stable configuration of a flexible wire to another." /></p>
</div>

<h2 id="flexible-receptor-docking">Flexible receptor docking</h2>

<div class="research">
  <div>
    <p>The study of molecular interactions is a central issue in biochemistry. The recognition between a ligand and its receptor plays a determinant role in virtually all biological processes. The goal in molecular docking is to quickly determine how well two molecules fit together. One of them (called the receptor) is a large protein whose function is influenced by the second molecule (called the ligand). This problem is central to computational drug design. Due to the formidable combinatorial complexity of modeling the interaction between a flexible protein and a flexible ligand, most current docking methods limit their search by approximating the receptor as a rigid structure. This approximation limits their applicability since it does not allow the receptor to adapt to the ligand in an induced fit model. In <a href="http://www.cs.rice.edu/~kavraki">Lydia Kavraki</a>’s group I have worked on modeling the flexibility of the receptor. Using approximations of the underlying degrees of freedom we can efficiently explore the low-energy parts of the configuration space of large flexible receptors.</p>
  </div>
  <p><img src="assets/pics/hiv_vdw-sm.jpg" alt="" title="The docked conformation of a ligand (shown in orange) and HIV-1 protease." /></p>
</div>

<h2 id="tactile-sensing"><a href="tactilesensing.html">Tactile sensing</a></h2>

<div class="research">
  <div>
    <p><a href="http://www.cs.cmu.edu/~me">Michael Erdmann</a> and I have developed a method to reconstruct the shape of an unknown object using tactile sensors without requiring object immobilization. Instead, the robot manipulates the object without prehension. The robot infers the shape, motion and center of mass of the object based on the motion of the contact points as measured by tactile sensors. This allows for a natural, continuous interaction between manipulation and sensing. Eventually, it will make robots more capable in the physical world by enabling them to pick up unknown objects.</p>

    <p>We have analyzed several different cases of the tactile shape reconstruction problem. First, we considered planar shapes with quasistatic dynamics. Simulations and experiments have validated the analytic results. Next, we extended the analysis to the full dynamics and prove observability of the nonlinear system describing the shape and motion of the object being manipulated. In our simulations, a simple observer based on Newton’s method for root finding can recover unknown shapes with almost negligible errors. Using the same framework we can also describe the shape and dynamics of three-dimensional objects. However, there are some fundamental differences between the planar and three-dimensional case, due to increased tangent dimensionality. Also, perfect global shape reconstruction is impossible in the 3D case, but it is almost trivial to obtain upper and lower bounds on the shape. The 3D shape reconstruction method has also been implemented and we present some simulation results.</p>
  </div>
  <p><img src="assets/pics/palms.png" alt="" title="Two palms reconstructing the shape of an unknown object from tactile data." width="200px" /></p>
</div>

<div class="researchtext">
  <p><a href="tactilesensing.html" class="btn btn-primary btn-sm">More »</a></p>
</div>

<h2 id="parts-orienting">Parts orienting</h2>

<div class="research">
  <div>
    <p>In parts orienting the goal is to bring a known part in an unknown orientation to a known orientation. Often this is done without any sensing. I have worked on this problem at the <em>macro-level</em> and the <em>micro-level.</em> At the <em>macro-level,</em> I have developed strategies to orient three-dimensional parts with minimal sensing and manipulation. That is, we would like to bring a part from an unknown position and orientation to a known orientation (but possibly unknown position) with minimal means. In general, it is not possible to orient a part completely without sensors, but it is sufficient if a particular orienting strategy can bring a part into one particular orientation with high probability. The sensing is then reduced to a binary decision: a sensor only has to detect whether the part is in the right orientation or not. If not, the part is fed back to the parts orienting device. Assuming the orienting strategy succeeds with high probability, on average it takes just a few tries to orient the part. An alternative view of this type of manipulation is to consider it as manipulation of the pose distribution. The goal then is to find the pose distribution with minimal entropy, thereby maximally reducing uncertainty.</p>
  </div>
  <p><img src="assets/pics/bowl.jpg" alt="" title="A bowl that minimizes the entropy of the pose distribution of a given part." width="200px" /></p>
</div>

<div class="research">
  <div>
    <p>During the Summer of 2001 I worked with <a href="http://www.ieor.berkeley.edu/~goldberg">Ken Goldberg</a> at UC Berkeley on orienting parts at the <em>micrometer scale.</em> As microelectromechanical systems (MEMS) become ubiquitous, the need to orient micro-scale parts becomes more important. At this scale sticking effects complicate hand-off motions and the degrees of freedom of micro-manipulators are limited. We found that polygonal parts can be oriented with a parallel jaw gripper using two simple primitives: squeezing and rolling the part between the jaws. The gripper does not need to be reoriented and no sensing is required. I developed an algorithm that finds a plan of length <em>O(n)</em> in <em>O(n)</em> time for orienting a polygonal micro-scale part with <em>n</em> vertices. To the best of my knowledge this is the first algorithmic approach to micromanipulation.</p>
  </div>
  <p><img src="assets/pics/gripper-sm.png" alt="" title="Design for parallel jaw micro-gripper." width="150px" /></p>
</div>

    </div></div>
  </div>

  <footer>
    <!-- Made with <a href="http://jekyllrb.com">Jekyll</a> and <a href="http://getbootstrap.com">Bootstrap</a>. -->
    <a href="mailto:%6D%6D%6F%6C%6C@%72%69%63%65.%65%64%75" data-toggle="tooltip" title="Email" class="btn-tooltip"><i class="far fa-envelope"></i></a><a href="https://www.linkedin.com/in/mmoll" data-toggle="tooltip" title="LinkedIn" class="btn-tooltip"><i class="fab fa-linkedin"></i></a><a href="https://scholar.google.com/citations?user=sLXafasAAAAJ&hl=en" data-toggle="tooltip" title="Google Scholar" class="btn-tooltip"><i class="ai ai-google-scholar"></i></a><a href="https://bitbucket.org/mamoll" data-toggle="tooltip" title="Bitbucket" class="btn-tooltip"><i class="fab fa-bitbucket"></i></a><a href="https://github.com/mamoll" data-toggle="tooltip" title="GitHub" class="btn-tooltip"><i class="fab fa-github"></i></a><a href="http://orcid.org/0000-0002-0451-2797" data-toggle="tooltip" title="ORCID" class="btn-tooltip"><i class="ai ai-orcid"></i></a><a href="https://www.semanticscholar.org/author/Mark-Moll/40268850" data-toggle="tooltip" title="Semantic Scholar"><i class="ai ai-semantic-scholar"></i></a><a href="https://www.researchgate.net/profile/Mark_Moll" data-toggle="tooltip" title="ResearchGate" class="btn-tooltip"><i class="ai ai-researchgate"></i></a><a href="http://www.researcherid.com/rid/A-1776-2011" data-toggle="tooltip" title="ResearcherID" class="btn-tooltip"><i class ="ai ai-researcherid"></i></a><a href="https://www.mendeley.com/profiles/mark-moll/" data-toggle="tooltip" title="Mendeley" class="btn-tooltip"><i class="ai ai-mendeley"></i></a><a href="https://impactstory.org/u/0000-0002-0451-2797" data-toggle="tooltip" title="Impactstory" class="btn-tooltip"><i class="ai ai-impactstory"></i></a>
</footer>
<script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.6/umd/popper.min.js" integrity="sha384-wHAiFfRlMFy6i5SRaxvfOCifBUQy1xHdJ/yoi7FRNXMRBu5WHdZYu1hA6ZOblgut" crossorigin="anonymous"></script>
<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.2.1/js/bootstrap.min.js" integrity="sha384-B0UglyR+jN6CkvvICOB2joaf5I4l3gm9GU6Hc1og6Ls7i6U/mkkaduKaBhlAXv9k" crossorigin="anonymous"></script>
<script>
  $('img[title]').each(function() {
    var img, caption;
    img = $(this).attr('src');
    caption = $(this).attr('title');
    if ($(this).is('[width]'))
      width = ' style="width:' + $(this).attr('width') +'"';
    else
      width=''
    if ($(this).is('[class]'))
      cls = ' class="' + $(this).attr('class') +'"';
    else
      cls=''
    $(this).parent().replaceWith($('<figure' + cls + width + '><img class="figure-img img-fluid" src="' + img + '"><figcaption>' + caption + '</figcaption></figure>'));
  });
</script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-3878170-1', 'auto');
  ga('send', 'pageview');
</script>

</body>
</html>
