<!doctype html> <!-- Minimal Mistakes Jekyll Theme 4.16.6 by Michael Rose Copyright 2013-2019 Michael Rose - mademistakes.com | @mmistakes Free for personal and commercial use under the MIT license https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE --><html lang="en" class="no-js"><head><meta charset="utf-8"> <!-- begin _includes/seo.html --><title>Research | Mark Moll</title><meta name="description" content=""><meta name="author" content="Mark Moll"><meta property="og:type" content="website"><meta property="og:locale" content="en_US"><meta property="og:site_name" content="Mark Moll"><meta property="og:title" content="Research"><meta property="og:url" content="https://moll.ai/research/"><link rel="canonical" href="https://moll.ai/research/"> <script type="application/ld+json"> { "@context": "https://schema.org", "@type": "Person", "name": "Mark Moll", "url": "https://moll.ai/" } </script> <!-- end _includes/seo.html --><link href="" type="application/atom+xml" rel="alternate" title="Mark Moll Feed"> <!-- https://t.co/dKP3o1e --><meta name="viewport" content="width=device-width, initial-scale=1.0"> <script> document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js '; </script> <!-- For all browsers --><link rel="stylesheet" href="/assets/css/main.css"> <!--[if IE]><style> /* old IE unsupported flexbox fixes */ .greedy-nav .site-title { padding-right: 3em; } .greedy-nav button { position: absolute; top: 0; right: 0; height: 100%; }</style><![endif]--> <!-- start custom head snippets --> <!-- insert favicons. use https://realfavicongenerator.net/ --> <!-- end custom head snippets --></head><body class="layout--single"><nav class="skip-links"><h2 class="screen-reader-text">Skip links</h2><ul><li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li><li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li><li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li></ul></nav><!--[if lt IE 9]><div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div><![endif]--><div class="masthead"><div class="masthead__inner-wrap"><div class="masthead__menu"><nav id="site-nav" class="greedy-nav"> <a class="site-title" href="/"> Mark Moll </a><ul class="visible-links"><li class="masthead__menu-item"> <a href="/publications/" >publications</a></li><li class="masthead__menu-item"> <a href="/research/" >research</a></li><li class="masthead__menu-item"> <a href="/software/" >software</a></li><li class="masthead__menu-item"> <a href="https://picknik.ai" >consulting</a></li></ul><button class="greedy-nav__toggle hidden" type="button"> <span class="visually-hidden">Toggle menu</span><div class="navicon"></div></button><ul class="hidden-links hidden"></ul></nav></div></div></div><div class="initial-content"><div id="main" role="main"><div class="sidebar sticky"><div itemscope itemtype="https://schema.org/Person"><div class="author__avatar"> <img src="/assets/images/MarkMoll2017.jpg" alt="Mark Moll" itemprop="image"></div><div class="author__content"><h3 class="author__name" itemprop="name">Mark Moll</h3></div><div class="author__urls-wrapper"> <button class="btn btn--inverse">Follow</button><ul class="author__urls social-icons"><li itemprop="homeLocation" itemscope itemtype="https://schema.org/Place"> <i class="fas fa-fw fa-map-marker-alt" aria-hidden="true"></i> <span itemprop="name"><a href="https://goo.gl/maps/VRbXssEZ5yxv5b326">Portland, OR</a></span></li><li><a href="mailto:mark@moll.ai" rel="nofollow noopener noreferrer"><i class="fas fa-fw fa-envelope-square" aria-hidden="true"></i> Email</a></li><li><a href="https://www.linkedin.com/in/mmoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i> LinkedIn</a></li><li><a href="https://scholar.google.com/citations?user=sLXafasAAAAJ&hl=en" rel="nofollow noopener noreferrer"><i class="ai ai-fw ai-google-scholar" aria-hidden="true"></i> Google Scholar</a></li><li><a href="http://orcid.org/0000-0002-0451-2797" rel="nofollow noopener noreferrer"><i class="ai ai-fw ai-orcid" aria-hidden="true"></i> ORCID</a></li><li><a href="https://github.com/mamoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li><li><a href="https://bitbucket.org/mamoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-bitbucket" aria-hidden="true"></i> BitBucket</a></li><!--<li> <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs" rel="nofollow noopener noreferrer"> <i class="fas fa-fw" aria-hidden="true"></i> Custom Social Profile Link </a></li>--></ul></div></div></div><article class="page" itemscope itemtype="https://schema.org/CreativeWork"><meta itemprop="headline" content="Research"><div class="page__inner-wrap"><header><h1 id="page-title" class="page__title" itemprop="headline">Research</h1></header><section class="page__content" itemprop="text"><aside class="sidebar__right "><nav class="toc"><header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On this page</h4></header><ul class="toc__menu"><li><a href="#sampling-based-motion-planning">Sampling-based motion planning</a></li><li><a href="#modeling-conformational-changes-in-proteins">Modeling conformational changes in proteins</a></li><li><a href="#functional-annotation-of-proteins">Functional annotation of proteins</a></li><li><a href="#self-reconfigurable-robots">Self-reconfigurable robots</a></li><li><a href="#protein-folding-analysis">Protein folding analysis</a></li><li><a href="#tactile-sensing">Tactile sensing</a></li><li><a href="#parts-orienting">Parts orienting</a></li></ul></nav></aside><h2 id="sampling-based-motion-planning">Sampling-based motion planning</h2><figure class="third "> <a href="/assets/images/r2-iss-path-sm.jpg" title="Simulation of Robonaut2 climbing across handrails in the International Space Station"> <img src="/assets/images/r2-iss-path.jpg" alt="" /> </a> <a href="/assets/images/r2.jpg" title="Robonaut2 performing a series of constrained motions: grabbing a handrail, pulling a bag off a shelf, and rotating a valve."> <img src="/assets/images/r2.jpg" alt="" /> </a> <a href="/assets/images/sparus.jpg" title="University of Girona's Sparus II autonomous underwater vehicle."> <img src="/assets/images/sparus.jpg" alt="" /> </a><figcaption>Sampling-based planning algorithms have been used for a variety of complex systems, ranging from systems with many degrees of freedom such as NASA’s Robonaut2 to systems subject to complex kinodynamic constraints such as the Sparus II AUV.</figcaption></figure><p>I have worked extensively on robot motion planning, focusing on a particular type of algorithms that rely on sampling random robot configurations. These algorithms have been shown to be effective at efficiently computing paths for a variety of complex systems. With several collaborators, I have developed new algorithms that are tailored for robots with many degrees of freedom and/or complex dynamics. Implementations of these algorithms are typically made available through the <a href="https://ompl.kavrakilab.org">Open Motion Planning Library</a>.</p><p>Recently, we have shown that many previously proposed algorithms for planning with geometric constraints (such as end-effector constraints) can be generalized by decoupling the motion planning strategy from the constraint adherence methodology. This enables us to easily optimize the combination of planning strategy and constraint adherence for a particular robot or application.</p><p>I have also been working with the University of Girona on algorithms that can be deployed on their Sparus II AUV. This robot has complex dynamics constraints, limited computational power, and is deployed in an initially unknown environment. On top of that, there are currents to deal with and very noisy sensor data. We have developed novel, practical online planning algorithms that produce high-quality solutions within a fixed time budget.</p><h2 id="modeling-conformational-changes-in-proteins">Modeling conformational changes in proteins</h2><figure class="half "> <a href="/assets/images/C3-C3b-path.gif" title="Conformational change between C3 and C3b (part of the complement pathway), modeled with SIMS"> <img src="/assets/images/C3-C3b-path.gif" alt="" /> </a> <a href="/assets/images/mmcpn.gif" title="Transition between the “open” and “closed” state of MmCpn (a large protein complex), modeled with SIMS"> <img src="/assets/images/mmcpn.gif" alt="" /> </a><figcaption>Conformational changes modeled with the Structured Intuitive Move Selector (SIMS), a conformational sampling algorithm using robotics-inspired techniques.</figcaption></figure><p>Many large protein complexes undergo extensive conformational changes as part of their functionality. Tracing these changes is important for understanding the way these proteins function. We have developed a novel computational methodology to efficiently model the conformational changes in biological macromolecules. The methodology borrows from robot motion planning techniques. A key component of the approach is the parametrization of the degrees of freedom in such a way that energetically favorable conformational changes are favored by the motion planning-inspired conformational sampling algorithm.</p><h2 id="functional-annotation-of-proteins">Functional annotation of proteins</h2><p><img src="/assets/images/labelhash-match.png" alt="" title="Query residues (in white) and matching residues (in green) superimposed. The rest of the matching protein is shown in ribbon representation." class="align-right" /></p><p>Broad and extensive knowledge of the biological function of proteins would have immense practical impact on the identification of novel drug targets, the reduction of potential side effects, and on finding the molecular causes of disease. Unfortunately, the experimental determination of protein function is an expensive and time-consuming process. In an effort to accelerate and guide the experimental process, computational techniques have been developed to annotate functional information about well-studied proteins onto predictably similar but less-studied proteins.</p><p>We are currently investigating the extent that patterns in the structure and physicochemical properties of small sets of residues can be used to predict function. Specifically, we have developed algorithms for the following problems:</p><p><em>The prediction of protein function.</em> Given a small spatial arrangement of residues characteristic of some function (a <em>motif</em>), we are able to quickly identify with high sensitivity and specificity all proteins in the entire Protein Data Bank with that function.</p><p><em>The identification of patterns of variation within the structures available for a protein family.</em> These patterns are often correlated with phylogeny or binding mode. Such patterns can be exploited to further improve the predictions in the previous problem.</p><p><em>The prediction of inhibitor affinity for the entire human kinome.</em> Kinases are all very similar to each other and are involved in many different biological processes. Being able to predict affinity of putative inhibitors is useful in computational drug design. Here, we do not assume a motif is known. Instead, we exhaustively explore all combinations of binding site residues whose structure and physicochemical properties are predictive of binding some inhibitor. There is not just one motif; the specificity-determining residues are inhibitor dependent.</p><h2 id="self-reconfigurable-robots"><a href="https://www.isi.edu/robots">Self-reconfigurable robots</a></h2><p><img src="/assets/images/SuperBotStand-sm.png" alt="" title="A homunculus composed of 6 SuperBot modules." class="align-right" /></p><p>Over the years many different robots have been developed for specific tasks. Self-reconfigurable robots are the only truly general purpose robots. For a given task, they can re-arrange themselves to perform that task. For instance, a number of modules can form an arm to help astronauts perform tasks during space walks, they can be reconfigured into planetary rover, form a network of communication beacons, and collect samples. At the <a href="https://www.isi.edu/robots">Polymorphic Robotics Laboratory</a> I worked on a system of self-reconfigurable robots called SuperBot. I was in charge of the software development, both on the modules and the simulator code.</p><p><a href="https://www.isi.edu/robots" class="btn btn--primary btn--small">More »</a></p><h2 id="protein-folding-analysis">Protein folding analysis</h2><p><img src="/assets/images/scimap-sm.png" alt="" title="Visualization of the free-energy landscape of src-SH3." class="align-right" /></p><p>The definition of reaction coordinates for the characterization of a protein folding reaction has long been a controversial issue, even for the “simple” case where one single free energy barrier separates the folded and unfolded ensemble. Together with <a href="https://www.cs.rice.edu/~kavraki">Lydia Kavraki</a>’s group and <a href="http://leonardo.rice.edu/~cecilia/research">Cecilia Clementi</a>’s group we have developed a general approach to this problem to obtain a few collective coordinates by using nonlinear dimensionality reduction. We validated the usefulness of this method by characterizing the folding landscape associated with a coarse-grained protein model of src homology 3 as sampled by molecular dynamics simulations. The folding free-energy landscape projected on the few relevant coordinates emerging from the dimensionality reduction can correctly identify the transition-state ensemble of the reaction. The first embedding dimension efficiently captures the evolution of the folding process along the main folding route. These results clearly show that the proposed method can efficiently find a low-dimensional representation of a complex process such as protein folding.</p><p>Although the work so far has focused on protein folding, the <a href="https://web.mit.edu/cocosci/isomap/isomap.html">isomap</a>-based method we have developed is applicable to other domains with large high-dimensional data sets. Especially if the data is non-uniformly sampled, our approach can offer a significant speed-up over plain landmark isomap.</p><h2 id="tactile-sensing"><a href="/tactilesensing/">Tactile sensing</a></h2><p><img src="/assets/images/palms.png" alt="" title="Two palms reconstructing the shape of an unknown object from tactile data." width="200px" class="align-right" /></p><p><a href="https://www.cs.cmu.edu/~me">Michael Erdmann</a> and I have developed a method to reconstruct the shape of an unknown object using tactile sensors without requiring object immobilization. Instead, the robot manipulates the object without prehension. The robot infers the shape, motion and center of mass of the object based on the motion of the contact points as measured by tactile sensors. This allows for a natural, continuous interaction between manipulation and sensing. Eventually, it will make robots more capable in the physical world by enabling them to pick up unknown objects.</p><p>We have analyzed several different cases of the tactile shape reconstruction problem. First, we considered planar shapes with quasistatic dynamics. Simulations and experiments have validated the analytic results. Next, we extended the analysis to the full dynamics and prove observability of the nonlinear system describing the shape and motion of the object being manipulated. In our simulations, a simple observer based on Newton’s method for root finding can recover unknown shapes with almost negligible errors. Using the same framework we can also describe the shape and dynamics of three-dimensional objects. However, there are some fundamental differences between the planar and three-dimensional case, due to increased tangent dimensionality. Also, perfect global shape reconstruction is impossible in the 3D case, but it is almost trivial to obtain upper and lower bounds on the shape. The 3D shape reconstruction method has also been implemented and we present some simulation results.</p><p><a href="/tactilesensing/" class="btn btn--primary btn--small">More »</a></p><h2 id="parts-orienting">Parts orienting</h2><p><img src="/assets/images/bowl.jpg" alt="" title="A bowl that minimizes the entropy of the pose distribution of a given part." width="200px" class="align-right" /></p><p>In parts orienting the goal is to bring a known part in an unknown orientation to a known orientation. Often this is done without any sensing. I have worked on this problem at the <em>macro-level</em> and the <em>micro-level.</em> At the <em>macro-level,</em> I have developed strategies to orient three-dimensional parts with minimal sensing and manipulation. That is, we would like to bring a part from an unknown position and orientation to a known orientation (but possibly unknown position) with minimal means. In general, it is not possible to orient a part completely without sensors, but it is sufficient if a particular orienting strategy can bring a part into one particular orientation with high probability. The sensing is then reduced to a binary decision: a sensor only has to detect whether the part is in the right orientation or not. If not, the part is fed back to the parts orienting device. Assuming the orienting strategy succeeds with high probability, on average it takes just a few tries to orient the part. An alternative view of this type of manipulation is to consider it as manipulation of the pose distribution. The goal then is to find the pose distribution with minimal entropy, thereby maximally reducing uncertainty.</p><p><img src="/assets/images/gripper-sm.png" alt="" title="Design for parallel jaw micro-gripper." width="150px" class="align-right" /></p><p>I worked with <a href="https://goldberg.berkeley.edu">Ken Goldberg</a> at UC Berkeley on orienting parts at the <em>micrometer scale.</em> As microelectromechanical systems (MEMS) become ubiquitous, the need to orient micro-scale parts becomes more important. At this scale sticking effects complicate hand-off motions and the degrees of freedom of micro-manipulators are limited. We found that polygonal parts can be oriented with a parallel jaw gripper using two simple primitives: squeezing and rolling the part between the jaws. The gripper does not need to be reoriented and no sensing is required. I developed an algorithm that finds a plan of length <em>O(n)</em> in <em>O(n)</em> time for orienting a polygonal micro-scale part with <em>n</em> vertices. To the best of my knowledge this is the first algorithmic approach to micromanipulation.</p></section><footer class="page__meta"></footer></div></article></div></div><div id="footer" class="page__footer"><footer> <!-- start custom footer snippets --> <!-- end custom footer snippets --><div class="page__footer-follow"><ul class="social-icons"><li><a href="mailto:mark@moll.ai" rel="nofollow noopener noreferrer"><i class="fas fa-fw fa-envelope-square" aria-hidden="true"></i> Email</a></li><li><a href="https://www.linkedin.com/in/mmoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i> LinkedIn</a></li><li><a href="https://scholar.google.com/citations?user=sLXafasAAAAJ&hl=en" rel="nofollow noopener noreferrer"><i class="ai ai-fw ai-google-scholar" aria-hidden="true"></i> Google Scholar</a></li><li><a href="http://orcid.org/0000-0002-0451-2797" rel="nofollow noopener noreferrer"><i class="ai ai-fw ai-orcid" aria-hidden="true"></i> ORCID</a></li><li><a href="https://github.com/mamoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li><li><a href="https://bitbucket.org/mamoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-bitbucket" aria-hidden="true"></i> BitBucket</a></li></ul></div><div class="page__footer-copyright">&copy; 2019 Mark Moll.</div></footer></div><script src="/assets/js/main.min.js"></script> <script src="https://kit.fontawesome.com/4eee35f757.js"></script> <script> var _gaq = _gaq || []; _gaq.push(['_setAccount', 'UA-3878170-1']); _gaq.push(['_trackPageview']); (function() { var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true; ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js'; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s); })(); </script> <script src="/assets/js/footer-script.js"></script></body></html>
