<!doctype html> <!-- Minimal Mistakes Jekyll Theme 4.16.6 by Michael Rose Copyright 2013-2019 Michael Rose - mademistakes.com | @mmistakes Free for personal and commercial use under the MIT license https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE --><html lang="en" class="no-js"><head><meta charset="utf-8"> <!-- begin _includes/seo.html --><title>Publications | Mark Moll</title><meta name="description" content=""><meta name="author" content="Mark Moll"><meta property="og:type" content="website"><meta property="og:locale" content="en_US"><meta property="og:site_name" content="Mark Moll"><meta property="og:title" content="Publications"><meta property="og:url" content="https://moll.ai/publications-bytype/"><meta property="og:description" content=""><link rel="canonical" href="https://moll.ai/publications-bytype/"> <script type="application/ld+json"> { "@context": "https://schema.org", "@type": "Person", "name": "Mark Moll", "url": "https://moll.ai/" } </script> <!-- end _includes/seo.html --><link href="/feed.xml" type="application/atom+xml" rel="alternate" title="Mark Moll Feed"> <!-- https://t.co/dKP3o1e --><meta name="viewport" content="width=device-width, initial-scale=1.0"> <script> document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js '; </script> <!-- For all browsers --><link rel="stylesheet" href="/assets/css/main.css"><link rel="preload" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'"> <noscript><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css"></noscript> <!-- start custom head snippets --> <!-- insert favicons. use https://realfavicongenerator.net/ --> <!-- end custom head snippets --></head><body class="layout--single"><nav class="skip-links"><ul><li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li><li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li><li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li></ul></nav><!--[if lt IE 9]><div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div><![endif]--><div class="masthead"><div class="masthead__inner-wrap"><div class="masthead__menu"><nav id="site-nav" class="greedy-nav"> <a class="site-title" href="/"> Mark Moll </a><ul class="visible-links"><li class="masthead__menu-item"> <a href="/publications/">publications</a></li><li class="masthead__menu-item"> <a href="/research/">research</a></li><li class="masthead__menu-item"> <a href="/teaching/">teaching</a></li><li class="masthead__menu-item"> <a href="/software/">software</a></li><li class="masthead__menu-item"> <a href="https://picknik.ai">consulting</a></li></ul><button class="greedy-nav__toggle hidden" type="button"> <span class="visually-hidden">Toggle menu</span><div class="navicon"></div></button><ul class="hidden-links hidden"></ul></nav></div></div></div><div class="initial-content"><div id="main" role="main"><div class="sidebar sticky"><div itemscope itemtype="https://schema.org/Person"><div class="author__avatar"> <img src="/assets/images/MarkMoll2017.jpg" alt="Mark Moll" itemprop="image"></div><div class="author__content"><h3 class="author__name" itemprop="name">Mark Moll</h3></div><div class="author__urls-wrapper"> <button class="btn btn--inverse">Follow</button><ul class="author__urls social-icons"><li itemprop="homeLocation" itemscope itemtype="https://schema.org/Place"> <i class="fas fa-fw fa-map-marker-alt" aria-hidden="true"></i> <span itemprop="name"><a href="https://goo.gl/maps/VRbXssEZ5yxv5b326">Portland, OR</a></span></li><li><a href="mailto:mark@moll.ai" rel="nofollow noopener noreferrer"><i class="fas fa-fw fa-envelope-square" aria-hidden="true"></i><span class="label">Email</span></a></li><li><a href="https://www.linkedin.com/in/mmoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i><span class="label">LinkedIn</span></a></li><li><a href="https://scholar.google.com/citations?user=sLXafasAAAAJ&hl=en" rel="nofollow noopener noreferrer"><i class="ai ai-fw ai-google-scholar" aria-hidden="true"></i><span class="label">Google Scholar</span></a></li><li><a href="http://orcid.org/0000-0002-0451-2797" rel="nofollow noopener noreferrer"><i class="ai ai-fw ai-orcid" aria-hidden="true"></i><span class="label">ORCID</span></a></li><li><a href="https://github.com/mamoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-github" aria-hidden="true"></i><span class="label">GitHub</span></a></li><!--<li> <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs" rel="nofollow noopener noreferrer"> <i class="fas fa-fw" aria-hidden="true"></i> Custom Social Profile Link </a></li>--></ul></div></div></div><article class="page" itemscope itemtype="https://schema.org/CreativeWork"><meta itemprop="headline" content="Publications"><div class="page__inner-wrap"><header><h1 id="page-title" class="page__title" itemprop="headline">Publications</h1></header><section class="page__content" itemprop="text"><aside class="sidebar__right sticky"><nav class="toc"><ul class="toc__menu"><li class="nav-item"><a class="nav-link" href="#secjournal-articles">Journal Articles</a></li><li class="nav-item"><a class="nav-link" href="#secbook-chapters">Book Chapters</a></li><li class="nav-item"><a class="nav-link" href="#secconference-articles">Conference Articles</a></li><li class="nav-item"><a class="nav-link" href="#sectheses">Theses</a></li><li class="nav-item"><a class="nav-link" href="#sectechnical-reports">Technical Reports</a></li></ul></nav></aside><h2 class="bibliography" id="secjournal-articles">Journal Articles</h2><ol class="bibliography"><li> <span id="hernandez2020increasing-robot-autonomy-via-motion">J. D. Hernández, S. Sobti, A. Sciola, <b>M. Moll</b>, and L. E. Kavraki, “Increasing Robot Autonomy via Motion Planning and an Augmented Reality Interface,” <i>IEEE Robotics and Automation Letters</i>, vol. 5, no. 2, pp. 1017–1023, Apr. 2020.</span> <br><div id="acchernandez2020increasing-robot-autonomy-via-motion"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/hernandez2020increasing-robot-autonomy-via-motion.pdf');" href="/publications/hernandez2020increasing-robot-autonomy-via-motion.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/LRA.2020.2967280');" href="https://dx.doi.org/10.1109/LRA.2020.2967280"> publisher </a> <button class="btn" type="button" onclick="$('#bibhernandez2020increasing-robot-autonomy-via-motion').slideToggle(600);$('#abshernandez2020increasing-robot-autonomy-via-motion').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abshernandez2020increasing-robot-autonomy-via-motion').slideToggle(600);$('#bibhernandez2020increasing-robot-autonomy-via-motion').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibhernandez2020increasing-robot-autonomy-via-motion"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">hernandez2020increasing-robot-autonomy-via-motion</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Recently, there has been a growing interest in robotic systems
        that are able to share workspaces and collaborate with humans. Such
        collaborative scenarios require efficient mechanisms to communicate
        human requests to a robot, as well as to transmit robot interpretations
        and intents to humans. Recent advances in augmented reality (AR)
        technologies have provided an alternative for such communication.
        Nonetheless, most of the existing work in human-robot interaction with
        AR devices is still limited to robot motion programming or
        teleoperation. In this paper, we present an alternative approach to
        command and collaborate with robots. Our approach uses an AR interface
        that allows a user to specify high-level requests to a robot, to
        preview, approve or modify the computed robot motions. The proposed
        approach exploits the robot's decisionmaking capabilities instead of
        requiring low-level motion specifications provided by the user. The
        latter is achieved by using a motion planner that can deal with
        high-level goals corresponding to regions in the robot configuration
        space. We present a proof of concept to validate our approach in
        different test scenarios, and we present a discussion of its
        applicability in collaborative environments."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Hern{\'a}ndez, Juan David and Sobti, Shlok and Sciola, Anthony and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/LRA.2020.2967280"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"IEEE Robotics and Automation Letters"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"Motion and path planning, virtual reality and interfaces"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">apr</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"2"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"1017-1023"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Increasing Robot Autonomy via Motion Planning and an Augmented
        Reality Interface"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"5"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2020"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abshernandez2020increasing-robot-autonomy-via-motion"><h3>Abstract</h3><p>Recently, there has been a growing interest in robotic systems that are able to share workspaces and collaborate with humans. Such collaborative scenarios require efficient mechanisms to communicate human requests to a robot, as well as to transmit robot interpretations and intents to humans. Recent advances in augmented reality (AR) technologies have provided an alternative for such communication. Nonetheless, most of the existing work in human-robot interaction with AR devices is still limited to robot motion programming or teleoperation. In this paper, we present an alternative approach to command and collaborate with robots. Our approach uses an AR interface that allows a user to specify high-level requests to a robot, to preview, approve or modify the computed robot motions. The proposed approach exploits the robot’s decisionmaking capabilities instead of requiring low-level motion specifications provided by the user. The latter is achieved by using a motion planner that can deal with high-level goals corresponding to regions in the robot configuration space. We present a proof of concept to validate our approach in different test scenarios, and we present a discussion of its applicability in collaborative environments.</p></div></div></li><li> <span id="kim2020improving-the-organization-and-interactivity-of-metabolic">S. M. Kim, M. I. Peña, <b>M. Moll</b>, G. N. Bennett, and L. E. Kavraki, “Improving the organization and interactivity of metabolic pathfinding with precomputed pathways,” <i>BMC Bioinformatics</i>, vol. 21, no. 1, p. 13, Jan. 2020.</span> <br><div id="acckim2020improving-the-organization-and-interactivity-of-metabolic"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/kim2020improving-the-organization-and-interactivity-of-metabolic.pdf');" href="/publications/kim2020improving-the-organization-and-interactivity-of-metabolic.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1186/s12859-019-3328-x');" href="https://dx.doi.org/10.1186/s12859-019-3328-x"> publisher </a> <button class="btn" type="button" onclick="$('#bibkim2020improving-the-organization-and-interactivity-of-metabolic').slideToggle(600);$('#abskim2020improving-the-organization-and-interactivity-of-metabolic').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abskim2020improving-the-organization-and-interactivity-of-metabolic').slideToggle(600);$('#bibkim2020improving-the-organization-and-interactivity-of-metabolic').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibkim2020improving-the-organization-and-interactivity-of-metabolic"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">kim2020improving-the-organization-and-interactivity-of-metabolic</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Background: The rapid growth of available knowledge on metabolic
        processes across thousands of species continues to expand the
        possibilities of producing chemicals by combining pathways found in
        different species. Several computational search algorithms have been
        developed for automating the identification of possible heterologous
        pathways; however, these searches may return thousands of pathway
        results. Although the large number of results are in part due to the
        large number of possible compounds and reactions, a subset of core
        reaction modules is repeatedly observed in pathway results across
        multiple searches, suggesting that some subpaths between common
        compounds were more consistently explored than others.
        
        To reduce the resources spent on searching the same metabolic space, a
        new meta-algorithm for metabolic pathfinding, Hub Pathway search with
        Atom Tracking (HPAT), was developed to take advantage of a precomputed
        network of subpath modules. To investigate the efficacy of this method,
        we created a table describing a network of common hub metabolites and
        how they are biochemically connected and only offloaded searches to and
        from this hub network onto an interactive webserver capable of
        visualizing the resulting pathways.
        
        Results: A test set of nineteen known pathways taken from literature
        and metabolic databases were used to evaluate if HPAT was capable of
        identifying known pathways. HPAT found the exact pathway for eleven of
        the nineteen test cases using a diverse set of precomputed subpaths,
        whereas a comparable pathfinding search algorithm that does not use
        precomputed subpaths found only seven of the nineteen test cases. The
        capability of HPAT to find novel pathways was demonstrated by its
        ability to identify novel 3-hydroxypropanoate (3-HP) synthesis
        pathways. As for pathway visualization, the new interactive pathway
        filters enable a reduction of the number of displayed pathways from
        hundreds down to less than ten pathways in several test cases,
        illustrating their utility in reducing the amount of presented
        information while retaining pathways of interest.
        
        Conclusions: This work presents the first step in incorporating a
        precomputed subpath network into metabolic pathfinding and demonstrates
        how this leads to a concise, interactive visualization of pathway
        results. The modular nature of metabolic pathways is exploited to
        facilitate efficient discovery of alternate pathways. "</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Kim, Sarah M. and Pe{\~n}a, Matthew I. and Moll, Mark and Bennett, George N. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1186/s12859-019-3328-x"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"BMC Bioinformatics"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"metabolic pathfinding; precomputation; atom mapping; graph
        search"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jan</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"13"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Improving the organization and interactivity of metabolic
        pathfinding with precomputed pathways"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"21"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2020"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abskim2020improving-the-organization-and-interactivity-of-metabolic"><h3>Abstract</h3><p>Background: The rapid growth of available knowledge on metabolic processes across thousands of species continues to expand the possibilities of producing chemicals by combining pathways found in different species. Several computational search algorithms have been developed for automating the identification of possible heterologous pathways; however, these searches may return thousands of pathway results. Although the large number of results are in part due to the large number of possible compounds and reactions, a subset of core reaction modules is repeatedly observed in pathway results across multiple searches, suggesting that some subpaths between common compounds were more consistently explored than others. To reduce the resources spent on searching the same metabolic space, a new meta-algorithm for metabolic pathfinding, Hub Pathway search with Atom Tracking (HPAT), was developed to take advantage of a precomputed network of subpath modules. To investigate the efficacy of this method, we created a table describing a network of common hub metabolites and how they are biochemically connected and only offloaded searches to and from this hub network onto an interactive webserver capable of visualizing the resulting pathways. Results: A test set of nineteen known pathways taken from literature and metabolic databases were used to evaluate if HPAT was capable of identifying known pathways. HPAT found the exact pathway for eleven of the nineteen test cases using a diverse set of precomputed subpaths, whereas a comparable pathfinding search algorithm that does not use precomputed subpaths found only seven of the nineteen test cases. The capability of HPAT to find novel pathways was demonstrated by its ability to identify novel 3-hydroxypropanoate (3-HP) synthesis pathways. As for pathway visualization, the new interactive pathway filters enable a reduction of the number of displayed pathways from hundreds down to less than ten pathways in several test cases, illustrating their utility in reducing the amount of presented information while retaining pathways of interest. Conclusions: This work presents the first step in incorporating a precomputed subpath network into metabolic pathfinding and demonstrates how this leads to a concise, interactive visualization of pathway results. The modular nature of metabolic pathways is exploited to facilitate efficient discovery of alternate pathways.</p></div></div></li><li> <span id="luna2020a-scalable-motion-planner-for-high-dimensional">R. Luna, <b>M. Moll</b>, J. M. Badger, and L. E. Kavraki, “A Scalable Motion Planner for High-Dimensional Kinematic Systems,” <i>Intl. J. of Robotics Research</i>, vol. 39, no. 4, pp. 361–388, 2020.</span> <br><div id="accluna2020a-scalable-motion-planner-for-high-dimensional"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/luna2020a-scalable-motion-planner-for-high-dimensional.pdf');" href="/publications/luna2020a-scalable-motion-planner-for-high-dimensional.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1177/0278364919890408');" href="https://dx.doi.org/10.1177/0278364919890408"> publisher </a> <button class="btn" type="button" onclick="$('#bibluna2020a-scalable-motion-planner-for-high-dimensional').slideToggle(600);$('#absluna2020a-scalable-motion-planner-for-high-dimensional').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absluna2020a-scalable-motion-planner-for-high-dimensional').slideToggle(600);$('#bibluna2020a-scalable-motion-planner-for-high-dimensional').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibluna2020a-scalable-motion-planner-for-high-dimensional"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">luna2020a-scalable-motion-planner-for-high-dimensional</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Sampling-based algorithms are known for their ability to
        effectively compute paths for high-dimensional robots in relatively
        short times. The same algorithms, however, are also notorious for poor
        quality solution paths, particularly as the dimensionality of the
        system grows. This work proposes a new probabilistically complete
        sampling-based algorithm, XXL, specially designed to plan the motions
        of high-dimensional mobile manipulators and related platforms. Using a
        novel sampling and connection strategy that guides a set of points
        mapped on the robot through the workspace, XXL scales to realistic
        manipulator platforms with dozens of joints by focusing the search of
        the robot's configuration space to specific degrees-of-freedom that
        affect motion in particular portions of the workspace. Simulated
        planning scenarios with the Robonaut2 platform and planar kinematic
        chains confirm that XXL exhibits competitive solution times relative to
        many existing works while obtaining execution-quality solution paths.
        Solutions from XXL are of comparable quality to costaware methods even
        though XXL does not explicitly optimize over any particular criteria,
        and are computed in an order of magnitude less time. Furthermore,
        observations about the performance of sampling-based algorithms on
        high-dimensional manipulator planning problems are presented that
        reveal a cautionary tale regarding two popular guiding heuristics used
        in these algorithms, indicating that a nearly random search may
        outperform the state-of-the-art when defining such heuristics is known
        to be difficult."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Luna, Ryan and Moll, Mark and Badger, Julia M. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1177/0278364919890408"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Intl.\ J.\ of Robotics Research"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"4"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"361--388"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"A Scalable Motion Planner for High-Dimensional Kinematic Systems"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"39"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2020"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absluna2020a-scalable-motion-planner-for-high-dimensional"><h3>Abstract</h3><p>Sampling-based algorithms are known for their ability to effectively compute paths for high-dimensional robots in relatively short times. The same algorithms, however, are also notorious for poor quality solution paths, particularly as the dimensionality of the system grows. This work proposes a new probabilistically complete sampling-based algorithm, XXL, specially designed to plan the motions of high-dimensional mobile manipulators and related platforms. Using a novel sampling and connection strategy that guides a set of points mapped on the robot through the workspace, XXL scales to realistic manipulator platforms with dozens of joints by focusing the search of the robot’s configuration space to specific degrees-of-freedom that affect motion in particular portions of the workspace. Simulated planning scenarios with the Robonaut2 platform and planar kinematic chains confirm that XXL exhibits competitive solution times relative to many existing works while obtaining execution-quality solution paths. Solutions from XXL are of comparable quality to costaware methods even though XXL does not explicitly optimize over any particular criteria, and are computed in an order of magnitude less time. Furthermore, observations about the performance of sampling-based algorithms on high-dimensional manipulator planning problems are presented that reveal a cautionary tale regarding two popular guiding heuristics used in these algorithms, indicating that a nearly random search may outperform the state-of-the-art when defining such heuristics is known to be difficult.</p></div></div></li><li> <span id="antunes2020hla-arena">D. A. Antunes, J. R. Abella, S. Hall-Swan, D. Devaurs, A. Conev, <b>M. Moll</b>, G. Lizée, and L. E. Kavraki, “HLA-Arena: A customizable environment for the structural modeling and analysis of peptide-HLA complexes for cancer immunotherapy,” <i>JCO Clinical Cancer Informatics</i>, vol. 4, pp. 623–636, 2020.</span> <br><div id="accantunes2020hla-arena"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/antunes2020hla-arena.pdf');" href="/publications/antunes2020hla-arena.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1200/CCI.19.00123');" href="https://dx.doi.org/10.1200/CCI.19.00123"> publisher </a> <button class="btn" type="button" onclick="$('#bibantunes2020hla-arena').slideToggle(600);$('#absantunes2020hla-arena').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absantunes2020hla-arena').slideToggle(600);$('#bibantunes2020hla-arena').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibantunes2020hla-arena"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">antunes2020hla-arena</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"PURPOSE HLA protein receptors play a key role in cellular
        immunity. They bind intracellular peptides and display them for
        recognition by T-cell lymphocytes. Because T-cell activation is
        partially driven by structural features of these peptide-HLA complexes,
        their structural modeling and analysis are becoming central components
        of cancer immunotherapy projects. Unfortunately, this kind of analysis
        is limited by the small number of experimentally determined structures
        of peptide-HLA complexes. Overcoming this limitation requires
        developing novel computational methods to model and analyze peptide-HLA
        structures.
        
        METHODS Here we describe a new platform for the structural modeling and
        analysis of peptide-HLA complexes, called HLA-Arena, which we have
        implemented using Jupyter Notebook and Docker. It is a customizable
        environment that facilitates the use of computational tools, such as
        APE-Gen and DINC, which we have previously applied to peptide-HLA
        complexes. By integrating other commonly used tools, such as Modeler
        and MHCflurry, this environment includes support for diverse tasks in
        structural modeling, analysis, and visualization.
        
        RESULTS To illustrate the capabilities of HLA-Arena, we describe 3
        example workflows applied to peptide-HLA complexes. Leveraging the
        strengths of our tools, DINC and APE-Gen, the first 2 workflows show
        how to perform geometry prediction for peptide-HLA complexes and
        structure-based binding prediction, respectively. The third workflow
        presents an example of large-scale virtual screening of peptides for
        multiple HLA alleles.
        
        CONCLUSION These workflows illustrate the potential benefits of
        HLA-Arena for the structural modeling and analysis of peptide-HLA
        complexes. Because HLA-Arena can easily be integrated within larger
        computational pipelines, we expect its potential impact to vastly
        increase. For instance, it could be used to conduct structural analyses
        for personalized cancer immunotherapy, neoantigen discovery, or vaccine
        development."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Antunes, Dinler A. and Abella, Jayvee R. and Hall-Swan, Sarah and Devaurs, Didier and Conev, Anja and Moll, Mark and Liz{\'e}e, Gregory and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1200/CCI.19.00123"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"{JCO} Clinical Cancer Informatics"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"Human leukocyte antigen (HLA); Peptide-HLA complex; Homology
        modeling; Molecular docking; Structural modeling; Structural analysis;
        Cancer immunotherapy"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"623--636"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"{HLA-Arena}: A customizable environment for the structural modeling
        and analysis of peptide-{HLA} complexes for cancer immunotherapy"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"4"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2020"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absantunes2020hla-arena"><h3>Abstract</h3><p>PURPOSE HLA protein receptors play a key role in cellular immunity. They bind intracellular peptides and display them for recognition by T-cell lymphocytes. Because T-cell activation is partially driven by structural features of these peptide-HLA complexes, their structural modeling and analysis are becoming central components of cancer immunotherapy projects. Unfortunately, this kind of analysis is limited by the small number of experimentally determined structures of peptide-HLA complexes. Overcoming this limitation requires developing novel computational methods to model and analyze peptide-HLA structures. METHODS Here we describe a new platform for the structural modeling and analysis of peptide-HLA complexes, called HLA-Arena, which we have implemented using Jupyter Notebook and Docker. It is a customizable environment that facilitates the use of computational tools, such as APE-Gen and DINC, which we have previously applied to peptide-HLA complexes. By integrating other commonly used tools, such as Modeler and MHCflurry, this environment includes support for diverse tasks in structural modeling, analysis, and visualization. RESULTS To illustrate the capabilities of HLA-Arena, we describe 3 example workflows applied to peptide-HLA complexes. Leveraging the strengths of our tools, DINC and APE-Gen, the first 2 workflows show how to perform geometry prediction for peptide-HLA complexes and structure-based binding prediction, respectively. The third workflow presents an example of large-scale virtual screening of peptides for multiple HLA alleles. CONCLUSION These workflows illustrate the potential benefits of HLA-Arena for the structural modeling and analysis of peptide-HLA complexes. Because HLA-Arena can easily be integrated within larger computational pipelines, we expect its potential impact to vastly increase. For instance, it could be used to conduct structural analyses for personalized cancer immunotherapy, neoantigen discovery, or vaccine development.</p></div></div></li><li> <span id="kingston2019exploring-implicit-spaces-for-constrained">Z. Kingston, <b>M. Moll</b>, and L. E. Kavraki, “Exploring Implicit Spaces for Constrained Sampling-Based Planning,” <i>Intl. J. of Robotics Research</i>, vol. 38, no. 10–11, pp. 1151–1178, Sep. 2019.</span> <br><div id="acckingston2019exploring-implicit-spaces-for-constrained"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/kingston2019exploring-implicit-spaces-for-constrained.pdf');" href="/publications/kingston2019exploring-implicit-spaces-for-constrained.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1177/0278364919868530');" href="https://dx.doi.org/10.1177/0278364919868530"> publisher </a> <button class="btn" type="button" onclick="$('#bibkingston2019exploring-implicit-spaces-for-constrained').slideToggle(600);$('#abskingston2019exploring-implicit-spaces-for-constrained').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abskingston2019exploring-implicit-spaces-for-constrained').slideToggle(600);$('#bibkingston2019exploring-implicit-spaces-for-constrained').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibkingston2019exploring-implicit-spaces-for-constrained"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">kingston2019exploring-implicit-spaces-for-constrained</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a review and reformulation of manifold constrained
        sampling-based motion planning within a unifying framework, IMACS
        (Implicit MAnifold Configuration Space). IMACS enables a broad class of
        motion planners to plan in the presence of manifold constraints,
        decoupling the choice of motion planning algorithm and method for
        constraint adherence into orthogonal choices. We show that implicit
        configuration spaces defined by constraints can be presented to
        sampling-based planners by addressing two key fundamental primitives:
        sampling and local planning, and that IMACS preserves theoretical
        properties of probabilistic completeness and asymptotic optimality
        through these primitives. Within IMACS, we implement projection- and
        continutation-based methods for constraint adherence, and demonstrate
        the framework on a range of planners with both methods in simulated and
        realistic scenarios. Our results show that the choice of method for
        constraint adherence depends on many factors and that novel
        combinations of planners and methods of constraint adherence can be
        more effective than previous approaches. Our implementation of IMACS is
        open source within the Open Motion Planning Library and is easily
        extended for novel planners and constraint spaces."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Kingston, Zachary and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1177/0278364919868530"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Intl.\ J.\ of Robotics Research"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"Constrained Planning, Manipulation Planning, Sampling-Based
        Motion Planning, Manifold Constraints, End-Effector Constraints, Loop
        Closure Constraints, Motion Planning"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"10--11"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"1151--1178"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Exploring Implicit Spaces for Constrained Sampling-Based Planning"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"38"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2019"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abskingston2019exploring-implicit-spaces-for-constrained"><h3>Abstract</h3><p>We present a review and reformulation of manifold constrained sampling-based motion planning within a unifying framework, IMACS (Implicit MAnifold Configuration Space). IMACS enables a broad class of motion planners to plan in the presence of manifold constraints, decoupling the choice of motion planning algorithm and method for constraint adherence into orthogonal choices. We show that implicit configuration spaces defined by constraints can be presented to sampling-based planners by addressing two key fundamental primitives: sampling and local planning, and that IMACS preserves theoretical properties of probabilistic completeness and asymptotic optimality through these primitives. Within IMACS, we implement projection- and continutation-based methods for constraint adherence, and demonstrate the framework on a range of planners with both methods in simulated and realistic scenarios. Our results show that the choice of method for constraint adherence depends on many factors and that novel combinations of planners and methods of constraint adherence can be more effective than previous approaches. Our implementation of IMACS is open source within the Open Motion Planning Library and is easily extended for novel planners and constraint spaces.</p></div></div></li><li> <span id="devaurs2019using-parallelized-incremental-meta-docking">D. Devaurs, D. A. Antunes, S. Hall-Swan, N. Mitchell, <b>M. Moll</b>, G. Lizée, and L. E. Kavraki, “Using parallelized incremental meta-docking can solve the conformational sampling issue when docking large ligands to proteins,” <i>BMC Molecular and Cell Biology</i>, vol. 20, no. 1, p. 42, Sep. 2019.</span> <br><div id="accdevaurs2019using-parallelized-incremental-meta-docking"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/devaurs2019using-parallelized-incremental-meta-docking.pdf');" href="/publications/devaurs2019using-parallelized-incremental-meta-docking.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1186/s12860-019-0218-z');" href="https://dx.doi.org/10.1186/s12860-019-0218-z"> publisher </a> <button class="btn" type="button" onclick="$('#bibdevaurs2019using-parallelized-incremental-meta-docking').slideToggle(600);$('#absdevaurs2019using-parallelized-incremental-meta-docking').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absdevaurs2019using-parallelized-incremental-meta-docking').slideToggle(600);$('#bibdevaurs2019using-parallelized-incremental-meta-docking').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibdevaurs2019using-parallelized-incremental-meta-docking"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">devaurs2019using-parallelized-incremental-meta-docking</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Background: Docking large ligands, and especially peptides, to
        protein receptors is still considered a challenge in computational
        structural biology. Besides the issue of accurately scoring the binding
        modes of a protein-ligand complex produced by a molecular docking tool,
        the conformational sampling of a large ligand is also often considered
        a challenge because of its underlying combinatorial complexity. In this
        study, we evaluate the impact of using parallelized and incremental
        paradigms on the accuracy and performance of conformational sampling
        when docking large ligands. We use five datasets of protein-ligand
        complexes involving ligands that could not be accurately docked by
        classical protein-ligand docking tools in previous similar studies.
        
        Results: Our computational evaluation shows that simply increasing the
        amount of conformational sampling performed by a protein-ligand docking
        tool, such as Vina, by running it for longer is rarely beneficial.
        Instead, it is more efficient and advantageous to run several short
        instances of this docking tool in parallel and group their results
        together, in a straightforward parallelized docking protocol. Even
        greater accuracy and efficiency are achieved by our parallelized
        incremental meta-docking tool, DINC, showing the additional benefits of
        its incremental paradigm. Using DINC, we could accurately reproduce the
        vast majority of the protein-ligand complexes we considered.
        
        Conclusions: Our study suggests that, even when trying to dock large
        ligands to proteins, the conformational sampling of the ligand should
        no longer be considered an issue, as simple docking protocols using
        existing tools can solve it. Therefore, scoring should currently be
        regarded as the biggest unmet challenge in molecular docking."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Devaurs, Didier and Antunes, Dinler A and Hall-Swan, Sarah and Mitchell, Nicole and Moll, Mark and Liz{\'e}e, Gregory and Kavraki, Lydia E"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1186/s12860-019-0218-z"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"BMC Molecular and Cell Biology"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"molecular docking; protein-ligand docking; protein-peptide
        docking; conformational sampling; scoring; parallelism; incremental
        protocol"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"42"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Using parallelized incremental meta-docking can solve the
        conformational sampling issue when docking large ligands to proteins"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"20"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2019"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absdevaurs2019using-parallelized-incremental-meta-docking"><h3>Abstract</h3><p>Background: Docking large ligands, and especially peptides, to protein receptors is still considered a challenge in computational structural biology. Besides the issue of accurately scoring the binding modes of a protein-ligand complex produced by a molecular docking tool, the conformational sampling of a large ligand is also often considered a challenge because of its underlying combinatorial complexity. In this study, we evaluate the impact of using parallelized and incremental paradigms on the accuracy and performance of conformational sampling when docking large ligands. We use five datasets of protein-ligand complexes involving ligands that could not be accurately docked by classical protein-ligand docking tools in previous similar studies. Results: Our computational evaluation shows that simply increasing the amount of conformational sampling performed by a protein-ligand docking tool, such as Vina, by running it for longer is rarely beneficial. Instead, it is more efficient and advantageous to run several short instances of this docking tool in parallel and group their results together, in a straightforward parallelized docking protocol. Even greater accuracy and efficiency are achieved by our parallelized incremental meta-docking tool, DINC, showing the additional benefits of its incremental paradigm. Using DINC, we could accurately reproduce the vast majority of the protein-ligand complexes we considered. Conclusions: Our study suggests that, even when trying to dock large ligands to proteins, the conformational sampling of the ligand should no longer be considered an issue, as simple docking protocols using existing tools can solve it. Therefore, scoring should currently be regarded as the biggest unmet challenge in molecular docking.</p></div></div></li><li> <span id="hernandez2019online-motion-planning-for-unexplored">J. D. Hernández, E. Vidal, <b>M. Moll</b>, N. Palomeras, M. Carreras, and L. E. Kavraki, “Online Motion Planning for Unexplored Underwater Environments using Autonomous Underwater Vehicles,” <i>Journal of Field Robotics</i>, vol. 36, no. 2, pp. 370–396, Mar. 2019.</span> <br><div id="acchernandez2019online-motion-planning-for-unexplored"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/hernandez2019online-motion-planning-for-unexplored.pdf');" href="/publications/hernandez2019online-motion-planning-for-unexplored.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1002/rob.21827');" href="https://dx.doi.org/10.1002/rob.21827"> publisher </a> <button class="btn" type="button" onclick="$('#bibhernandez2019online-motion-planning-for-unexplored').slideToggle(600);$('#abshernandez2019online-motion-planning-for-unexplored').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abshernandez2019online-motion-planning-for-unexplored').slideToggle(600);$('#bibhernandez2019online-motion-planning-for-unexplored').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibhernandez2019online-motion-planning-for-unexplored"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">hernandez2019online-motion-planning-for-unexplored</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present an approach to endow an autonomous underwater vehicle
        (AUV) with the capabilities to move through unexplored environments. To
        do so, we propose a computational framework for planning feasible and
        safe paths. The framework allows the vehicle to incrementally build a
        map of the surroundings, while simultaneously (re)planning a feasible
        path to a specified goal. To accomplish this, the framework considers
        motion constraints to plan feasible 3D paths, i.e., those that meet the
        vehicle's motion capabilities. It also incorporates a risk function to
        avoid navigating close to nearby obstacles. Furthermore, the framework
        makes use of two strategies to ensure meeting online computation
        limitations. The first one is to reuse the last best known solution to
        eliminate time-consuming pruning routines. The second one is to
        opportunistically check the states' risk of collision.
        
        To evaluate the proposed approach, we use the Sparus II performing
        autonomous missions in different real-world scenarios. These
        experiments consist of simulated and in-water trials for different
        tasks. The conducted tasks include the exploration of challenging
        scenarios such as artificial marine structures, natural marine
        structures, and confined natural environments. All these applications
        allow us to extensively prove the efficacy of the presented approach,
        not only for constant-depth missions (2D), but, more importantly, for
        situations in which the vehicle must vary its depth (3D)."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Hern{\'a}ndez, Juan David and Vidal, Eduard and Moll, Mark and Palomeras, N. and Carreras, Marc and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1002/rob.21827"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Journal of Field Robotics"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"2"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"370--396"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Online Motion Planning for Unexplored Underwater Environments using
        Autonomous Underwater Vehicles"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"36"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2019"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abshernandez2019online-motion-planning-for-unexplored"><h3>Abstract</h3><p>We present an approach to endow an autonomous underwater vehicle (AUV) with the capabilities to move through unexplored environments. To do so, we propose a computational framework for planning feasible and safe paths. The framework allows the vehicle to incrementally build a map of the surroundings, while simultaneously (re)planning a feasible path to a specified goal. To accomplish this, the framework considers motion constraints to plan feasible 3D paths, i.e., those that meet the vehicle’s motion capabilities. It also incorporates a risk function to avoid navigating close to nearby obstacles. Furthermore, the framework makes use of two strategies to ensure meeting online computation limitations. The first one is to reuse the last best known solution to eliminate time-consuming pruning routines. The second one is to opportunistically check the states’ risk of collision. To evaluate the proposed approach, we use the Sparus II performing autonomous missions in different real-world scenarios. These experiments consist of simulated and in-water trials for different tasks. The conducted tasks include the exploration of challenging scenarios such as artificial marine structures, natural marine structures, and confined natural environments. All these applications allow us to extensively prove the efficacy of the presented approach, not only for constant-depth missions (2D), but, more importantly, for situations in which the vehicle must vary its depth (3D).</p></div></div></li><li> <span id="litsa2019machine-learning-guided-atom">E. E. Litsa, M. I. Peña, <b>M. Moll</b>, G. Giannakopoulos, G. N. Bennett, and L. E. Kavraki, “Machine Learning Guided Atom Mapping of Metabolic Reactions,” <i>Journal of Chemical Information and Modeling</i>, vol. 59, no. 3, pp. 1121–1135, Mar. 2019.</span> <br><div id="acclitsa2019machine-learning-guided-atom"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/litsa2019machine-learning-guided-atom.pdf');" href="/publications/litsa2019machine-learning-guided-atom.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1021/acs.jcim.8b00434');" href="https://dx.doi.org/10.1021/acs.jcim.8b00434"> publisher </a> <button class="btn" type="button" onclick="$('#biblitsa2019machine-learning-guided-atom').slideToggle(600);$('#abslitsa2019machine-learning-guided-atom').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abslitsa2019machine-learning-guided-atom').slideToggle(600);$('#biblitsa2019machine-learning-guided-atom').slideUp(600);"> abstract </button></div><div class="card collapse" id="biblitsa2019machine-learning-guided-atom"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">litsa2019machine-learning-guided-atom</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Atom mapping of a chemical reaction is a mapping between the
        atoms in the reactant molecules and the atoms in the product molecules.
        It encodes the underlying reaction mechanism and, as such, constitutes
        essential information in computational studies in metabolic
        engineering. Various techniques have been investigated for the
        automatic computation of the atom mapping of a chemical reaction,
        approaching the problem as a graph matching problem. The graph
        abstraction of the chemical problem, though, eliminates crucial
        chemical information. There have been efforts for enhancing the graph
        representation by introducing the bond stabilities as edge weights, as
        they are estimated based on experimental evidence. Here, we present a
        fully automated optimization-based approach, named AMLGAM, (Automated
        Machine Learning Guided Atom Mapping), that uses machine learning
        techniques for the estimation of the bond stabilities based on the
        chemical environment of each bond. The optimization method finds the
        reaction mechanism which favors the breakage/formation of the less
        stable bonds. We evaluated our method on a manually curated data set of
        382 chemical reactions and ran our method on a much larger and diverse
        data set of 7400 chemical reactions. We show that the proposed method
        improves the accuracy over existing techniques based on results
        published by earlier studies on a common data set and is capable of
        handling unbalanced reactions."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Litsa, Eleni E. and Pe{\~n}a, Matthew I. and Moll, Mark and Giannakopoulos, George and Bennett, George N. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1021/acs.jcim.8b00434"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Journal of Chemical Information and Modeling"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"3"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"1121--1135"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Machine Learning Guided Atom Mapping of Metabolic Reactions"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"59"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2019"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abslitsa2019machine-learning-guided-atom"><h3>Abstract</h3><p>Atom mapping of a chemical reaction is a mapping between the atoms in the reactant molecules and the atoms in the product molecules. It encodes the underlying reaction mechanism and, as such, constitutes essential information in computational studies in metabolic engineering. Various techniques have been investigated for the automatic computation of the atom mapping of a chemical reaction, approaching the problem as a graph matching problem. The graph abstraction of the chemical problem, though, eliminates crucial chemical information. There have been efforts for enhancing the graph representation by introducing the bond stabilities as edge weights, as they are estimated based on experimental evidence. Here, we present a fully automated optimization-based approach, named AMLGAM, (Automated Machine Learning Guided Atom Mapping), that uses machine learning techniques for the estimation of the bond stabilities based on the chemical environment of each bond. The optimization method finds the reaction mechanism which favors the breakage/formation of the less stable bonds. We evaluated our method on a manually curated data set of 382 chemical reactions and ran our method on a much larger and diverse data set of 7400 chemical reactions. We show that the proposed method improves the accuracy over existing techniques based on results published by earlier studies on a common data set and is capable of handling unbalanced reactions.</p></div></div></li><li> <span id="kingston2018sampling-based-methods-for-motion-planning">Z. Kingston, <b>M. Moll</b>, and L. E. Kavraki, “Sampling-Based Methods for Motion Planning with Constraints,” <i>Annual Review of Control, Robotics, and Autonomous Systems</i>, vol. 1, pp. 159–185, May 2018.</span> <br><div id="acckingston2018sampling-based-methods-for-motion-planning"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/kingston2018sampling-based-methods-for-motion-planning.pdf');" href="/publications/kingston2018sampling-based-methods-for-motion-planning.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1146/annurev-control-060117-105226');" href="https://dx.doi.org/10.1146/annurev-control-060117-105226"> publisher </a> <button class="btn" type="button" onclick="$('#bibkingston2018sampling-based-methods-for-motion-planning').slideToggle(600);$('#abskingston2018sampling-based-methods-for-motion-planning').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abskingston2018sampling-based-methods-for-motion-planning').slideToggle(600);$('#bibkingston2018sampling-based-methods-for-motion-planning').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibkingston2018sampling-based-methods-for-motion-planning"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">kingston2018sampling-based-methods-for-motion-planning</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Robots with many degrees of freedom (e.g., humanoid robots and
        mobile manipulators) have increasingly been employed to accomplish
        realistic tasks in domains such as disaster relief, spacecraft
        logistics, and home caretaking. Finding feasible motions for these
        robots autonomously is essential for their operation. Sampling-based
        motion planning algorithms have been shown to be effective for these
        high-dimensional systems. However, incorporating task constraints
        (e.g., keeping a cup level, writing on a board) into the planning
        process introduces significant challenges. is survey describes the
        families of methods for sampling-based planning with constraints and
        places them on a spectrum delineated by their complexity. Constrained
        sampling-based methods are based upon two core primitive operations:
        (1) sampling constraint-satisfying configurations and (2) generating
        constraint-satisfying continuous motion. Although the basics of
        sampling-based planning are presented for contextual background, the
        survey focuses on the representation of constraints and sampling- based
        planners that incorporate constraints."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Kingston, Zachary and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1146/annurev-control-060117-105226"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Annual Review of Control, Robotics, and Autonomous Systems"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"robotics, robot motion planning, sampling-based planning,
        constraints, planning with constraints, planning for high-dimensional
        robotic systems"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">may</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"159--185"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Sampling-Based Methods for Motion Planning with Constraints"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"1"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2018"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abskingston2018sampling-based-methods-for-motion-planning"><h3>Abstract</h3><p>Robots with many degrees of freedom (e.g., humanoid robots and mobile manipulators) have increasingly been employed to accomplish realistic tasks in domains such as disaster relief, spacecraft logistics, and home caretaking. Finding feasible motions for these robots autonomously is essential for their operation. Sampling-based motion planning algorithms have been shown to be effective for these high-dimensional systems. However, incorporating task constraints (e.g., keeping a cup level, writing on a board) into the planning process introduces significant challenges. is survey describes the families of methods for sampling-based planning with constraints and places them on a spectrum delineated by their complexity. Constrained sampling-based methods are based upon two core primitive operations: (1) sampling constraint-satisfying configurations and (2) generating constraint-satisfying continuous motion. Although the basics of sampling-based planning are presented for contextual background, the survey focuses on the representation of constraints and sampling- based planners that incorporate constraints.</p></div></div></li><li> <span id="muhayyuddin2018randomized-physics-based-motion-planning">Muhayyuddin, <b>M. Moll</b>, L. E. Kavraki, and J. Rosell, “Randomized Physics-based Motion Planning for Grasping in Cluttered and Uncertain Environments,” <i>IEEE Robotics and Automation Letters</i>, vol. 3, no. 2, pp. 712–719, Apr. 2018.</span> <br><div id="accmuhayyuddin2018randomized-physics-based-motion-planning"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/muhayyuddin2018randomized-physics-based-motion-planning.pdf');" href="/publications/muhayyuddin2018randomized-physics-based-motion-planning.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/LRA.2017.2783445');" href="https://dx.doi.org/10.1109/LRA.2017.2783445"> publisher </a> <button class="btn" type="button" onclick="$('#bibmuhayyuddin2018randomized-physics-based-motion-planning').slideToggle(600);$('#absmuhayyuddin2018randomized-physics-based-motion-planning').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmuhayyuddin2018randomized-physics-based-motion-planning').slideToggle(600);$('#bibmuhayyuddin2018randomized-physics-based-motion-planning').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmuhayyuddin2018randomized-physics-based-motion-planning"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">muhayyuddin2018randomized-physics-based-motion-planning</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Planning motions to grasp an object in cluttered and uncertain
        environments is a challenging task, particularly when a collision-free
        trajectory does not exist and objects obstructing the way are required
        to be carefully grasped and moved out. This paper takes a different
        approach and proposes to address this problem by using a randomized
        physics-based motion planner that permits robot-object and
        object-object interactions. The main idea is to avoid an explicit
        high-level reasoning of the task by providing the motion planner with a
        physics engine to evaluate possible complex multi-body dynamical
        interactions. The approach is able to solve the problem in complex
        scenarios, also considering uncertainty in the objects' pose and in the
        contact dynamics. The work enhances the state validity checker, the
        control sampler and the tree exploration strategy of a kinodynamic
        motion planner called KPIECE. The enhanced algorithm, called p-KPIECE,
        has been validated in simulation and with real experiments. The results
        have been compared with an ontological physics-based motion planner and
        with task and motion planning approaches, resulting in a significant
        improvement in terms of planning time, success rate and quality of the
        solution path."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Muhayyuddin and Moll, Mark and Kavraki, Lydia E. and Rosell, Jan"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/LRA.2017.2783445"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"IEEE Robotics and Automation Letters"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"Physics-based motion planning, planning under uncertainties,
        clutter grasping"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">apr</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"2"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"712--719"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Randomized Physics-based Motion Planning for Grasping in Cluttered
        and Uncertain Environments"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"3"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2018"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmuhayyuddin2018randomized-physics-based-motion-planning"><h3>Abstract</h3><p>Planning motions to grasp an object in cluttered and uncertain environments is a challenging task, particularly when a collision-free trajectory does not exist and objects obstructing the way are required to be carefully grasped and moved out. This paper takes a different approach and proposes to address this problem by using a randomized physics-based motion planner that permits robot-object and object-object interactions. The main idea is to avoid an explicit high-level reasoning of the task by providing the motion planner with a physics engine to evaluate possible complex multi-body dynamical interactions. The approach is able to solve the problem in complex scenarios, also considering uncertainty in the objects’ pose and in the contact dynamics. The work enhances the state validity checker, the control sampler and the tree exploration strategy of a kinodynamic motion planner called KPIECE. The enhanced algorithm, called p-KPIECE, has been validated in simulation and with real experiments. The results have been compared with an ontological physics-based motion planner and with task and motion planning approaches, resulting in a significant improvement in terms of planning time, success rate and quality of the solution path.</p></div></div></li><li> <span id="devaurs2018native-state-of-complement-protein">D. Devaurs, M. Papanastasiou, D. Antunes, J. Abella, <b>M. Moll</b>, D. Ricklin, J. Lambris, and L. E. Kavraki, “Native State of Complement Protein C3d Analyzed via Hydrogen Exchange and Conformational Sampling,” <i>Intl. J. of Computational Biology and Drug Design</i>, vol. 11, no. 1/2, pp. 90–113, Mar. 2018.</span> Special issue on the 2016 Intl. Conf. on Intelligent Biology and Medicine (ICIBM). <br><div id="accdevaurs2018native-state-of-complement-protein"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/devaurs2018native-state-of-complement-protein.pdf');" href="/publications/devaurs2018native-state-of-complement-protein.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1504/IJCBDD.2018.10011903');" href="https://dx.doi.org/10.1504/IJCBDD.2018.10011903"> publisher </a> <button class="btn" type="button" onclick="$('#bibdevaurs2018native-state-of-complement-protein').slideToggle(600);$('#absdevaurs2018native-state-of-complement-protein').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absdevaurs2018native-state-of-complement-protein').slideToggle(600);$('#bibdevaurs2018native-state-of-complement-protein').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibdevaurs2018native-state-of-complement-protein"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">devaurs2018native-state-of-complement-protein</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Hydrogen/deuterium exchange detected by mass spectrometry
        (HDX-MS) provides valuable information on protein structure and
        dynamics. Although HDX-MS data is often interpreted using crystal
        structures, it was suggested that conformational ensembles produced by
        molecular dynamics simulations yield more accurate interpretations. In
        this paper, we analyse the complement protein C3d through HDX-MS data
        and evaluate several interpretation methodologies, using an existing
        prediction model to derive HDX-MS data from protein structure. We
        perform an HDX-MS experiment on C3d and, then, to interpret and refine
        the obtained data, we look for a conformation (or conformational
        ensemble) of C3d that allows computationally replicating this data.
        First, we confirm that crystal structures are not a good choice.
        Second, we suggest that conformational ensembles produced by molecular
        dynamics simulations might not always be satisfactory either. Finally,
        we show that coarse-grained conformational sampling of C3d produces a
        conformation from which the HDX-MS data can be replicated and
        refined."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Devaurs, Didier and Papanastasiou, Malvina and Antunes, Dinler and Abella, Jayvee and Moll, Mark and Ricklin, Daniel and Lambris, John and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1504/IJCBDD.2018.10011903"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Intl.\ J.\ of Computational Biology and Drug Design"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"complement protein C3d; hydrogen exchange; mass spectrometry;
        protein conformational sampling; coarse-grained conformational
        sampling; native state; X-ray crystallography; molecular dynamics;
        protein structures; conformational ensembles"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">note</span> <span class="p">=</span> <span class="s">"Special issue on the 2016 Intl.\ Conf.\ on Intelligent Biology and
        Medicine (ICIBM)."</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1/2"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"90--113"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Native State of Complement Protein {C3d} Analyzed via Hydrogen
        Exchange and Conformational Sampling"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"11"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2018"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absdevaurs2018native-state-of-complement-protein"><h3>Abstract</h3><p>Hydrogen/deuterium exchange detected by mass spectrometry (HDX-MS) provides valuable information on protein structure and dynamics. Although HDX-MS data is often interpreted using crystal structures, it was suggested that conformational ensembles produced by molecular dynamics simulations yield more accurate interpretations. In this paper, we analyse the complement protein C3d through HDX-MS data and evaluate several interpretation methodologies, using an existing prediction model to derive HDX-MS data from protein structure. We perform an HDX-MS experiment on C3d and, then, to interpret and refine the obtained data, we look for a conformation (or conformational ensemble) of C3d that allows computationally replicating this data. First, we confirm that crystal structures are not a good choice. Second, we suggest that conformational ensembles produced by molecular dynamics simulations might not always be satisfactory either. Finally, we show that coarse-grained conformational sampling of C3d produces a conformation from which the HDX-MS data can be replicated and refined.</p></div></div></li><li> <span id="antunes2018general-prediction-of-peptide-mhc-binding">D. A. Antunes, D. Devaurs, <b>M. Moll</b>, G. Lizée, and L. E. Kavraki, “General Prediction of Peptide-MHC Binding Modes Using Incremental Docking: A Proof of Concept,” <i>Scientific Reports</i>, vol. 8, no. 1, p. 4327, Mar. 2018.</span> <br><div id="accantunes2018general-prediction-of-peptide-mhc-binding"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/antunes2018general-prediction-of-peptide-mhc-binding.pdf');" href="/publications/antunes2018general-prediction-of-peptide-mhc-binding.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1038/s41598-018-22173-4');" href="https://dx.doi.org/10.1038/s41598-018-22173-4"> publisher </a> <button class="btn" type="button" onclick="$('#bibantunes2018general-prediction-of-peptide-mhc-binding').slideToggle(600);$('#absantunes2018general-prediction-of-peptide-mhc-binding').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absantunes2018general-prediction-of-peptide-mhc-binding').slideToggle(600);$('#bibantunes2018general-prediction-of-peptide-mhc-binding').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibantunes2018general-prediction-of-peptide-mhc-binding"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">antunes2018general-prediction-of-peptide-mhc-binding</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The class I major histocompatibility complex (MHC) is capable of
        binding peptides derived from intracellular proteins and displaying
        them at the cell surface. The recognition of these peptide-MHC (pMHC)
        complexes by T-cells is the cornerstone of cellular immunity, enabling
        the elimination of infected or tumoral cells. T-cell-based
        immunotherapies against cancer, which leverage this mechanism, can
        greatly benefit from structural analyses of pMHC complexes. Several
        attempts have been made to use molecular docking for such analyses, but
        pMHC structure remains too challenging for even state-of-the-art
        docking tools. To overcome these limitations, we describe the use of an
        incremental meta-docking approach for structural prediction of pMHC
        complexes. Previous methods applied in this context used specific
        constraints to reduce the complexity of this prediction problem, at the
        expense of generality. Our strategy makes no assumption and can
        potentially be used to predict binding modes for any pMHC complex. Our
        method has been tested in a re-docking experiment, reproducing the
        binding modes of 25 pMHC complexes whose crystal structures are
        available. This study is a proof of concept that incremental docking
        strategies can lead to general geometry prediction of pMHC complexes,
        with potential applications for immunotherapy against cancer or
        infectious diseases."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Antunes, Dinler A. and Devaurs, Didier and Moll, Mark and Liz{\'e}e, Gregory and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1038/s41598-018-22173-4"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Scientific Reports"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"Peptide-MHC, HLA, molecular docking, geometry prediction, cancer
        immunotherapy"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"4327"</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">"Nature Publishing Group"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"General Prediction of Peptide-{MHC} Binding Modes Using Incremental
        Docking: A Proof of Concept"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"8"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2018"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absantunes2018general-prediction-of-peptide-mhc-binding"><h3>Abstract</h3><p>The class I major histocompatibility complex (MHC) is capable of binding peptides derived from intracellular proteins and displaying them at the cell surface. The recognition of these peptide-MHC (pMHC) complexes by T-cells is the cornerstone of cellular immunity, enabling the elimination of infected or tumoral cells. T-cell-based immunotherapies against cancer, which leverage this mechanism, can greatly benefit from structural analyses of pMHC complexes. Several attempts have been made to use molecular docking for such analyses, but pMHC structure remains too challenging for even state-of-the-art docking tools. To overcome these limitations, we describe the use of an incremental meta-docking approach for structural prediction of pMHC complexes. Previous methods applied in this context used specific constraints to reduce the complexity of this prediction problem, at the expense of generality. Our strategy makes no assumption and can potentially be used to predict binding modes for any pMHC complex. Our method has been tested in a re-docking experiment, reproducing the binding modes of 25 pMHC complexes whose crystal structures are available. This study is a proof of concept that incremental docking strategies can lead to general geometry prediction of pMHC complexes, with potential applications for immunotherapy against cancer or infectious diseases.</p></div></div></li><li> <span id="abella2018maintaining-and-enhancing-diversity-of-sampled">J. R. Abella, <b>M. Moll</b>, and L. E. Kavraki, “Maintaining and Enhancing Diversity of Sampled Protein Conformations in Robotics-Inspired Methods,” <i>Journal of Computational Biology</i>, vol. 25, no. 1, pp. 3–20, Jan. 2018.</span> <br><div id="accabella2018maintaining-and-enhancing-diversity-of-sampled"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/abella2018maintaining-and-enhancing-diversity-of-sampled.pdf');" href="/publications/abella2018maintaining-and-enhancing-diversity-of-sampled.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1089/cmb.2017.0164');" href="https://dx.doi.org/10.1089/cmb.2017.0164"> publisher </a> <button class="btn" type="button" onclick="$('#bibabella2018maintaining-and-enhancing-diversity-of-sampled').slideToggle(600);$('#absabella2018maintaining-and-enhancing-diversity-of-sampled').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absabella2018maintaining-and-enhancing-diversity-of-sampled').slideToggle(600);$('#bibabella2018maintaining-and-enhancing-diversity-of-sampled').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibabella2018maintaining-and-enhancing-diversity-of-sampled"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">abella2018maintaining-and-enhancing-diversity-of-sampled</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The ability to efficiently sample structurally diverse protein
        conformations allows one to gain a high-level view of a protein's
        energy landscape. Algorithms from robot motion planning have been used
        for conformational sampling and promote diversity by keeping track of
        ``coverage'' in conformational space based on the local sampling
        density. However, large proteins present special challenges. In
        particular, larger systems require running many concurrent instances of
        these algorithms, but these algorithms can quickly become memory
        intensive because they typically keep previously sampled conformations
        in memory to maintain coverage estimates. Additionally, many of these
        algorithms depend on defining useful perturbation strategies for
        exploring the conformational space, which is a very difficult task for
        large proteins because such systems are typically more constrained and
        exhibit complex motions. In this paper, we introduce two methodologies
        for maintaining and enhancing diversity in robotics-inspired
        conformational sampling. The first method leverages the use of a
        low-dimensional projection to define a global coverage grid that
        maintains coverage across concurrent runs of sampling. The second
        method is an automatic definition of a perturbation strategy through
        readily available flexibility information derived from B-factors,
        secondary structure, and rigidity analysis. Our results show a
        significant increase in the diversity of the conformations sampled for
        proteins consisting of up to 500 residues. The methodologies presented
        in this paper may be vital components for the scalability of
        robotics-inspired approaches. "</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Abella, Jayvee R. and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1089/cmb.2017.0164"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Journal of Computational Biology"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"protein conformational sampling, robotics-inspired sampling,
        perturbation strategies, parallelization"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jan</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"3--20"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Maintaining and Enhancing Diversity of Sampled Protein Conformations
        in Robotics-Inspired Methods"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"25"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2018"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absabella2018maintaining-and-enhancing-diversity-of-sampled"><h3>Abstract</h3><p>The ability to efficiently sample structurally diverse protein conformations allows one to gain a high-level view of a protein’s energy landscape. Algorithms from robot motion planning have been used for conformational sampling and promote diversity by keeping track of “coverage” in conformational space based on the local sampling density. However, large proteins present special challenges. In particular, larger systems require running many concurrent instances of these algorithms, but these algorithms can quickly become memory intensive because they typically keep previously sampled conformations in memory to maintain coverage estimates. Additionally, many of these algorithms depend on defining useful perturbation strategies for exploring the conformational space, which is a very difficult task for large proteins because such systems are typically more constrained and exhibit complex motions. In this paper, we introduce two methodologies for maintaining and enhancing diversity in robotics-inspired conformational sampling. The first method leverages the use of a low-dimensional projection to define a global coverage grid that maintains coverage across concurrent runs of sampling. The second method is an automatic definition of a perturbation strategy through readily available flexibility information derived from B-factors, secondary structure, and rigidity analysis. Our results show a significant increase in the diversity of the conformations sampled for proteins consisting of up to 500 residues. The methodologies presented in this paper may be vital components for the scalability of robotics-inspired approaches.</p></div></div></li><li> <span id="antunes2017dinc2">D. A. Antunes, <b>M. Moll</b>, D. Devaurs, K. Jackson, G. Lizée, and L. E. Kavraki, “DINC 2.0: a new protein-peptide docking webserver using an incremental approach,” <i>Cancer Research</i>, vol. 77, no. 21, pp. e55–57, Nov. 2017.</span> <br><div id="accantunes2017dinc2"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/antunes2017dinc2.pdf');" href="/publications/antunes2017dinc2.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1158/0008-5472.CAN-17-0511');" href="https://dx.doi.org/10.1158/0008-5472.CAN-17-0511"> publisher </a> <button class="btn" type="button" onclick="$('#bibantunes2017dinc2').slideToggle(600);$('#absantunes2017dinc2').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absantunes2017dinc2').slideToggle(600);$('#bibantunes2017dinc2').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibantunes2017dinc2"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">antunes2017dinc2</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Molecular docking is a standard computational approach to predict
        binding modes of protein-ligand complexes, by exploring alternative
        orientations and conformations of the ligand (i.e., by exploring ligand
        flexibility). Docking tools are largely used for virtual screening of
        small drug-like molecules, but their accuracy and efficiency greatly
        decays for ligands with more than 10 flexible bonds. This prevents a
        broader use of these tools to dock larger ligands such as peptides,
        which are molecules of growing interest for cancer research. To
        overcome this limitation, our group has previously proposed a
        meta-docking strategy, called DINC, to predict binding modes of large
        ligands. By incrementally docking overlapping fragments of a ligand,
        DINC allowed predicting binding modes of peptide-based inhibitors of
        transcription factors involved in cancer. Here we describe DINC 2.0, a
        revamped version of the DINC webserver with enhanced capabilities and a
        user-friendly interface. DINC 2.0 allows docking ligands that were
        previously too challenging for DINC, such as peptides with more than 25
        flexible bonds. The webserver is freely accessible at
        http://dinc.kavrakilab.org, together with additional documentation and
        video tutorials. Our team will provide continuous support for this tool
        and is working on extending its applicability to other challenging
        fields, such as personalized immunotherapy against cancer."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Antunes, Dinler A. and Moll, Mark and Devaurs, Didier and Jackson, Kyle and Liz{\'e}e, Gregory and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1158/0008-5472.CAN-17-0511"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Cancer Research"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">nov</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"21"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"e55-57"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"{DINC} 2.0: a new protein-peptide docking webserver using an
        incremental approach"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"77"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2017"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absantunes2017dinc2"><h3>Abstract</h3><p>Molecular docking is a standard computational approach to predict binding modes of protein-ligand complexes, by exploring alternative orientations and conformations of the ligand (i.e., by exploring ligand flexibility). Docking tools are largely used for virtual screening of small drug-like molecules, but their accuracy and efficiency greatly decays for ligands with more than 10 flexible bonds. This prevents a broader use of these tools to dock larger ligands such as peptides, which are molecules of growing interest for cancer research. To overcome this limitation, our group has previously proposed a meta-docking strategy, called DINC, to predict binding modes of large ligands. By incrementally docking overlapping fragments of a ligand, DINC allowed predicting binding modes of peptide-based inhibitors of transcription factors involved in cancer. Here we describe DINC 2.0, a revamped version of the DINC webserver with enhanced capabilities and a user-friendly interface. DINC 2.0 allows docking ligands that were previously too challenging for DINC, such as peptides with more than 25 flexible bonds. The webserver is freely accessible at http://dinc.kavrakilab.org, together with additional documentation and video tutorials. Our team will provide continuous support for this tool and is working on extending its applicability to other challenging fields, such as personalized immunotherapy against cancer.</p></div></div></li><li> <span id="kim2017a-review-of-parameters-and-heuristics-for-guiding">S. M. Kim, M. I. Peña, <b>M. Moll</b>, G. N. Bennett, and L. E. Kavraki, “A Review of Parameters and Heuristics for Guiding Metabolic Pathfinding,” <i>Journal of Cheminformatics</i>, vol. 9, no. 1, p. 51, Sep. 2017.</span> <br><div id="acckim2017a-review-of-parameters-and-heuristics-for-guiding"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/kim2017a-review-of-parameters-and-heuristics-for-guiding.pdf');" href="/publications/kim2017a-review-of-parameters-and-heuristics-for-guiding.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1186/s13321-017-0239-6');" href="https://dx.doi.org/10.1186/s13321-017-0239-6"> publisher </a> <button class="btn" type="button" onclick="$('#bibkim2017a-review-of-parameters-and-heuristics-for-guiding').slideToggle(600);$('#abskim2017a-review-of-parameters-and-heuristics-for-guiding').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abskim2017a-review-of-parameters-and-heuristics-for-guiding').slideToggle(600);$('#bibkim2017a-review-of-parameters-and-heuristics-for-guiding').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibkim2017a-review-of-parameters-and-heuristics-for-guiding"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">kim2017a-review-of-parameters-and-heuristics-for-guiding</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Recent developments in metabolic engineering have led to the
        successful biosynthesis of valuable products, such as the precursor of
        the antimalarial compound, artemisinin, and opioid precursor, thebaine.
        Synthesizing these traditionally plant-derived compounds in genetically
        modified yeast cells introduces the possibility of significantly
        reducing the total time and resources required for their production,
        and in turn, allows these valuable compounds to become cheaper and more
        readily available. Most biosynthesis pathways used in metabolic
        engineering applications have been discovered manually, requiring a
        tedious search of existing literature and metabolic databases. However,
        the recent rapid development of available metabolic information has
        enabled the development of automated approaches for identifying novel
        pathways. Computer-assisted pathfinding has the potential to save
        biochemists time in the initial discovery steps of metabolic
        engineering. In this paper, we review the parameters and heuristics
        used to guide the search in recent pathfinding algorithms. These
        parameters and heuristics capture information on the metabolic network
        structure, compound structures, reaction features, and
        organism-specificity of pathways. No one metabolic pathfinding
        algorithm or search parameter stands out as the best to use broadly for
        solving the pathfinding problem, as each method and parameter has its
        own strengths and shortcomings. As assisted pathfinding approaches
        continue to become more sophisticated, the development of better
        methods for visualizing pathway results and integrating these results
        into existing metabolic engineering practices is also important for
        encouraging wider use of these pathfinding methods."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Kim, Sarah M. and Pe{\~n}a, Matthew I. and Moll, Mark and Bennett, George N. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1186/s13321-017-0239-6"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Journal of Cheminformatics"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"metabolic pathfinding; graph-based search; metabolic
        engineering"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"51"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"A Review of Parameters and Heuristics for Guiding Metabolic
        Pathfinding"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"9"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2017"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abskim2017a-review-of-parameters-and-heuristics-for-guiding"><h3>Abstract</h3><p>Recent developments in metabolic engineering have led to the successful biosynthesis of valuable products, such as the precursor of the antimalarial compound, artemisinin, and opioid precursor, thebaine. Synthesizing these traditionally plant-derived compounds in genetically modified yeast cells introduces the possibility of significantly reducing the total time and resources required for their production, and in turn, allows these valuable compounds to become cheaper and more readily available. Most biosynthesis pathways used in metabolic engineering applications have been discovered manually, requiring a tedious search of existing literature and metabolic databases. However, the recent rapid development of available metabolic information has enabled the development of automated approaches for identifying novel pathways. Computer-assisted pathfinding has the potential to save biochemists time in the initial discovery steps of metabolic engineering. In this paper, we review the parameters and heuristics used to guide the search in recent pathfinding algorithms. These parameters and heuristics capture information on the metabolic network structure, compound structures, reaction features, and organism-specificity of pathways. No one metabolic pathfinding algorithm or search parameter stands out as the best to use broadly for solving the pathfinding problem, as each method and parameter has its own strengths and shortcomings. As assisted pathfinding approaches continue to become more sophisticated, the development of better methods for visualizing pathway results and integrating these results into existing metabolic engineering practices is also important for encouraging wider use of these pathfinding methods.</p></div></div></li><li> <span id="devaurs2017coarse-grained-conformational-sampling-of-protein">D. Devaurs, D. A. Antunes, M. Papanastasiou, <b>M. Moll</b>, D. Ricklin, J. D. Lambris, and L. E. Kavraki, “Coarse-Grained Conformational Sampling of Protein Structure Improves the Fit to Experimental Hydrogen-Exchange Data,” <i>Frontiers in Molecular Biosciences</i>, vol. 4, no. 13, Mar. 2017.</span> <br><div id="accdevaurs2017coarse-grained-conformational-sampling-of-protein"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/devaurs2017coarse-grained-conformational-sampling-of-protein.pdf');" href="/publications/devaurs2017coarse-grained-conformational-sampling-of-protein.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.3389/fmolb.2017.00013');" href="https://dx.doi.org/10.3389/fmolb.2017.00013"> publisher </a> <button class="btn" type="button" onclick="$('#bibdevaurs2017coarse-grained-conformational-sampling-of-protein').slideToggle(600);$('#absdevaurs2017coarse-grained-conformational-sampling-of-protein').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absdevaurs2017coarse-grained-conformational-sampling-of-protein').slideToggle(600);$('#bibdevaurs2017coarse-grained-conformational-sampling-of-protein').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibdevaurs2017coarse-grained-conformational-sampling-of-protein"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">devaurs2017coarse-grained-conformational-sampling-of-protein</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Monitoring hydrogen/deuterium exchange (HDX) undergone by a
        protein in solution produces experimental data that translates into
        valuable information about the protein's structure. Data produced by
        HDX experiments is often interpreted using a crystal structure of the
        protein, when available. However, it has been shown that the
        correspondence between experimental HDX data and crystal structures is
        often not satisfactory. This creates difficulties when trying to
        perform a structural analysis of the HDX data. In this paper, we
        evaluate several strategies to obtain a conformation providing a good
        fit to the experimental HDX data, which is a premise of an accurate
        structural analysis. We show that performing molecular dynamics
        simulations can be inadequate to obtain such conformations, and we
        propose a novel methodology involving a coarse-grained conformational
        sampling approach instead. By extensively exploring the intrinsic
        flexibility of a protein with this approach, we produce a
        conformational ensemble from which we extract a single conformation
        providing a good fit to the experimental HDX data. We successfully
        demonstrate the applicability of our method to four small and
        medium-sized proteins."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Devaurs, Didier and Antunes, Dinler A and Papanastasiou, Malvina and Moll, Mark and Ricklin, Daniel and Lambris, John D and Kavraki, Lydia E"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.3389/fmolb.2017.00013"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Frontiers in Molecular Biosciences"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"13"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Coarse-Grained Conformational Sampling of Protein Structure Improves
        the Fit to Experimental Hydrogen-Exchange Data"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"4"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2017"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absdevaurs2017coarse-grained-conformational-sampling-of-protein"><h3>Abstract</h3><p>Monitoring hydrogen/deuterium exchange (HDX) undergone by a protein in solution produces experimental data that translates into valuable information about the protein’s structure. Data produced by HDX experiments is often interpreted using a crystal structure of the protein, when available. However, it has been shown that the correspondence between experimental HDX data and crystal structures is often not satisfactory. This creates difficulties when trying to perform a structural analysis of the HDX data. In this paper, we evaluate several strategies to obtain a conformation providing a good fit to the experimental HDX data, which is a premise of an accurate structural analysis. We show that performing molecular dynamics simulations can be inadequate to obtain such conformations, and we propose a novel methodology involving a coarse-grained conformational sampling approach instead. By extensively exploring the intrinsic flexibility of a protein with this approach, we produce a conformational ensemble from which we extract a single conformation providing a good fit to the experimental HDX data. We successfully demonstrate the applicability of our method to four small and medium-sized proteins.</p></div></div></li><li> <span id="novinskaya2017defining-low-dimensional-projections-to-guide">A. Novinskaya, D. Devaurs, <b>M. Moll</b>, and L. E. Kavraki, “Defining Low-Dimensional Projections to Guide Protein Conformational Sampling,” <i>Journal of Computational Biology</i>, vol. 24, no. 1, pp. 79–89, Jan. 2017.</span> <br><div id="accnovinskaya2017defining-low-dimensional-projections-to-guide"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/novinskaya2017defining-low-dimensional-projections-to-guide.pdf');" href="/publications/novinskaya2017defining-low-dimensional-projections-to-guide.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1089/cmb.2016.0144');" href="https://dx.doi.org/10.1089/cmb.2016.0144"> publisher </a> <button class="btn" type="button" onclick="$('#bibnovinskaya2017defining-low-dimensional-projections-to-guide').slideToggle(600);$('#absnovinskaya2017defining-low-dimensional-projections-to-guide').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absnovinskaya2017defining-low-dimensional-projections-to-guide').slideToggle(600);$('#bibnovinskaya2017defining-low-dimensional-projections-to-guide').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibnovinskaya2017defining-low-dimensional-projections-to-guide"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">novinskaya2017defining-low-dimensional-projections-to-guide</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Exploring the conformational space of proteins is critical to
        characterizing their functions. Numerous methods have been proposed to
        sample a protein's conformational space, including techniques developed
        in the field of robotics and known as sampling-based motion-planning
        algorithms (or sampling-based planners). However, these algorithms
        suffer from the curse of dimensionality when applied to large proteins.
        Many sampling-based planners attempt to mitigate this issue by keeping
        track of sampling density to guide conformational sampling toward
        unexplored regions of the conformational space. This is often done
        using low-dimensional projections as an indirect way to reduce the
        dimensionality of the exploration problem. However, how to choose an
        appropriate projection and how much it influences the planner's
        performance are still poorly understood problems. In this paper, we
        introduce two methodologies defining low-dimensional projections that
        can be used by sampling-based planners for protein conformational
        sampling. The first method leverages information about a protein's
        flexibility to construct projections that can efficiently guide
        conformational sampling, when expert knowledge is available. The second
        method builds similar projections automatically, without expert
        intervention. We evaluate the projections produced by both
        methodologies on two conformational-search problems involving three
        middle-size proteins. Our experiments demonstrate that (i) defining
        projections based on expert knowledge can benefit conformational
        sampling, and (ii) automatically constructing such projections is a
        reasonable alternative."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Novinskaya, Anastasia and Devaurs, Didier and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1089/cmb.2016.0144"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Journal of Computational Biology"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"protein conformational sampling, protein flexibility, sampling-
        based planning, low-dimensional projection"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jan</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"79-89"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Defining Low-Dimensional Projections to Guide Protein Conformational
        Sampling"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"24"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2017"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absnovinskaya2017defining-low-dimensional-projections-to-guide"><h3>Abstract</h3><p>Exploring the conformational space of proteins is critical to characterizing their functions. Numerous methods have been proposed to sample a protein’s conformational space, including techniques developed in the field of robotics and known as sampling-based motion-planning algorithms (or sampling-based planners). However, these algorithms suffer from the curse of dimensionality when applied to large proteins. Many sampling-based planners attempt to mitigate this issue by keeping track of sampling density to guide conformational sampling toward unexplored regions of the conformational space. This is often done using low-dimensional projections as an indirect way to reduce the dimensionality of the exploration problem. However, how to choose an appropriate projection and how much it influences the planner’s performance are still poorly understood problems. In this paper, we introduce two methodologies defining low-dimensional projections that can be used by sampling-based planners for protein conformational sampling. The first method leverages information about a protein’s flexibility to construct projections that can efficiently guide conformational sampling, when expert knowledge is available. The second method builds similar projections automatically, without expert intervention. We evaluate the projections produced by both methodologies on two conformational-search problems involving three middle-size proteins. Our experiments demonstrate that (i) defining projections based on expert knowledge can benefit conformational sampling, and (ii) automatically constructing such projections is a reasonable alternative.</p></div></div></li><li> <span id="moll2016structure-guided-selection-of-specificity-determining"><b>M. Moll</b>, P. W. Finn, and L. E. Kavraki, “Structure-Guided Selection of Specificity Determining Positions in the Human Kinome,” <i>BMC Genomics</i>, vol. 17 (Suppl. 4), p. 431, Aug. 2016.</span> <br><div id="accmoll2016structure-guided-selection-of-specificity-determining"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2016structure-guided-selection-of-specificity-determining.pdf');" href="/publications/moll2016structure-guided-selection-of-specificity-determining.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1186/s12864-016-2790-3');" href="https://dx.doi.org/10.1186/s12864-016-2790-3"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2016structure-guided-selection-of-specificity-determining').slideToggle(600);$('#absmoll2016structure-guided-selection-of-specificity-determining').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2016structure-guided-selection-of-specificity-determining').slideToggle(600);$('#bibmoll2016structure-guided-selection-of-specificity-determining').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2016structure-guided-selection-of-specificity-determining"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll2016structure-guided-selection-of-specificity-determining</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Background: The human kinome contains many important drug
        targets. It is well-known that inhibitors of protein kinases bind with
        very different selectivity profiles. This is also the case for
        inhibitors of many other protein families. The increased availability
        of protein 3D structures has provided much information on the
        structural variation within a given protein family. However, the
        relationship between structural variations and binding specificity is
        complex and incompletely understood. We have developed a structural
        bioinformatics approach which provides an analysis of key determinants
        of binding selectivity as a tool to enhance the rational design of
        drugs with a specific selectivity profile.
        
        Results: We propose a greedy algorithm that computes a subset of
        residue positions in a multiple sequence alignment such that structural
        and chemical variation in those positions helps explain known binding
        affinities. By providing this information, the main purpose of the
        algorithm is to provide experimentalists with possible insights into
        how the selectivity profile of certain inhibitors is achieved, which is
        useful for lead optimization. In addition, the algorithm can also be
        used to predict binding affinities for structures whose affinity for a
        given inhibitor is unknown. The algorithm's performance is demonstrated
        using an extensive dataset for the human kinome.
        
        Conclusion: We show that the binding affinity of 38 different kinase
        inhibitors can be explained with consistently high precision and
        accuracy using the variation of at most six residue positions in the
        kinome binding site. We show for several inhibitors that we are able to
        identify residues that are known to be functionally important."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Finn, Paul W. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1186/s12864-016-2790-3"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"BMC Genomics"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"protein kinases; specificity determining positions; binding
        affinity"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">aug</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"431"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Structure-Guided Selection of Specificity Determining Positions in
        the Human Kinome"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"17 (Suppl.\ 4)"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2016"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2016structure-guided-selection-of-specificity-determining"><h3>Abstract</h3><p>Background: The human kinome contains many important drug targets. It is well-known that inhibitors of protein kinases bind with very different selectivity profiles. This is also the case for inhibitors of many other protein families. The increased availability of protein 3D structures has provided much information on the structural variation within a given protein family. However, the relationship between structural variations and binding specificity is complex and incompletely understood. We have developed a structural bioinformatics approach which provides an analysis of key determinants of binding selectivity as a tool to enhance the rational design of drugs with a specific selectivity profile. Results: We propose a greedy algorithm that computes a subset of residue positions in a multiple sequence alignment such that structural and chemical variation in those positions helps explain known binding affinities. By providing this information, the main purpose of the algorithm is to provide experimentalists with possible insights into how the selectivity profile of certain inhibitors is achieved, which is useful for lead optimization. In addition, the algorithm can also be used to predict binding affinities for structures whose affinity for a given inhibitor is unknown. The algorithm’s performance is demonstrated using an extensive dataset for the human kinome. Conclusion: We show that the binding affinity of 38 different kinase inhibitors can be explained with consistently high precision and accuracy using the variation of at most six residue positions in the kinome binding site. We show for several inhibitors that we are able to identify residues that are known to be functionally important.</p></div></div></li><li> <span id="kavraki2016rss2014">L. E. Kavraki and <b>M. Moll</b>, “Special Issue on the 2014 ‘Robotics: Science &amp; Systems,’ <i>Intl. J. of Robotics Research</i>, vol. 35, no. 1–3, pp. 3–4, Mar. 2016.</span> <br><div id="acckavraki2016rss2014"><div> <button class="btn disabled">pdf</button> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1177/0278364915608299');" href="https://dx.doi.org/10.1177/0278364915608299"> publisher </a> <button class="btn" type="button" onclick="$('#bibkavraki2016rss2014').slideToggle(600);$('#abskavraki2016rss2014').slideUp(600);"> bibtex </button> <button class="btn disabled" type="button">abstract</button></div><div class="card collapse" id="bibkavraki2016rss2014"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">kavraki2016rss2014</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Kavraki, Lydia E. and Moll, Mark"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1177/0278364915608299"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Intl.\ J.\ of Robotics Research"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1--3"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"3--4"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Special Issue on the 2014 ``{R}obotics: {S}cience \&amp; {S}ystems''
        Conference (Guest Editorial)"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"35"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2016"</span>
<span class="p">}</span></code></pre></figure></div></div></li><li> <span id="moll2015benchmarking-motion-planning-algorithms"><b>M. Moll</b>, I. A. Şucan, and L. E. Kavraki, “Benchmarking Motion Planning Algorithms: An Extensible Infrastructure for Analysis and Visualization,” <i>IEEE Robotics &amp; Automation Magazine (Special Issue on Replicable and Measurable Robotics Research)</i>, vol. 22, no. 3, pp. 96–102, Sep. 2015.</span> <br><div id="accmoll2015benchmarking-motion-planning-algorithms"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2015benchmarking-motion-planning-algorithms.pdf');" href="/publications/moll2015benchmarking-motion-planning-algorithms.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/MRA.2015.2448276');" href="https://dx.doi.org/10.1109/MRA.2015.2448276"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2015benchmarking-motion-planning-algorithms').slideToggle(600);$('#absmoll2015benchmarking-motion-planning-algorithms').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2015benchmarking-motion-planning-algorithms').slideToggle(600);$('#bibmoll2015benchmarking-motion-planning-algorithms').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2015benchmarking-motion-planning-algorithms"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll2015benchmarking-motion-planning-algorithms</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Sampling-based planning algorithms are widely used on many robot
        platforms. Within this class of algorithms, many variants have been
        proposed over the last 20 years, yet there is still no characterization
        of which algorithms are well-suited for which classes of problems. This
        has motivated us to develop a benchmarking infrastructure for motion
        planning algorithms. It consists of three main components. First, we
        have created an extensive benchmarking software framework that is
        included with the Open Motion Planning Library (OMPL), a C++ library
        that contains implementations of many sampling-based algorithms.
        Second, we have defined extensible formats for storing benchmark
        results. The formats are fairly straightforward so that other planning
        libraries could easily produce compatible output. Finally, we have
        created an interactive, versatile visualization tool for compact
        presentation of collected benchmark data. The tool and underlying
        database facilitate the analysis of performance across benchmark
        problems and planners."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and {\c{S}}ucan, Ioan A. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/MRA.2015.2448276"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"{IEEE} Robotics \&amp; Automation Magazine (Special Issue on
        Replicable and Measurable Robotics Research)"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"3"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"96--102"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Benchmarking Motion Planning Algorithms: An Extensible
        Infrastructure for Analysis and Visualization"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"22"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2015"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2015benchmarking-motion-planning-algorithms"><h3>Abstract</h3><p>Sampling-based planning algorithms are widely used on many robot platforms. Within this class of algorithms, many variants have been proposed over the last 20 years, yet there is still no characterization of which algorithms are well-suited for which classes of problems. This has motivated us to develop a benchmarking infrastructure for motion planning algorithms. It consists of three main components. First, we have created an extensive benchmarking software framework that is included with the Open Motion Planning Library (OMPL), a C++ library that contains implementations of many sampling-based algorithms. Second, we have defined extensible formats for storing benchmark results. The formats are fairly straightforward so that other planning libraries could easily produce compatible output. Finally, we have created an interactive, versatile visualization tool for compact presentation of collected benchmark data. The tool and underlying database facilitate the analysis of performance across benchmark problems and planners.</p></div></div></li><li> <span id="grady2015extending-the-applicability-of-pomdp-solutions">D. K. Grady, <b>M. Moll</b>, and L. E. Kavraki, “Extending the Applicability of POMDP Solutions to Robotic Tasks,” <i>IEEE Trans. on Robotics</i>, vol. 31, no. 4, pp. 948–961, Aug. 2015.</span> <br><div id="accgrady2015extending-the-applicability-of-pomdp-solutions"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/grady2015extending-the-applicability-of-pomdp-solutions.pdf');" href="/publications/grady2015extending-the-applicability-of-pomdp-solutions.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/TRO.2015.2441511');" href="https://dx.doi.org/10.1109/TRO.2015.2441511"> publisher </a> <button class="btn" type="button" onclick="$('#bibgrady2015extending-the-applicability-of-pomdp-solutions').slideToggle(600);$('#absgrady2015extending-the-applicability-of-pomdp-solutions').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absgrady2015extending-the-applicability-of-pomdp-solutions').slideToggle(600);$('#bibgrady2015extending-the-applicability-of-pomdp-solutions').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibgrady2015extending-the-applicability-of-pomdp-solutions"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">grady2015extending-the-applicability-of-pomdp-solutions</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Partially-Observable Markov Decision Processes (POMDPs) are used
        in many robotic task classes from soccer to household chores.
        Determining an approximately optimal action policy for POMDPs is
        PSPACE-complete, and the exponential growth of computation time
        prohibits solving large tasks. This paper describes two techniques to
        extend the range of robotic tasks that can be solved using a POMDP. Our
        first technique reduces the motion constraints of a robot, and then
        uses state-of-the-art robotic motion planning techniques to respect the
        true motion constraints at runtime. We then propose a novel task
        decomposition that can be applied to some indoor robotic tasks. This
        decomposition transforms a long time horizon task into a set of shorter
        tasks. We empirically demonstrate the performance gain provided by
        these two techniques through simulated execution in a variety of
        environments. Comparing a direct formulation of a POMDP to solving our
        proposed reductions, we conclude that the techniques proposed in this
        paper can provide significant enhancement to current POMDP solution
        techniques, extending the POMDP instances that can be solved to include
        large, continuous-state robotic tasks."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Grady, Devin K. and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/TRO.2015.2441511"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"{IEEE} Trans.\ on Robotics"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">aug</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"4"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"948--961"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Extending the Applicability of {POMDP} Solutions to Robotic Tasks"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"31"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2015"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absgrady2015extending-the-applicability-of-pomdp-solutions"><h3>Abstract</h3><p>Partially-Observable Markov Decision Processes (POMDPs) are used in many robotic task classes from soccer to household chores. Determining an approximately optimal action policy for POMDPs is PSPACE-complete, and the exponential growth of computation time prohibits solving large tasks. This paper describes two techniques to extend the range of robotic tasks that can be solved using a POMDP. Our first technique reduces the motion constraints of a robot, and then uses state-of-the-art robotic motion planning techniques to respect the true motion constraints at runtime. We then propose a novel task decomposition that can be applied to some indoor robotic tasks. This decomposition transforms a long time horizon task into a set of shorter tasks. We empirically demonstrate the performance gain provided by these two techniques through simulated execution in a variety of environments. Comparing a direct formulation of a POMDP to solving our proposed reductions, we conclude that the techniques proposed in this paper can provide significant enhancement to current POMDP solution techniques, extending the POMDP instances that can be solved to include large, continuous-state robotic tasks.</p></div></div></li><li> <span id="gipson2013sims-a-hybrid-method-for-rapid">B. Gipson, <b>M. Moll</b>, and L. E. Kavraki, “SIMS: A hybrid method for rapid conformational analysis,” <i>PLOS ONE</i>, vol. 8, no. 7, p. e68826, Jul. 2013.</span> <br><div id="accgipson2013sims-a-hybrid-method-for-rapid"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/gipson2013sims-a-hybrid-method-for-rapid.pdf');" href="/publications/gipson2013sims-a-hybrid-method-for-rapid.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1371/journal.pone.0068826');" href="https://dx.doi.org/10.1371/journal.pone.0068826"> publisher </a> <button class="btn" type="button" onclick="$('#bibgipson2013sims-a-hybrid-method-for-rapid').slideToggle(600);$('#absgipson2013sims-a-hybrid-method-for-rapid').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absgipson2013sims-a-hybrid-method-for-rapid').slideToggle(600);$('#bibgipson2013sims-a-hybrid-method-for-rapid').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibgipson2013sims-a-hybrid-method-for-rapid"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">gipson2013sims-a-hybrid-method-for-rapid</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Proteins are at the root of many biological functions, often
        performing complex tasks as the result of large changes in their
        structure. Describing the exact details of these conformational
        changes, however, remains a central challenge for computational biology
        due the enormous computational requirements of the problem. This has
        engendered the development of a rich variety of useful methods designed
        to answer specific questions at different levels of spatial, temporal,
        and energetic resolution. These methods fall largely into two classes:
        physically accurate, but computationally demanding methods and fast,
        approximate methods. We introduce here a new hybrid modeling tool, the
        Structured Intuitive Move Selector (SIMS), designed to bridge the
        divide between these two classes, while allowing the benefits of both
        to be seamlessly integrated into a single framework. This is achieved
        by applying a modern motion planning algorithm, borrowed from the field
        of robotics, in tandem with a well-established protein modeling
        library. sims can combine precise energy calculations with approximate
        or specialized conformational sampling routines to produce rapid, yet
        accurate, analysis of the large-scale conformational variability of
        protein systems. Several key advancements are shown, including the
        abstract use of generically defined moves (conformational sampling
        methods) and an expansive probabilistic conformational exploration. We
        present three example problems that SIMS is applied to and demonstrate
        a rapid solution for each. These include the automatic determination of
        ``active'' residues for the hinge-based system Cyanovirin-N, exploring
        conformational changes involving long-range coordinated motion between
        non-sequential residues in Ribose-Binding Protein, and the rapid
        discovery of a transient conformational state of Maltose-Binding
        Protein, previously only determined by Molecular Dynamics. For all
        cases we provide energetic validations using well-established energy
        fields, demonstrating this framework as a fast and accurate tool for
        the analysis of a wide range of protein flexibility problems."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Gipson, Bryant and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1371/journal.pone.0068826"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"PLOS ONE"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jul</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"7"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"e68826"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"{SIMS}: A hybrid method for rapid conformational analysis"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"8"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2013"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absgipson2013sims-a-hybrid-method-for-rapid"><h3>Abstract</h3><p>Proteins are at the root of many biological functions, often performing complex tasks as the result of large changes in their structure. Describing the exact details of these conformational changes, however, remains a central challenge for computational biology due the enormous computational requirements of the problem. This has engendered the development of a rich variety of useful methods designed to answer specific questions at different levels of spatial, temporal, and energetic resolution. These methods fall largely into two classes: physically accurate, but computationally demanding methods and fast, approximate methods. We introduce here a new hybrid modeling tool, the Structured Intuitive Move Selector (SIMS), designed to bridge the divide between these two classes, while allowing the benefits of both to be seamlessly integrated into a single framework. This is achieved by applying a modern motion planning algorithm, borrowed from the field of robotics, in tandem with a well-established protein modeling library. sims can combine precise energy calculations with approximate or specialized conformational sampling routines to produce rapid, yet accurate, analysis of the large-scale conformational variability of protein systems. Several key advancements are shown, including the abstract use of generically defined moves (conformational sampling methods) and an expansive probabilistic conformational exploration. We present three example problems that SIMS is applied to and demonstrate a rapid solution for each. These include the automatic determination of “active” residues for the hinge-based system Cyanovirin-N, exploring conformational changes involving long-range coordinated motion between non-sequential residues in Ribose-Binding Protein, and the rapid discovery of a transient conformational state of Maltose-Binding Protein, previously only determined by Molecular Dynamics. For all cases we provide energetic validations using well-established energy fields, demonstrating this framework as a fast and accurate tool for the analysis of a wide range of protein flexibility problems.</p></div></div></li><li> <span id="bryant2013combinatorial-clustering-of-residue-position">D. H. Bryant, <b>M. Moll</b>, P. W. Finn, and L. E. Kavraki, “Combinatorial clustering of residue position subsets predicts inhibitor affinity across the human kinome,” <i>PLOS Computational Biology</i>, vol. 9, no. 6, p. e1003087, Jun. 2013.</span> <br><div id="accbryant2013combinatorial-clustering-of-residue-position"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/bryant2013combinatorial-clustering-of-residue-position.pdf');" href="/publications/bryant2013combinatorial-clustering-of-residue-position.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1371/journal.pcbi.1003087');" href="https://dx.doi.org/10.1371/journal.pcbi.1003087"> publisher </a> <button class="btn" type="button" onclick="$('#bibbryant2013combinatorial-clustering-of-residue-position').slideToggle(600);$('#absbryant2013combinatorial-clustering-of-residue-position').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absbryant2013combinatorial-clustering-of-residue-position').slideToggle(600);$('#bibbryant2013combinatorial-clustering-of-residue-position').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibbryant2013combinatorial-clustering-of-residue-position"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bryant2013combinatorial-clustering-of-residue-position</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The protein kinases are a large family of enzymes that play
        fundamental roles in propagating signals within the cell. Because of
        the high degree of binding site similarity shared among protein
        kinases, designing drug compounds with high specificity among the
        kinases has proven difficult. However, computational approaches to
        comparing the 3-dimensional geometry and physicochemical properties of
        key binding site residue positions have been shown to be informative of
        inhibitor selectivity. The Combinatorial Clustering Of Residue Position
        Subsets (ccorps) method, introduced here, provides a semi-supervised
        learning approach for identifying structural features that are
        correlated with a given set of annotation labels. Here, ccorps is
        applied to the problem of identifying structural features of the kinase
        atp binding site that are informative of inhibitor binding. ccorps is
        demonstrated to make perfect or near-perfect predictions for the
        binding affinity profile of 8 of the 38 kinase inhibitors studied.
        Additionally, ccorps is shown to identify shared structural features
        across phylogenetically diverse groups of kinases that are correlated
        with binding affinity for particular inhibitors; such instances of
        structural similarity among phylogenetically diverse kinases are also
        shown to not be rare among kinases. Finally, these function-specific
        structural features may serve as potential starting points for the
        development of highly specific kinase inhibitors."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Bryant, Drew H. and Moll, Mark and Finn, Paul W. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1371/journal.pcbi.1003087"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"PLOS Computational Biology"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"kinases, binding affinity"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"6"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"e1003087"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Combinatorial clustering of residue position subsets predicts
        inhibitor affinity across the human kinome"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"9"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2013"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absbryant2013combinatorial-clustering-of-residue-position"><h3>Abstract</h3><p>The protein kinases are a large family of enzymes that play fundamental roles in propagating signals within the cell. Because of the high degree of binding site similarity shared among protein kinases, designing drug compounds with high specificity among the kinases has proven difficult. However, computational approaches to comparing the 3-dimensional geometry and physicochemical properties of key binding site residue positions have been shown to be informative of inhibitor selectivity. The Combinatorial Clustering Of Residue Position Subsets (ccorps) method, introduced here, provides a semi-supervised learning approach for identifying structural features that are correlated with a given set of annotation labels. Here, ccorps is applied to the problem of identifying structural features of the kinase atp binding site that are informative of inhibitor binding. ccorps is demonstrated to make perfect or near-perfect predictions for the binding affinity profile of 8 of the 38 kinase inhibitors studied. Additionally, ccorps is shown to identify shared structural features across phylogenetically diverse groups of kinases that are correlated with binding affinity for particular inhibitors; such instances of structural similarity among phylogenetically diverse kinases are also shown to not be rare among kinases. Finally, these function-specific structural features may serve as potential starting points for the development of highly specific kinase inhibitors.</p></div></div></li><li> <span id="moll2013software-for-project-based-learning-of-robot"><b>M. Moll</b>, J. Bordeaux, and L. E. Kavraki, “Software for Project-Based Learning of Robot Motion Planning,” <i>Computer Science Education, Special Issue on Robotics in CS Education</i>, vol. 23, no. 4, pp. 332–348, 2013.</span> <br><div id="accmoll2013software-for-project-based-learning-of-robot"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2013software-for-project-based-learning-of-robot.pdf');" href="/publications/moll2013software-for-project-based-learning-of-robot.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1080/08993408.2013.847167');" href="https://dx.doi.org/10.1080/08993408.2013.847167"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2013software-for-project-based-learning-of-robot').slideToggle(600);$('#absmoll2013software-for-project-based-learning-of-robot').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2013software-for-project-based-learning-of-robot').slideToggle(600);$('#bibmoll2013software-for-project-based-learning-of-robot').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2013software-for-project-based-learning-of-robot"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll2013software-for-project-based-learning-of-robot</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Motion planning is a core problem in robotics concerned with
        finding feasible paths for a given robot. Motion planning algorithms
        perform a search in the high-dimensional continuous space of robot
        configurations and exemplify many of the core algorithmic concepts of
        search algorithms and associated data structures. Motion planning
        algorithms can be explained in a simplified two-dimensional setting,
        but this masks many of the subtleties and complexities of the
        underlying problem. We have developed software for Project-Based
        Learning of motion planning that enables deep learning. The projects
        that we have developed allow advanced undergraduate students and
        graduate students to reflect on the performance of existing textbook
        algorithms and their own variations on such algorithms. Formative
        assessment has been conducted at three institutions. The core of the
        software used for this teaching module is also used within the Robot
        Operating System (ROS), a widely adopted platform by the robotics
        research community. This allows for transfer of knowledge and skills to
        robotics research projects involving a large variety robot hardware
        platforms."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Bordeaux, Janice and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1080/08993408.2013.847167"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Computer Science Education, Special Issue on Robotics in CS
        Education"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"robotics education; motion planning; project-based learning;
        educational software"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"4"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"332--348"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Software for Project-Based Learning of Robot Motion Planning"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"23"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2013"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2013software-for-project-based-learning-of-robot"><h3>Abstract</h3><p>Motion planning is a core problem in robotics concerned with finding feasible paths for a given robot. Motion planning algorithms perform a search in the high-dimensional continuous space of robot configurations and exemplify many of the core algorithmic concepts of search algorithms and associated data structures. Motion planning algorithms can be explained in a simplified two-dimensional setting, but this masks many of the subtleties and complexities of the underlying problem. We have developed software for Project-Based Learning of motion planning that enables deep learning. The projects that we have developed allow advanced undergraduate students and graduate students to reflect on the performance of existing textbook algorithms and their own variations on such algorithms. Formative assessment has been conducted at three institutions. The core of the software used for this teaching module is also used within the Robot Operating System (ROS), a widely adopted platform by the robotics research community. This allows for transfer of knowledge and skills to robotics research projects involving a large variety robot hardware platforms.</p></div></div></li><li> <span id="sucan2012the-open-motion-planning-library">I. A. Şucan, <b>M. Moll</b>, and L. E. Kavraki, “The Open Motion Planning Library,” <i>IEEE Robotics &amp; Automation Magazine</i>, vol. 19, no. 4, pp. 72–82, Dec. 2012.</span> <a href="http://ompl.kavrakilab.org">http://ompl.kavrakilab.org</a> <br><div id="accsucan2012the-open-motion-planning-library"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/sucan2012the-open-motion-planning-library.pdf');" href="/publications/sucan2012the-open-motion-planning-library.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/MRA.2012.2205651');" href="https://dx.doi.org/10.1109/MRA.2012.2205651"> publisher </a> <button class="btn" type="button" onclick="$('#bibsucan2012the-open-motion-planning-library').slideToggle(600);$('#abssucan2012the-open-motion-planning-library').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abssucan2012the-open-motion-planning-library').slideToggle(600);$('#bibsucan2012the-open-motion-planning-library').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibsucan2012the-open-motion-planning-library"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">sucan2012the-open-motion-planning-library</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We describe the Open Motion Planning Library (OMPL), a new
        library for sampling-based motion planning, which contains
        implementations of many state-of-the-art planning algorithms. The
        library is designed in a way that allows the user to easily solve a
        variety of complex motion planning problems with minimal input. OMPL
        facilitates the addition of new motion planning algorithms and it can
        be conveniently interfaced with other software components. A simple
        graphical user interface (GUI) built on top of the library, a number of
        tutorials, demos and programming assignments have been designed to
        teach students about sampling-based motion planning. Finally, the
        library is also available for use through the Robot Operating System
        (ROS)."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"{\c{S}}ucan, Ioan A. and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/MRA.2012.2205651"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"{IEEE} Robotics \&amp; Automation Magazine"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">dec</span><span class="p">,</span>
  <span class="na">note</span> <span class="p">=</span> <span class="s">"\url{http://ompl.kavrakilab.org}"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"4"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"72--82"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"The {O}pen {M}otion {P}lanning {L}ibrary"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"19"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2012"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abssucan2012the-open-motion-planning-library"><h3>Abstract</h3><p>We describe the Open Motion Planning Library (OMPL), a new library for sampling-based motion planning, which contains implementations of many state-of-the-art planning algorithms. The library is designed in a way that allows the user to easily solve a variety of complex motion planning problems with minimal input. OMPL facilitates the addition of new motion planning algorithms and it can be conveniently interfaced with other software components. A simple graphical user interface (GUI) built on top of the library, a number of tutorials, demos and programming assignments have been designed to teach students about sampling-based motion planning. Finally, the library is also available for use through the Robot Operating System (ROS).</p></div></div></li><li> <span id="bekris2012safe-distributed-motion-coordination">K. E. Bekris, D. K. Grady, <b>M. Moll</b>, and L. E. Kavraki, “Safe Distributed Motion Coordination for Second-Order Systems with Different Planning Cycles,” <i>Intl. J. of Robotics Research</i>, vol. 31, no. 2, pp. 129–149, Feb. 2012.</span> <br><div id="accbekris2012safe-distributed-motion-coordination"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/bekris2012safe-distributed-motion-coordination.pdf');" href="/publications/bekris2012safe-distributed-motion-coordination.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1177/0278364911430420');" href="https://dx.doi.org/10.1177/0278364911430420"> publisher </a> <button class="btn" type="button" onclick="$('#bibbekris2012safe-distributed-motion-coordination').slideToggle(600);$('#absbekris2012safe-distributed-motion-coordination').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absbekris2012safe-distributed-motion-coordination').slideToggle(600);$('#bibbekris2012safe-distributed-motion-coordination').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibbekris2012safe-distributed-motion-coordination"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bekris2012safe-distributed-motion-coordination</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"When multiple robots operate in the same environment, it is
        critical to coordinate their motion in a distributed fashion so that
        they reach their goals safely. If the robots have to respect
        second-order dynamics, this becomes a challenging problem, even for
        known static environments. This work presents a replanning framework
        where each robot computes partial plans independently during a planning
        cycle, while executing a previously computed trajectory. Each robot
        communicates with its neighbors to select a trajectory that is safe
        over an infinite time horizon. The proposed approach does not require
        synchronization and allows the robots to dynamically change their
        planning cycles over time. This paper proves that the proposed motion
        coordination algorithm guarantees safety under this general setting. An
        important advantage is that this framework is not specific to the
        underlying robot dynamics nor to the planning or control algorithm used
        to generate the robot trajectories. The performance of the approach is
        evaluated using a distributed multi-robot simulator on a computing
        cluster, where the simulated robots are forced to communicate by
        exchanging messages. The simulation results confirm the safety of the
        algorithm in various environments with up to 32 robots governed by
        second-order dynamics."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Bekris, Kostas E. and Grady, Devin K. and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1177/0278364911430420"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Intl.\ J.\ of Robotics Research"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">feb</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"2"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"129--149"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Safe Distributed Motion Coordination for Second-Order Systems with
        Different Planning Cycles"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"31"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2012"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absbekris2012safe-distributed-motion-coordination"><h3>Abstract</h3><p>When multiple robots operate in the same environment, it is critical to coordinate their motion in a distributed fashion so that they reach their goals safely. If the robots have to respect second-order dynamics, this becomes a challenging problem, even for known static environments. This work presents a replanning framework where each robot computes partial plans independently during a planning cycle, while executing a previously computed trajectory. Each robot communicates with its neighbors to select a trajectory that is safe over an infinite time horizon. The proposed approach does not require synchronization and allows the robots to dynamically change their planning cycles over time. This paper proves that the proposed motion coordination algorithm guarantees safety under this general setting. An important advantage is that this framework is not specific to the underlying robot dynamics nor to the planning or control algorithm used to generate the robot trajectories. The performance of the approach is evaluated using a distributed multi-robot simulator on a computing cluster, where the simulated robots are forced to communicate by exchanging messages. The simulation results confirm the safety of the algorithm in various environments with up to 32 robots governed by second-order dynamics.</p></div></div></li><li> <span id="moll2011the-labelhash-server-and-tools-for-substructure-based"><b>M. Moll</b>, D. H. Bryant, and L. E. Kavraki, “The LabelHash Server and Tools for Substructure-Based Functional Annotation,” <i>Bioinformatics</i>, vol. 27, no. 15, pp. 2161–2162, Jun. 2011.</span> <br><div id="accmoll2011the-labelhash-server-and-tools-for-substructure-based"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2011the-labelhash-server-and-tools-for-substructure-based.pdf');" href="/publications/moll2011the-labelhash-server-and-tools-for-substructure-based.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1093/bioinformatics/btr343');" href="https://dx.doi.org/10.1093/bioinformatics/btr343"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2011the-labelhash-server-and-tools-for-substructure-based').slideToggle(600);$('#absmoll2011the-labelhash-server-and-tools-for-substructure-based').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2011the-labelhash-server-and-tools-for-substructure-based').slideToggle(600);$('#bibmoll2011the-labelhash-server-and-tools-for-substructure-based').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2011the-labelhash-server-and-tools-for-substructure-based"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll2011the-labelhash-server-and-tools-for-substructure-based</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Summary: The LabelHash server and tools are designed for large-
        scale substructure comparison. The main use is to predict the function
        of unknown proteins. Given a set of (putative) functional residues,
        LabelHash finds all occurrences of matching substructures in the entire
        Protein Data Bank, along with a statistical significance estimate and
        known functional annotations for each match. The results can be
        downloaded for further analysis in any molecular viewer. For Chimera
        there is a plugin to facilitate this process.
        
        Availability: The website is free and open to all users with no login
        requirements at http://labelhash.kavrakilab.org.
        
        Contact: mmoll@rice.edu"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Bryant, Drew H. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1093/bioinformatics/btr343"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Bioinformatics"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"15"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"2161--2162"</span><span class="p">,</span>
  <span class="na">pmid</span> <span class="p">=</span> <span class="s">"21659320"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"The {LabelHash} Server and Tools for Substructure-Based Functional
        Annotation"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"27"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2011"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2011the-labelhash-server-and-tools-for-substructure-based"><h3>Abstract</h3><p>Summary: The LabelHash server and tools are designed for large- scale substructure comparison. The main use is to predict the function of unknown proteins. Given a set of (putative) functional residues, LabelHash finds all occurrences of matching substructures in the entire Protein Data Bank, along with a statistical significance estimate and known functional annotations for each match. The results can be downloaded for further analysis in any molecular viewer. For Chimera there is a plugin to facilitate this process. Availability: The website is free and open to all users with no login requirements at http://labelhash.kavrakilab.org. Contact: mmoll@rice.edu</p></div></div></li><li> <span id="moll2010the-labelhash-algorithm-for-substructure-matching"><b>M. Moll</b>, D. H. Bryant, and L. E. Kavraki, “The LabelHash Algorithm for Substructure Matching,” <i>BMC Bioinformatics</i>, vol. 11, no. 555, Nov. 2010.</span> <br><div id="accmoll2010the-labelhash-algorithm-for-substructure-matching"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2010the-labelhash-algorithm-for-substructure-matching.pdf');" href="/publications/moll2010the-labelhash-algorithm-for-substructure-matching.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1186/1471-2105-11-555');" href="https://dx.doi.org/10.1186/1471-2105-11-555"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2010the-labelhash-algorithm-for-substructure-matching').slideToggle(600);$('#absmoll2010the-labelhash-algorithm-for-substructure-matching').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2010the-labelhash-algorithm-for-substructure-matching').slideToggle(600);$('#bibmoll2010the-labelhash-algorithm-for-substructure-matching').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2010the-labelhash-algorithm-for-substructure-matching"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll2010the-labelhash-algorithm-for-substructure-matching</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Background There is an increasing number of proteins with known
        structure but unknown function. Determining their function would have a
        significant impact on understanding diseases and designing new
        therapeutics. However, experimental protein function determination is
        expensive and very time-consuming. Computational methods can facilitate
        function determination by identifying proteins that have high
        structural and chemical similarity.
        
        Results We present LabelHash, a novel algorithm for matching
        substructural motifs to large collections of protein structures. The
        algorithm consists of two phases. In the first phase the proteins are
        preprocessed in a fashion that allows for instant lookup of partial
        matches to any motif. In the second phase, partial matches for a given
        motif are expanded to complete matches. The general applicability of
        the algorithm is demonstrated with three different case studies. First,
        we show that we can accurately identify members of the enolase
        superfamily with a single motif. Next, we demonstrate how LabelHash can
        complement SOIPPA, an algorithm for motif identification and pairwise
        substructure alignment. Finally, a large collection of Catalytic Site
        Atlas motifs is used to benchmark the performance of the algorithm.
        LabelHash runs very efficiently in parallel; matching a motif against
        all proteins in the 95% sequence identity filtered non-redundant
        Protein Data Bank typically takes no more than a few minutes. The
        LabelHash algorithm is available through a web server and as a suite of
        standalone programs at http://labelhash.kavrakilab.org. The output of
        the LabelHash algorithm can be further analyzed with Chimera through a
        plugin that we developed for this purpose.
        
        Conclusions LabelHash is an efficient, versatile algorithm for
        large-scale substructure matching. When LabelHash is running in
        parallel, motifs can typically be matched against the entire PDB on the
        order of minutes. The algorithm is able to identify functional homologs
        beyond the twilight zone of sequence identity and even beyond fold
        similarity. The three case studies presented in this paper illustrate
        the versatility of the algorithm."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Bryant, Drew H. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1186/1471-2105-11-555"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"BMC Bioinformatics"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">nov</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"555"</span><span class="p">,</span>
  <span class="na">pmcid</span> <span class="p">=</span> <span class="s">"PMC2996407"</span><span class="p">,</span>
  <span class="na">pmid</span> <span class="p">=</span> <span class="s">"21070651"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"The {LabelHash} Algorithm for Substructure Matching"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"11"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2010"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2010the-labelhash-algorithm-for-substructure-matching"><h3>Abstract</h3><p>Background There is an increasing number of proteins with known structure but unknown function. Determining their function would have a significant impact on understanding diseases and designing new therapeutics. However, experimental protein function determination is expensive and very time-consuming. Computational methods can facilitate function determination by identifying proteins that have high structural and chemical similarity. Results We present LabelHash, a novel algorithm for matching substructural motifs to large collections of protein structures. The algorithm consists of two phases. In the first phase the proteins are preprocessed in a fashion that allows for instant lookup of partial matches to any motif. In the second phase, partial matches for a given motif are expanded to complete matches. The general applicability of the algorithm is demonstrated with three different case studies. First, we show that we can accurately identify members of the enolase superfamily with a single motif. Next, we demonstrate how LabelHash can complement SOIPPA, an algorithm for motif identification and pairwise substructure alignment. Finally, a large collection of Catalytic Site Atlas motifs is used to benchmark the performance of the algorithm. LabelHash runs very efficiently in parallel; matching a motif against all proteins in the 95% sequence identity filtered non-redundant Protein Data Bank typically takes no more than a few minutes. The LabelHash algorithm is available through a web server and as a suite of standalone programs at http://labelhash.kavrakilab.org. The output of the LabelHash algorithm can be further analyzed with Chimera through a plugin that we developed for this purpose. Conclusions LabelHash is an efficient, versatile algorithm for large-scale substructure matching. When LabelHash is running in parallel, motifs can typically be matched against the entire PDB on the order of minutes. The algorithm is able to identify functional homologs beyond the twilight zone of sequence identity and even beyond fold similarity. The three case studies presented in this paper illustrate the versatility of the algorithm.</p></div></div></li><li> <span id="bryant2010analysis-of-substructural-variation-in-families">D. H. Bryant, <b>M. Moll</b>, B. Y. Chen, V. Y. Fofanov, and L. E. Kavraki, “Analysis of substructural variation in families of enzymatic proteins with applications to protein function prediction,” <i>BMC Bioinformatics</i>, vol. 11, no. 242, May 2010.</span> <br><div id="accbryant2010analysis-of-substructural-variation-in-families"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/bryant2010analysis-of-substructural-variation-in-families.pdf');" href="/publications/bryant2010analysis-of-substructural-variation-in-families.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1186/1471-2105-11-242');" href="https://dx.doi.org/10.1186/1471-2105-11-242"> publisher </a> <button class="btn" type="button" onclick="$('#bibbryant2010analysis-of-substructural-variation-in-families').slideToggle(600);$('#absbryant2010analysis-of-substructural-variation-in-families').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absbryant2010analysis-of-substructural-variation-in-families').slideToggle(600);$('#bibbryant2010analysis-of-substructural-variation-in-families').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibbryant2010analysis-of-substructural-variation-in-families"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bryant2010analysis-of-substructural-variation-in-families</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Background Structural variations caused by a wide range of
        physicochemical and biological sources directly influence the function
        of a protein. For enzymatic proteins, the structure and chemistry of
        the catalytic binding site residues can be loosely defined as a
        substructure of the protein. Comparative analysis of drug-receptor
        substructures across and within species has been used for lead
        evaluation. Substructure-level similarity between the binding sites of
        functionally similar proteins has also been used to identify instances
        of convergent evolution among proteins. In functionally homologous
        protein families, shared chemistry and geometry at catalytic sites
        provide a common, local point of comparison among proteins that may
        differ significantly at the sequence, fold, or domain topology levels.
        
        Results This paper describes two key results that can be used
        separately or in combination for protein function analysis. The
        Family-wise Analysis of SubStructural Templates (FASST) method uses
        all-against-all substructure comparison to determine Substructural
        Clusters (SCs). SCs characterize the binding site substructural
        variation within a protein family. In this paper we focus on examples
        of automatically determined SCs that can be linked to phylogenetic
        distance between family members, segregation by conformation, and
        organization by homology among convergent protein lineages. The Motif
        Ensemble Statistical Hypothesis (MESH) framework constructs a
        representative motif for each protein cluster among the SCs determined
        by FASST to build motif ensembles that are shown through a series of
        function prediction experiments to improve the function prediction
        power of existing motifs.
        
        Conclusions FASST contributes a critical feedback and assessment step
        to existing binding site substructure identification methods and can be
        used for the thorough investigation of structure-function
        relationships. The application of MESH allows for an automated,
        statistically rigorous procedure for incorporating structural variation
        data into protein function prediction pipelines. Our work provides an
        unbiased, automated assessment of the structural variability of
        identified binding site substructures among protein structure families
        and a technique for exploring the relation of substructural variation
        to protein function. As available proteomic data continues to expand,
        the techniques proposed will be indispensable for the large-scale
        analysis and interpretation of structural data.
        
        "</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Bryant, Drew H. and Moll, Mark and Chen, Brian Y. and Fofanov, Viacheslav Y. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1186/1471-2105-11-242"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"BMC Bioinformatics"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">may</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"242"</span><span class="p">,</span>
  <span class="na">pmcid</span> <span class="p">=</span> <span class="s">"PMC2885373"</span><span class="p">,</span>
  <span class="na">pmid</span> <span class="p">=</span> <span class="s">"20459833"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Analysis of substructural variation in families of enzymatic
        proteins with applications to protein function prediction"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"11"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2010"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absbryant2010analysis-of-substructural-variation-in-families"><h3>Abstract</h3><p>Background Structural variations caused by a wide range of physicochemical and biological sources directly influence the function of a protein. For enzymatic proteins, the structure and chemistry of the catalytic binding site residues can be loosely defined as a substructure of the protein. Comparative analysis of drug-receptor substructures across and within species has been used for lead evaluation. Substructure-level similarity between the binding sites of functionally similar proteins has also been used to identify instances of convergent evolution among proteins. In functionally homologous protein families, shared chemistry and geometry at catalytic sites provide a common, local point of comparison among proteins that may differ significantly at the sequence, fold, or domain topology levels. Results This paper describes two key results that can be used separately or in combination for protein function analysis. The Family-wise Analysis of SubStructural Templates (FASST) method uses all-against-all substructure comparison to determine Substructural Clusters (SCs). SCs characterize the binding site substructural variation within a protein family. In this paper we focus on examples of automatically determined SCs that can be linked to phylogenetic distance between family members, segregation by conformation, and organization by homology among convergent protein lineages. The Motif Ensemble Statistical Hypothesis (MESH) framework constructs a representative motif for each protein cluster among the SCs determined by FASST to build motif ensembles that are shown through a series of function prediction experiments to improve the function prediction power of existing motifs. Conclusions FASST contributes a critical feedback and assessment step to existing binding site substructure identification methods and can be used for the thorough investigation of structure-function relationships. The application of MESH allows for an automated, statistically rigorous procedure for incorporating structural variation data into protein function prediction pipelines. Our work provides an unbiased, automated assessment of the structural variability of identified binding site substructures among protein structure families and a technique for exploring the relation of substructural variation to protein function. As available proteomic data continues to expand, the techniques proposed will be indispensable for the large-scale analysis and interpretation of structural data.</p></div></div></li><li> <span id="moll2010teaching-robot-motion-planning"><b>M. Moll</b>, J. Bordeaux, and L. E. Kavraki, “Teaching Robot Motion Planning,” <i>ASEE Computers in Education Journal (Special Issue on Novel Approaches to Robotics Education)</i>, vol. 20, no. 3, pp. 50–59, 2010.</span> <br><div id="accmoll2010teaching-robot-motion-planning"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2010teaching-robot-motion-planning.pdf');" href="/publications/moll2010teaching-robot-motion-planning.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://www.asee.org/papers-and-publications/publications/division-publications/computers-in-education-journal/volume-xx');" href="https://www.asee.org/papers-and-publications/publications/division-publications/computers-in-education-journal/volume-xx"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2010teaching-robot-motion-planning').slideToggle(600);$('#absmoll2010teaching-robot-motion-planning').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2010teaching-robot-motion-planning').slideToggle(600);$('#bibmoll2010teaching-robot-motion-planning').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2010teaching-robot-motion-planning"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll2010teaching-robot-motion-planning</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Robot motion planning is a fairly intuitive and engaging topic,
        yet it is difficult to teach. The material is taught in undergraduate
        and graduate robotics classes in computer science, electrical
        engineering, mechanical engineering and aeronautical engineering, but
        at an abstract level. Deep learning could be achieved by having
        students implement and test different motion planning strategies.
        However, it is practically impossible in the context of a single class
        to have undergraduates implement motion planning algorithms that are
        powerful and fun to use, even when the students have proficient
        programming skills. Due to lack of supporting educational material,
        students are often asked to implement simple (and uninteresting) motion
        planning algorithms from scratch, or access thousands of lines of code
        and just figure out how things work. We present an ongoing project to
        develop microworld software and a modeling curriculum that supports
        undergraduate acquisition of motion planning knowledge and tool use by
        computer science and engineering students. The goal is to open the
        field of motion planning and robotics to young and enthusiastic
        talent."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Bordeaux, Janice and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"ASEE Computers in Education Journal (Special Issue on Novel
        Approaches to Robotics Education)"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"3"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"50--59"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Teaching Robot Motion Planning"</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">"https://www.asee.org/papers-and-publications/publications/division-publications/computers-in-education-journal/volume-xx"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"20"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2010"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2010teaching-robot-motion-planning"><h3>Abstract</h3><p>Robot motion planning is a fairly intuitive and engaging topic, yet it is difficult to teach. The material is taught in undergraduate and graduate robotics classes in computer science, electrical engineering, mechanical engineering and aeronautical engineering, but at an abstract level. Deep learning could be achieved by having students implement and test different motion planning strategies. However, it is practically impossible in the context of a single class to have undergraduates implement motion planning algorithms that are powerful and fun to use, even when the students have proficient programming skills. Due to lack of supporting educational material, students are often asked to implement simple (and uninteresting) motion planning algorithms from scratch, or access thousands of lines of code and just figure out how things work. We present an ongoing project to develop microworld software and a modeling curriculum that supports undergraduate acquisition of motion planning knowledge and tool use by computer science and engineering students. The goal is to open the field of motion planning and robotics to young and enthusiastic talent.</p></div></div></li><li> <span id="haspel2010tracing-conformational-changes-in-proteins">N. Haspel, <b>M. Moll</b>, M. L. Baker, W. Chiu, and L. E. Kavraki, “Tracing Conformational Changes in Proteins,” <i>BMC Structural Biology</i>, vol. 10, no. Suppl. 1, p. S1, 2010.</span> <br><div id="acchaspel2010tracing-conformational-changes-in-proteins"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/haspel2010tracing-conformational-changes-in-proteins.pdf');" href="/publications/haspel2010tracing-conformational-changes-in-proteins.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1186/1472-6807-10-S1-S1');" href="https://dx.doi.org/10.1186/1472-6807-10-S1-S1"> publisher </a> <button class="btn" type="button" onclick="$('#bibhaspel2010tracing-conformational-changes-in-proteins').slideToggle(600);$('#abshaspel2010tracing-conformational-changes-in-proteins').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abshaspel2010tracing-conformational-changes-in-proteins').slideToggle(600);$('#bibhaspel2010tracing-conformational-changes-in-proteins').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibhaspel2010tracing-conformational-changes-in-proteins"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">haspel2010tracing-conformational-changes-in-proteins</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Background Many proteins undergo extensive conformational changes
        as part of their functionality. Tracing these changes is important for
        understanding the way these proteins function. Traditional
        biophysics-based conformational search methods require a large number
        of calculations and are hard to apply to large-scale conformational
        motions.
        
        Results In this work we investigate the application of a
        robotics-inspired method, using backbone and limited side chain
        representation and a coarse grained energy function to trace
        large-scale conformational motions. We tested the algorithm on four
        well known medium to large proteins and we show that even with
        relatively little information we are able to trace low-energy
        conformational pathways efficiently. The conformational pathways
        produced by our methods can be further filtered and refined to produce
        more useful information on the way proteins function under
        physiological conditions.
        
        Conclusions The proposed method effectively captures large-scale
        conformational changes and produces pathways that are consistent with
        experimental data and other computational studies. The method
        represents an important first step towards a larger scale modeling of
        more complex biological systems."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Haspel, Nurit and Moll, Mark and Baker, Matthew L. and Chiu, Wah and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1186/1472-6807-10-S1-S1"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"BMC Structural Biology"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"Suppl. 1"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"S1"</span><span class="p">,</span>
  <span class="na">pmcid</span> <span class="p">=</span> <span class="s">"PMC2873824"</span><span class="p">,</span>
  <span class="na">pmid</span> <span class="p">=</span> <span class="s">"20487508"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Tracing Conformational Changes in Proteins"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"10"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2010"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abshaspel2010tracing-conformational-changes-in-proteins"><h3>Abstract</h3><p>Background Many proteins undergo extensive conformational changes as part of their functionality. Tracing these changes is important for understanding the way these proteins function. Traditional biophysics-based conformational search methods require a large number of calculations and are hard to apply to large-scale conformational motions. Results In this work we investigate the application of a robotics-inspired method, using backbone and limited side chain representation and a coarse grained energy function to trace large-scale conformational motions. We tested the algorithm on four well known medium to large proteins and we show that even with relatively little information we are able to trace low-energy conformational pathways efficiently. The conformational pathways produced by our methods can be further filtered and refined to produce more useful information on the way proteins function under physiological conditions. Conclusions The proposed method effectively captures large-scale conformational changes and produces pathways that are consistent with experimental data and other computational studies. The method represents an important first step towards a larger scale modeling of more complex biological systems.</p></div></div></li><li> <span id="moll2008workshop-on-self-reconfigurable-modular-robots"><b>M. Moll</b> and D. Rus, “Special Issue on Self-Reconfiguring Modular Robots (Guest Editorial),” <i>Intl. J. of Robotics Research</i>, vol. 27, no. 3/4, pp. 277–278, Mar. 2008.</span> <br><div id="accmoll2008workshop-on-self-reconfigurable-modular-robots"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2008workshop-on-self-reconfigurable-modular-robots.pdf');" href="/publications/moll2008workshop-on-self-reconfigurable-modular-robots.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1177/0278364908089348');" href="https://dx.doi.org/10.1177/0278364908089348"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2008workshop-on-self-reconfigurable-modular-robots').slideToggle(600);$('#absmoll2008workshop-on-self-reconfigurable-modular-robots').slideUp(600);"> bibtex </button> <button class="btn disabled" type="button">abstract</button></div><div class="card collapse" id="bibmoll2008workshop-on-self-reconfigurable-modular-robots"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll2008workshop-on-self-reconfigurable-modular-robots</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Rus, Daniela"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1177/0278364908089348"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Intl.\ J.\ of Robotics Research"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"3/4"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"277-278"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Special Issue on Self-Reconfiguring Modular Robots (Guest
        Editorial)"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"27"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2008"</span>
<span class="p">}</span></code></pre></figure></div></div></li><li> <span id="yim2007modular-self-reconfigurable-robot-systems">M. Yim, W.-M. Shen, B. Salemi, D. Rus, <b>M. Moll</b>, H. Lipson, and E. Klavins, “Modular Self-reconfigurable Robot Systems: Challenges and Opportunities for the Future,” <i>IEEE Robotics &amp; Automation Magazine</i>, vol. 14, no. 1, pp. 43–52, Mar. 2007.</span> <br><div id="accyim2007modular-self-reconfigurable-robot-systems"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/yim2007modular-self-reconfigurable-robot-systems.pdf');" href="/publications/yim2007modular-self-reconfigurable-robot-systems.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/MRA.2007.339623');" href="https://dx.doi.org/10.1109/MRA.2007.339623"> publisher </a> <button class="btn" type="button" onclick="$('#bibyim2007modular-self-reconfigurable-robot-systems').slideToggle(600);$('#absyim2007modular-self-reconfigurable-robot-systems').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absyim2007modular-self-reconfigurable-robot-systems').slideToggle(600);$('#bibyim2007modular-self-reconfigurable-robot-systems').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibyim2007modular-self-reconfigurable-robot-systems"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">yim2007modular-self-reconfigurable-robot-systems</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The field of modular self-reconfigurable robotic systems
        addresses the design, fabrication, motion planning, and control of
        autonomous kinematic machines with variable morphology. Beyond
        conventional actuation, sensing, and control typically found in
        fixed-morphology robots, self-reconfigurable robots are also able to
        deliberately change their own shape by rearranging the connectivity of
        their parts in order to adapt to new circumstances, perform new tasks,
        or recover from damage. Over the last two decades, this field has
        advanced from proof-of-concept systems to elaborate physical
        implementations and simulations. The goal of this article is to outline
        some of this progress and identify key challenges and opportunities
        that lay ahead."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Yim, Mark and Shen, Wei-Min and Salemi, Benham and Rus, Daniela and Moll, Mark and Lipson, Hod and Klavins, Eric"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/MRA.2007.339623"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"{IEEE} Robotics \&amp; Automation Magazine"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"43--52"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Modular Self-reconfigurable Robot Systems: Challenges and
        Opportunities for the Future"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"14"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2007"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absyim2007modular-self-reconfigurable-robot-systems"><h3>Abstract</h3><p>The field of modular self-reconfigurable robotic systems addresses the design, fabrication, motion planning, and control of autonomous kinematic machines with variable morphology. Beyond conventional actuation, sensing, and control typically found in fixed-morphology robots, self-reconfigurable robots are also able to deliberately change their own shape by rearranging the connectivity of their parts in order to adapt to new circumstances, perform new tasks, or recover from damage. Over the last two decades, this field has advanced from proof-of-concept systems to elaborate physical implementations and simulations. The goal of this article is to outline some of this progress and identify key challenges and opportunities that lay ahead.</p></div></div></li><li> <span id="moll2006path-planning-for-deformable-linear"><b>M. Moll</b> and L. E. Kavraki, “Path Planning for Deformable Linear Objects,” <i>IEEE Trans. on Robotics</i>, vol. 22, no. 4, pp. 625–636, Aug. 2006.</span> <br><div id="accmoll2006path-planning-for-deformable-linear"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2006path-planning-for-deformable-linear.pdf');" href="/publications/moll2006path-planning-for-deformable-linear.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/TRO.2006.878933');" href="https://dx.doi.org/10.1109/TRO.2006.878933"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2006path-planning-for-deformable-linear').slideToggle(600);$('#absmoll2006path-planning-for-deformable-linear').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2006path-planning-for-deformable-linear').slideToggle(600);$('#bibmoll2006path-planning-for-deformable-linear').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2006path-planning-for-deformable-linear"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll2006path-planning-for-deformable-linear</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a new approach to path planning for deformable linear
        (one-dimensional) objects such as flexible wires. We introduce a method
        for efficiently computing stable configurations of a wire subject to
        manipulation constraints. These configurations correspond to
        minimal-energy curves. By restricting the planner to minimal-energy
        curves, the execution of a path becomes easier. Our curve
        representation is adaptive in the sense that the number of parameters
        automatically varies with the complexity of the underlying curve. We
        introduce a planner that computes paths from one minimal-energy curve
        to another such that all intermediate curves are also minimal-energy
        curves. This planner can be used as a powerful local planner in a
        sampling-based roadmap method. This makes it possible to compute a
        roadmap of the entire ``shape space,'' which is not possible with
        previous approaches. Using a simplified model for obstacles, we can
        find minimal-energy curves of fixed length that pass through specified
        tangents at given control points. Our work has applications in cable
        routing, and motion planning for surgical suturing and snake-like
        robots."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/TRO.2006.878933"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"{IEEE} Trans.\ on Robotics"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">aug</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"4"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"625--636"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Path Planning for Deformable Linear Objects"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"22"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2006"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2006path-planning-for-deformable-linear"><h3>Abstract</h3><p>We present a new approach to path planning for deformable linear (one-dimensional) objects such as flexible wires. We introduce a method for efficiently computing stable configurations of a wire subject to manipulation constraints. These configurations correspond to minimal-energy curves. By restricting the planner to minimal-energy curves, the execution of a path becomes easier. Our curve representation is adaptive in the sense that the number of parameters automatically varies with the complexity of the underlying curve. We introduce a planner that computes paths from one minimal-energy curve to another such that all intermediate curves are also minimal-energy curves. This planner can be used as a powerful local planner in a sampling-based roadmap method. This makes it possible to compute a roadmap of the entire “shape space,” which is not possible with previous approaches. Using a simplified model for obstacles, we can find minimal-energy curves of fixed length that pass through specified tangents at given control points. Our work has applications in cable routing, and motion planning for surgical suturing and snake-like robots.</p></div></div></li><li> <span id="das2006low-dimensional-free-energy-landscapes">P. Das, <b>M. Moll</b>, H. Stamati, L. E. Kavraki, and C. Clementi, “Low-dimensional, free-energy landscapes of protein-folding reactions by nonlinear dimensionality reduction,” <i>Proc. Natl. Acad. of Science USA</i>, vol. 103, no. 26, pp. 9885–9890, Jun. 2006.</span> <br><div id="accdas2006low-dimensional-free-energy-landscapes"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/das2006low-dimensional-free-energy-landscapes.pdf');" href="/publications/das2006low-dimensional-free-energy-landscapes.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1073/pnas.0603553103');" href="https://dx.doi.org/10.1073/pnas.0603553103"> publisher </a> <button class="btn" type="button" onclick="$('#bibdas2006low-dimensional-free-energy-landscapes').slideToggle(600);$('#absdas2006low-dimensional-free-energy-landscapes').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absdas2006low-dimensional-free-energy-landscapes').slideToggle(600);$('#bibdas2006low-dimensional-free-energy-landscapes').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibdas2006low-dimensional-free-energy-landscapes"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">das2006low-dimensional-free-energy-landscapes</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The definition of reaction coordinates for the characterization
        of a protein-folding reaction has long been a controversial issue, even
        for the ``simple'' case in which one single free-energy barrier
        separates the folded and unfolded ensemble. We propose a general
        approach to this problem to obtain a few collective coordinates by
        using nonlinear dimensionality reduction. We validate the usefulness of
        this method by characterizing the folding landscape associated with a
        coarse-grained protein model of src homology 3 as sampled by molecular
        dynamics simulations. The folding free-energy landscape projected on
        the few relevant coordinates emerging from the dimensionality reduction
        can correctly identify the transition-state ensemble of the reaction.
        The first embedding dimension efficiently captures the evolution of the
        folding process along the main folding route. These results clearly
        show that the proposed method can efficiently find a low-dimensional
        representation of a complex process such as protein folding."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Das, Payel and Moll, Mark and Stamati, Hernan and Kavraki, Lydia E. and Clementi, Cecilia"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1073/pnas.0603553103"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Proc.\ Natl.\ Acad.\ of Science USA"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"reaction coordinate, transition state, manifold, embedding,
        ISOMAP, ScIMAP"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"26"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"9885--9890"</span><span class="p">,</span>
  <span class="na">pmcid</span> <span class="p">=</span> <span class="s">"PMC1502548"</span><span class="p">,</span>
  <span class="na">pmid</span> <span class="p">=</span> <span class="s">"16785435"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Low-dimensional, free-energy landscapes of protein-folding reactions
        by nonlinear dimensionality reduction"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"103"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2006"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absdas2006low-dimensional-free-energy-landscapes"><h3>Abstract</h3><p>The definition of reaction coordinates for the characterization of a protein-folding reaction has long been a controversial issue, even for the “simple” case in which one single free-energy barrier separates the folded and unfolded ensemble. We propose a general approach to this problem to obtain a few collective coordinates by using nonlinear dimensionality reduction. We validate the usefulness of this method by characterizing the folding landscape associated with a coarse-grained protein model of src homology 3 as sampled by molecular dynamics simulations. The folding free-energy landscape projected on the few relevant coordinates emerging from the dimensionality reduction can correctly identify the transition-state ensemble of the reaction. The first embedding dimension efficiently captures the evolution of the folding process along the main folding route. These results clearly show that the proposed method can efficiently find a low-dimensional representation of a complex process such as protein folding.</p></div></div></li><li> <span id="moll-erdmann2002manip-pose-distr"><b>M. Moll</b> and M. A. Erdmann, “Manipulation of Pose Distributions,” <i>Intl. J. of Robotics Research</i>, vol. 21, no. 3, pp. 277–292, Mar. 2002.</span> <br><div id="accmoll-erdmann2002manip-pose-distr"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll-erdmann2002manip-pose-distr.pdf');" href="/publications/moll-erdmann2002manip-pose-distr.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1177/027836402320556449');" href="https://dx.doi.org/10.1177/027836402320556449"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll-erdmann2002manip-pose-distr').slideToggle(600);$('#absmoll-erdmann2002manip-pose-distr').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll-erdmann2002manip-pose-distr').slideToggle(600);$('#bibmoll-erdmann2002manip-pose-distr').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll-erdmann2002manip-pose-distr"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll-erdmann2002manip-pose-distr</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"For assembly tasks parts often have to be oriented before they
        can be put in an assembly. The results presented in this paper are a
        component of the automated design of parts orienting devices. The focus
        is on orienting parts with minimal sensing and manipulation. We present
        a new approach to parts orienting through the manipulation of pose
        distributions. Through dynamic simulation we can determine the pose
        distribution for an object being dropped from an arbitrary height on an
        arbitrary surface. By varying the drop height and the shape of the
        support surface we can find the initial conditions that will result in
        a pose distribution with minimal entropy. We are trying to uniquely
        orient a part with high probability just by varying the initial
        conditions. We will derive a condition on the pose and velocity of a
        simple planar object in contact with a sloped surface that will allow
        us to quickly determine the final resting configuration of the object.
        This condition can then be used to quickly compute the pose
        distribution. We also present simulation and experimental results that
        show how dynamic simulation can be used to find optimal shapes and drop
        heights for a given part."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Erdmann, Michael A."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1177/027836402320556449"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Intl.\ J.\ of Robotics Research"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"pose distributions, parts orienting, dynamic simulation"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"3"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"277--292"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Manipulation of Pose Distributions"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"21"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2002"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll-erdmann2002manip-pose-distr"><h3>Abstract</h3><p>For assembly tasks parts often have to be oriented before they can be put in an assembly. The results presented in this paper are a component of the automated design of parts orienting devices. The focus is on orienting parts with minimal sensing and manipulation. We present a new approach to parts orienting through the manipulation of pose distributions. Through dynamic simulation we can determine the pose distribution for an object being dropped from an arbitrary height on an arbitrary surface. By varying the drop height and the shape of the support surface we can find the initial conditions that will result in a pose distribution with minimal entropy. We are trying to uniquely orient a part with high probability just by varying the initial conditions. We will derive a condition on the pose and velocity of a simple planar object in contact with a sloped surface that will allow us to quickly determine the final resting configuration of the object. This condition can then be used to quickly compute the pose distribution. We also present simulation and experimental results that show how dynamic simulation can be used to find optimal shapes and drop heights for a given part.</p></div></div></li><li> <span id="moll2002align-parts-micro-assem"><b>M. Moll</b>, K. Goldberg, M. A. Erdmann, and R. Fearing, “Aligning Parts for Micro Assemblies,” <i>Assembly Automation</i>, vol. 22, no. 1, pp. 46–54, Feb. 2002.</span> <br><div id="accmoll2002align-parts-micro-assem"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2002align-parts-micro-assem.pdf');" href="/publications/moll2002align-parts-micro-assem.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1108/01445150210416673');" href="https://dx.doi.org/10.1108/01445150210416673"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2002align-parts-micro-assem').slideToggle(600);$('#absmoll2002align-parts-micro-assem').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2002align-parts-micro-assem').slideToggle(600);$('#bibmoll2002align-parts-micro-assem').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2002align-parts-micro-assem"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll2002align-parts-micro-assem</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Orienting parts that measure only a few micrometers in diameter
        introduces several challenges that need not be considered at the
        macro-scale. First, there are several kinds of sticking effects due to
        Van der Waals forces and static electricity which complicate hand-off
        motions and release of a part. Second, the degrees of freedom of
        micro-manipulators are limited. This paper proposes a pair of
        manipulation primitives and a complete algorithm that addresses these
        challenges. We will show that a sequence of these two manipulation
        primitives can uniquely orient any asymmetric part while maintaining
        contact without sensing. This allows us to apply the same plan to many
        (identical) parts simultaneously. For asymmetric parts we can find a
        plan of length O(n) in O(n) time that orients the part, where n is the
        number of vertices."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Goldberg, Ken and Erdmann, Michael A. and Fearing, Ron"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1108/01445150210416673"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Assembly Automation"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"micromanipulation, parts orienting, parts feeding, rolling"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">feb</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"1"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"46--54"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Aligning Parts for Micro Assemblies"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"22"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2002"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2002align-parts-micro-assem"><h3>Abstract</h3><p>Orienting parts that measure only a few micrometers in diameter introduces several challenges that need not be considered at the macro-scale. First, there are several kinds of sticking effects due to Van der Waals forces and static electricity which complicate hand-off motions and release of a part. Second, the degrees of freedom of micro-manipulators are limited. This paper proposes a pair of manipulation primitives and a complete algorithm that addresses these challenges. We will show that a sequence of these two manipulation primitives can uniquely orient any asymmetric part while maintaining contact without sensing. This allows us to apply the same plan to many (identical) parts simultaneously. For asymmetric parts we can find a plan of length O(n) in O(n) time that orients the part, where n is the number of vertices.</p></div></div></li><li> <span id="moll-miikkulainen1997conver-zone-episod-memor"><b>M. Moll</b> and R. Miikkulainen, “Convergence-Zone Episodic Memory: Analysis and Simulations,” <i>Neural Networks</i>, vol. 10, no. 6, pp. 1017–1036, Aug. 1997.</span> <br><div id="accmoll-miikkulainen1997conver-zone-episod-memor"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll-miikkulainen1997conver-zone-episod-memor.pdf');" href="/publications/moll-miikkulainen1997conver-zone-episod-memor.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1016/S0893-6080(97)00016-6');" href="https://dx.doi.org/10.1016/S0893-6080(97)00016-6"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll-miikkulainen1997conver-zone-episod-memor').slideToggle(600);$('#absmoll-miikkulainen1997conver-zone-episod-memor').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll-miikkulainen1997conver-zone-episod-memor').slideToggle(600);$('#bibmoll-miikkulainen1997conver-zone-episod-memor').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll-miikkulainen1997conver-zone-episod-memor"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">moll-miikkulainen1997conver-zone-episod-memor</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Human episodic memory provides a seemingly unlimited storage for
        everyday experiences, and a retrieval system that allows us to access
        the experiences with partial activation of their components. The system
        is believed to consist of a fast, temporary storage in the hippocampus,
        and a slow, long-term storage within the neocortex. This paper presents
        a neural network model of the hippocampal episodic memory inspired by
        Damasio's idea of Convergence Zones. The model consists of a layer of
        perceptual feature maps and a binding layer. A perceptual feature
        pattern is coarse coded in the binding layer, and stored on the weights
        between layers. A partial activation of the stored features activates
        the binding pattern, which in turn reactivates the entire stored
        pattern. For many configurations of the model, a theoretical lower
        bound for the memory capacity can be derived, and it can be an order of
        magnitude or higher than the number of all units in the model, and
        several orders of magnitude higher than the number of binding-layer
        units. Computational simulations further indicate that the average
        capacity is an order of magnitude larger than the theoretical lower
        bound, and making the connectivity between layers sparser causes an
        even further increase in capacity. Simulations also show that if more
        descriptive binding patterns are used, the errors tend to be more
        plausible (patterns are confused with other similar patterns), with a
        slight cost in capacity. The convergence-zone episodic memory therefore
        accounts for the immediate storage and associative retrieval capability
        and large capacity of the hippocampal memory, and shows why the memory
        encoding areas can be much smaller than the perceptual maps, consist of
        rather coarse computational units, and be only sparsely connected to
        the perceptual maps."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Miikkulainen, Risto"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1016/S0893-6080(97)00016-6"</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">"Neural Networks"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">aug</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"6"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"1017--1036"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Convergence-Zone Episodic Memory: Analysis and Simulations"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"10"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"1997"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll-miikkulainen1997conver-zone-episod-memor"><h3>Abstract</h3><p>Human episodic memory provides a seemingly unlimited storage for everyday experiences, and a retrieval system that allows us to access the experiences with partial activation of their components. The system is believed to consist of a fast, temporary storage in the hippocampus, and a slow, long-term storage within the neocortex. This paper presents a neural network model of the hippocampal episodic memory inspired by Damasio’s idea of Convergence Zones. The model consists of a layer of perceptual feature maps and a binding layer. A perceptual feature pattern is coarse coded in the binding layer, and stored on the weights between layers. A partial activation of the stored features activates the binding pattern, which in turn reactivates the entire stored pattern. For many configurations of the model, a theoretical lower bound for the memory capacity can be derived, and it can be an order of magnitude or higher than the number of all units in the model, and several orders of magnitude higher than the number of binding-layer units. Computational simulations further indicate that the average capacity is an order of magnitude larger than the theoretical lower bound, and making the connectivity between layers sparser causes an even further increase in capacity. Simulations also show that if more descriptive binding patterns are used, the errors tend to be more plausible (patterns are confused with other similar patterns), with a slight cost in capacity. The convergence-zone episodic memory therefore accounts for the immediate storage and associative retrieval capability and large capacity of the hippocampal memory, and shows why the memory encoding areas can be much smaller than the perceptual maps, consist of rather coarse computational units, and be only sparsely connected to the perceptual maps.</p></div></div></li></ol><h2 class="bibliography" id="secbook-chapters">Book Chapters</h2><ol class="bibliography"><li> <span id="butler2020a-general-algorithm-for-time-optimal-trajectory">S. Butler, <b>M. Moll</b>, and L. E. Kavraki, “A General Algorithm for Time-Optimal Trajectory Generation Subject to Minimum and Maximum Constraints,” in <i>Algorithmic Foundations of Robotics XII</i>, K. Goldberg, P. Abbeel, K. Bekris, and L. Miller, Eds. Springer, 2020, pp. 368–383.</span> <br><div id="accbutler2020a-general-algorithm-for-time-optimal-trajectory"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/butler2020a-general-algorithm-for-time-optimal-trajectory.pdf');" href="/publications/butler2020a-general-algorithm-for-time-optimal-trajectory.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1007/978-3-030-43089-4_24');" href="https://dx.doi.org/10.1007/978-3-030-43089-4_24"> publisher </a> <button class="btn" type="button" onclick="$('#bibbutler2020a-general-algorithm-for-time-optimal-trajectory').slideToggle(600);$('#absbutler2020a-general-algorithm-for-time-optimal-trajectory').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absbutler2020a-general-algorithm-for-time-optimal-trajectory').slideToggle(600);$('#bibbutler2020a-general-algorithm-for-time-optimal-trajectory').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibbutler2020a-general-algorithm-for-time-optimal-trajectory"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@incollection</span><span class="p">{</span><span class="nl">butler2020a-general-algorithm-for-time-optimal-trajectory</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"This paper presents a new algorithm which generates time-optimal
        trajectories given a path as input. The algorithm improves on previous
        approaches by generically handling a broader class of constraints on
        the dynamics. It eliminates the need for heuristics to select
        trajectory segments that are part of the optimal trajectory through an
        exhaustive, but efficient search. We also present an algorithm for
        computing all achievable velocities at the end of a path given an
        initial range of velocities. This algorithm effectively computes
        bundles of feasible trajectories for a given path and is a first step
        toward a new generation of more efficient kinodynamic motion planning
        algorithms. We present results for both algorithms using a simulated
        WAM arm with a Barrett hand subject to dynamics constraints on joint
        torque, joint velocity, momentum, and end effector velocity. The new
        algorithms are compared with a state-of-the-art alternative approach."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Butler, Stephen and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Algorithmic Foundations of Robotics {XII}"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1007/978-3-030-43089-4_24"</span><span class="p">,</span>
  <span class="na">editor</span> <span class="p">=</span> <span class="s">"Goldberg, Ken and Abbeel, Pieter and Bekris, Kostas and Miller, Lauren"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"368-383"</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">"Springer"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"A General Algorithm for Time-Optimal Trajectory Generation Subject
        to Minimum and Maximum Constraints"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2020"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absbutler2020a-general-algorithm-for-time-optimal-trajectory"><h3>Abstract</h3><p>This paper presents a new algorithm which generates time-optimal trajectories given a path as input. The algorithm improves on previous approaches by generically handling a broader class of constraints on the dynamics. It eliminates the need for heuristics to select trajectory segments that are part of the optimal trajectory through an exhaustive, but efficient search. We also present an algorithm for computing all achievable velocities at the end of a path given an initial range of velocities. This algorithm effectively computes bundles of feasible trajectories for a given path and is a first step toward a new generation of more efficient kinodynamic motion planning algorithms. We present results for both algorithms using a simulated WAM arm with a Barrett hand subject to dynamics constraints on joint torque, joint velocity, momentum, and end effector velocity. The new algorithms are compared with a state-of-the-art alternative approach.</p></div></div></li><li> <span id="kingston2017decoupling-constraints-from-sampling-based">Z. Kingston, <b>M. Moll</b>, and L. E. Kavraki, “Decoupling Constraints from Sampling-Based Planners,” in <i>Robotics Research (The 18th International Symposium ISRR)</i>, N. M. Amato, G. Hager, S. Thomas, and M. Torres-Torriti, Eds. Springer, 2020, pp. 913–928.</span> <br><div id="acckingston2017decoupling-constraints-from-sampling-based"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/kingston2017decoupling-constraints-from-sampling-based.pdf');" href="/publications/kingston2017decoupling-constraints-from-sampling-based.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1007/978-3-030-28619-4_62');" href="https://dx.doi.org/10.1007/978-3-030-28619-4_62"> publisher </a> <button class="btn" type="button" onclick="$('#bibkingston2017decoupling-constraints-from-sampling-based').slideToggle(600);$('#abskingston2017decoupling-constraints-from-sampling-based').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abskingston2017decoupling-constraints-from-sampling-based').slideToggle(600);$('#bibkingston2017decoupling-constraints-from-sampling-based').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibkingston2017decoupling-constraints-from-sampling-based"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@incollection</span><span class="p">{</span><span class="nl">kingston2017decoupling-constraints-from-sampling-based</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a general unifying framework for sampling-based motion
        planning under kinematic task constraints which enables a broad class
        of planners to compute plans that satisfy a given constraint function
        that encodes, e.g., loop closure, balance, and end-effector
        constraints. The framework decouples a planner's method for exploration
        from constraint satisfaction by representing the implicit configuration
        space defined by a constraint function. We emulate three constraint
        satisfaction methodologies from the literature, and demonstrate the
        framework with a range of planners utilizing these constraint
        methodologies. Our results show that the appropriate choice of
        constrained satisfaction methodology depends on many factors, e.g., the
        dimension of the configuration space and implicit constraint manifold,
        and number of obstacles. Furthermore, we show that novel combinations
        of planners and constraint satisfaction methodologies can be more
        effective than previous approaches. The framework is also easily
        extended for novel planners and constraint spaces."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Kingston, Zachary and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Robotics Research (The 18th International Symposium ISRR)"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1007/978-3-030-28619-4_62"</span><span class="p">,</span>
  <span class="na">editor</span> <span class="p">=</span> <span class="s">"Amato, N. M. and Hager, G. and Thomas, S. and Torres-Torriti, M."</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"913--928"</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">"Springer"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Decoupling Constraints from Sampling-Based Planners"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2020"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abskingston2017decoupling-constraints-from-sampling-based"><h3>Abstract</h3><p>We present a general unifying framework for sampling-based motion planning under kinematic task constraints which enables a broad class of planners to compute plans that satisfy a given constraint function that encodes, e.g., loop closure, balance, and end-effector constraints. The framework decouples a planner’s method for exploration from constraint satisfaction by representing the implicit configuration space defined by a constraint function. We emulate three constraint satisfaction methodologies from the literature, and demonstrate the framework with a range of planners utilizing these constraint methodologies. Our results show that the appropriate choice of constrained satisfaction methodology depends on many factors, e.g., the dimension of the configuration space and implicit constraint manifold, and number of obstacles. Furthermore, we show that novel combinations of planners and constraint satisfaction methodologies can be more effective than previous approaches. The framework is also easily extended for novel planners and constraint spaces.</p></div></div></li><li> <span id="luna2015asymptotically-optimal-stochastic-motion">R. Luna, M. Lahijanian, <b>M. Moll</b>, and L. E. Kavraki, “Asymptotically Optimal Stochastic Motion Planning with Temporal Goals,” in <i>Algorithmic Foundations of Robotics XI</i>, vol. 107, H. L. Akin, N. M. Amato, V. Isler, and A. F. van der Stappen, Eds. Springer Verlag, 2015, pp. 335–352.</span> <br><div id="accluna2015asymptotically-optimal-stochastic-motion"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/luna2015asymptotically-optimal-stochastic-motion.pdf');" href="/publications/luna2015asymptotically-optimal-stochastic-motion.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1007/978-3-319-16595-0_20');" href="https://dx.doi.org/10.1007/978-3-319-16595-0_20"> publisher </a> <button class="btn" type="button" onclick="$('#bibluna2015asymptotically-optimal-stochastic-motion').slideToggle(600);$('#absluna2015asymptotically-optimal-stochastic-motion').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absluna2015asymptotically-optimal-stochastic-motion').slideToggle(600);$('#bibluna2015asymptotically-optimal-stochastic-motion').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibluna2015asymptotically-optimal-stochastic-motion"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@incollection</span><span class="p">{</span><span class="nl">luna2015asymptotically-optimal-stochastic-motion</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"This work presents a planning framework that allows a robot with
        stochastic action uncertainty to achieve a high-level task given in the
        form of a temporal logic formula. The objective is to quickly compute a
        feedback control policy to satisfy the task specification with maximum
        probability. A top-down framework is proposed that abstracts the motion
        of a continuous stochastic system to a discrete, bounded- parameter
        Markov decision process (bmdp), and then computes a control policy over
        the product of the bmdp abstraction and a dfa representing the temporal
        logic specification. Analysis of the framework reveals that as the
        resolution of the bmdp abstraction becomes finer, the policy obtained
        converges to optimal. Simulations show that high-quality policies to
        satisfy complex temporal logic specifications can be obtained in
        seconds, orders of magnitude faster than existing methods. "</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Luna, Ryan and Lahijanian, Morteza and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Algorithmic Foundations of Robotics XI"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1007/978-3-319-16595-0_20"</span><span class="p">,</span>
  <span class="na">editor</span> <span class="p">=</span> <span class="s">"Akin, H. Levent and Amato, Nancy M. and Isler, Volkan and van der Stappen, A. Frank"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"planning under uncertainty, temporal logic planning, stochastic
        systems, formal control synthesis"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"335--352"</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">"Springer Verlag"</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">"Springer Tracts in Advanced Robotics"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Asymptotically Optimal Stochastic Motion Planning with Temporal
        Goals"</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">"107"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2015"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absluna2015asymptotically-optimal-stochastic-motion"><h3>Abstract</h3><p>This work presents a planning framework that allows a robot with stochastic action uncertainty to achieve a high-level task given in the form of a temporal logic formula. The objective is to quickly compute a feedback control policy to satisfy the task specification with maximum probability. A top-down framework is proposed that abstracts the motion of a continuous stochastic system to a discrete, bounded- parameter Markov decision process (bmdp), and then computes a control policy over the product of the bmdp abstraction and a dfa representing the temporal logic specification. Analysis of the framework reveals that as the resolution of the bmdp abstraction becomes finer, the policy obtained converges to optimal. Simulations show that high-quality policies to satisfy complex temporal logic specifications can be obtained in seconds, orders of magnitude faster than existing methods.</p></div></div></li><li> <span id="moll2007roadmap-methods-for-protein-folding"><b>M. Moll</b>, D. Schwarz, and L. E. Kavraki, “Roadmap Methods for Protein Folding,” in <i>Protein Structure Prediction: Methods and Protocols</i>, Second., M. Zaki and C. Bystroff, Eds. Humana Press, 2007.</span> <br><div id="accmoll2007roadmap-methods-for-protein-folding"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2007roadmap-methods-for-protein-folding.pdf');" href="/publications/moll2007roadmap-methods-for-protein-folding.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1007/978-1-59745-574-9_9');" href="https://dx.doi.org/10.1007/978-1-59745-574-9_9"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2007roadmap-methods-for-protein-folding').slideToggle(600);$('#absmoll2007roadmap-methods-for-protein-folding').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2007roadmap-methods-for-protein-folding').slideToggle(600);$('#bibmoll2007roadmap-methods-for-protein-folding').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2007roadmap-methods-for-protein-folding"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@incollection</span><span class="p">{</span><span class="nl">moll2007roadmap-methods-for-protein-folding</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Protein folding refers to the process whereby a protein assumes
        its intricate three-dimensional shape. Different aspects of this
        problem have attracted much attention in the last decade. Both
        experimental and computational methods have been used to study protein
        folding and there has been considerable progress This chapter reviews a
        class of methods for studying protein folding called roadmap methods.
        These methods are relatively new and are still under active
        development. Roadmap methods are computational methods that have been
        developed to understand the process or the mechanism by which a protein
        folds or unfolds. It is typically assumed that the folded state is
        already known. Note that this is not a comprehensive survey of all
        existing computational protein folding methods. In particular, it does
        not cover Molecular Dynamics (MD) methods, Monte Carlo methods (MC),
        the use of coarse grain models in simulations and many others. "</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Schwarz, David and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Protein Structure Prediction: Methods and Protocols"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1007/978-1-59745-574-9_9"</span><span class="p">,</span>
  <span class="na">edition</span> <span class="p">=</span> <span class="s">"Second"</span><span class="p">,</span>
  <span class="na">editor</span> <span class="p">=</span> <span class="s">"Zaki, Mohammed and Bystroff, Chris"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">pmid</span> <span class="p">=</span> <span class="s">"18075168"</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">"Humana Press"</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">"Methods In Molecular Biology"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Roadmap Methods for Protein Folding"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2007"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2007roadmap-methods-for-protein-folding"><h3>Abstract</h3><p>Protein folding refers to the process whereby a protein assumes its intricate three-dimensional shape. Different aspects of this problem have attracted much attention in the last decade. Both experimental and computational methods have been used to study protein folding and there has been considerable progress This chapter reviews a class of methods for studying protein folding called roadmap methods. These methods are relatively new and are still under active development. Roadmap methods are computational methods that have been developed to understand the process or the mechanism by which a protein folds or unfolds. It is typically assumed that the folded state is already known. Note that this is not a comprehensive survey of all existing computational protein folding methods. In particular, it does not cover Molecular Dynamics (MD) methods, Monte Carlo methods (MC), the use of coarse grain models in simulations and many others.</p></div></div></li><li> <span id="moll2004reconstructing-the-shape-and-motion-of-unknown"><b>M. Moll</b> and M. A. Erdmann, “Reconstructing the Shape and Motion of Unknown Objects with Active Tactile Sensors,” in <i>Algorithmic Foundations of Robotics V</i>, J.-D. Boissonnat, J. Burdick, K. Goldberg, and S. Hutchinson, Eds. Springer Verlag, 2004, pp. 293–310.</span> <br><div id="accmoll2004reconstructing-the-shape-and-motion-of-unknown"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2004reconstructing-the-shape-and-motion-of-unknown.pdf');" href="/publications/moll2004reconstructing-the-shape-and-motion-of-unknown.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1007/b80173');" href="https://dx.doi.org/10.1007/b80173"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2004reconstructing-the-shape-and-motion-of-unknown').slideToggle(600);$('#absmoll2004reconstructing-the-shape-and-motion-of-unknown').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2004reconstructing-the-shape-and-motion-of-unknown').slideToggle(600);$('#bibmoll2004reconstructing-the-shape-and-motion-of-unknown').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2004reconstructing-the-shape-and-motion-of-unknown"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@incollection</span><span class="p">{</span><span class="nl">moll2004reconstructing-the-shape-and-motion-of-unknown</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a method to simultaneously reconstruct the shape and
        motion of an unknown smooth convex object. The object is manipulated by
        planar palms covered with tactile elements. The shape and dynamics of
        the object can be expressed as a function of the sensor values and the
        motion of the palms. We present a brief review of previous results for
        the planar case. In this paper we show that the 3D case is
        fundamentally different from the planar case, due to increased tangent
        dimensionality. The main contribution of this paper is a shape-dynamics
        analysis in 3D, and the synthesis of shape approximation methods via
        reconstructed contact point curves."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Erdmann, Michael A."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Algorithmic Foundations of Robotics V"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1007/b80173"</span><span class="p">,</span>
  <span class="na">editor</span> <span class="p">=</span> <span class="s">"Boissonnat, Jean-Daniel and Burdick, Joel and Goldberg, Ken and Hutchinson, Seth"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"tactile sensing, shape reconstruction, nonprehensile
        manipulation, contact kinematics"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"293--310"</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">"Springer Verlag"</span><span class="p">,</span>
  <span class="na">series</span> <span class="p">=</span> <span class="s">"Springer Tracts in Advanced Robotics"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Reconstructing the Shape and Motion of Unknown Objects with Active
        Tactile Sensors"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2004"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2004reconstructing-the-shape-and-motion-of-unknown"><h3>Abstract</h3><p>We present a method to simultaneously reconstruct the shape and motion of an unknown smooth convex object. The object is manipulated by planar palms covered with tactile elements. The shape and dynamics of the object can be expressed as a function of the sensor values and the motion of the palms. We present a brief review of previous results for the planar case. In this paper we show that the 3D case is fundamentally different from the planar case, due to increased tangent dimensionality. The main contribution of this paper is a shape-dynamics analysis in 3D, and the synthesis of shape approximation methods via reconstructed contact point curves.</p></div></div></li><li> <span id="moll-erdmann2001manip-pose-distr"><b>M. Moll</b> and M. A. Erdmann, “Manipulation of Pose Distributions,” in <i>Algorithmic and Computational Robotics: New Directions</i>, B. R. Donald, K. M. Lynch, and D. Rus, Eds. A. K. Peters, 2001, pp. 127–141.</span> <br><div id="accmoll-erdmann2001manip-pose-distr"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll-erdmann2001manip-pose-distr.pdf');" href="/publications/moll-erdmann2001manip-pose-distr.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('http://www.crcpress.com/product/isbn/9781568811253');" href="http://www.crcpress.com/product/isbn/9781568811253"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll-erdmann2001manip-pose-distr').slideToggle(600);$('#absmoll-erdmann2001manip-pose-distr').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll-erdmann2001manip-pose-distr').slideToggle(600);$('#bibmoll-erdmann2001manip-pose-distr').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll-erdmann2001manip-pose-distr"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@incollection</span><span class="p">{</span><span class="nl">moll-erdmann2001manip-pose-distr</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"For assembly tasks parts often have to be oriented before they
        can be put in an assembly. The results presented in this paper are a
        component of the automated design of parts orienting devices. The focus
        is on orienting parts with minimal sensing and manipulation. We present
        a new approach to parts orienting through the manipulation of pose
        distributions. Through dynamic simulation we can determine the pose
        distribution for an object being dropped from an arbitrary height on an
        arbitrary surface. By varying the drop height and the shape of the
        support surface we can find the initial conditions that will result in
        a pose distribution with minimal entropy. We are trying to uniquely
        orient a part with high probability just by varying the initial
        conditions. We will derive a condition on the pose and velocity of an
        object in contact with a sloped surface that will allow us to quickly
        determine the final resting configuration of the object. This condition
        can then be used to quickly compute the pose distribution. We also
        present simulation and experimental results that show how dynamic
        simulation can be used to find optimal shapes and drop heights for a
        given part."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Erdmann, Michael A."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Algorithmic and Computational Robotics: New Directions"</span><span class="p">,</span>
  <span class="na">editor</span> <span class="p">=</span> <span class="s">"Donald, Bruce R. and Lynch, Kevin M. and Rus, Daniela"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"pose distributions, parts orienting, dynamic simulation"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"127--141"</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">"A. K. Peters"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Manipulation of Pose Distributions"</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">"http://www.crcpress.com/product/isbn/9781568811253"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2001"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll-erdmann2001manip-pose-distr"><h3>Abstract</h3><p>For assembly tasks parts often have to be oriented before they can be put in an assembly. The results presented in this paper are a component of the automated design of parts orienting devices. The focus is on orienting parts with minimal sensing and manipulation. We present a new approach to parts orienting through the manipulation of pose distributions. Through dynamic simulation we can determine the pose distribution for an object being dropped from an arbitrary height on an arbitrary surface. By varying the drop height and the shape of the support surface we can find the initial conditions that will result in a pose distribution with minimal entropy. We are trying to uniquely orient a part with high probability just by varying the initial conditions. We will derive a condition on the pose and velocity of an object in contact with a sloped surface that will allow us to quickly determine the final resting configuration of the object. This condition can then be used to quickly compute the pose distribution. We also present simulation and experimental results that show how dynamic simulation can be used to find optimal shapes and drop heights for a given part.</p></div></div></li></ol><h2 class="bibliography" id="secconference-articles">Conference Articles</h2><ol class="bibliography"><li> <span id="moll2021hyperplan"><b>M. Moll</b>, C. Chamzas, Z. Kingston, and L. E. Kavraki, “HyperPlan: A Framework for Motion Planning Algorithm Selection and Parameter Optimization,” in <i>IEEE/RSJ Intl. Conf. on Intelligent Robots and Systems</i>, 2021.</span> <br><div id="accmoll2021hyperplan"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2021hyperplan.pdf');" href="/publications/moll2021hyperplan.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibmoll2021hyperplan').slideToggle(600);$('#absmoll2021hyperplan').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2021hyperplan').slideToggle(600);$('#bibmoll2021hyperplan').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2021hyperplan"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll2021hyperplan</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Over the years, many motion planning algorithms have been
        proposed. It is often unclear which algorithm might be best suited for
        a particular class of problems. The problem is compounded by the fact
        that algorithm performance can be highly dependent on parameter
        settings. This paper shows that hyperparameter optimization is an
        effective tool in both algorithm selection and parameter tuning over a
        given set of motion planning problems. We present different loss
        functions for optimization that capture different notions of
        optimality. The approach is evaluated on a broad range of scenes using
        two different manipulators, a Fetch and a Baxter. We show that
        optimized planning algorithm performance significantly improves upon
        baseline performance and generalizes broadly in the sense that
        performance improvements carry over to problems that are very different
        from the ones considered during optimization."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Chamzas, Constantinos and Kingston, Zachary and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE/RSJ} Intl.\ Conf.\ on Intelligent Robots and Systems"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"{HyperPlan}: A Framework for Motion Planning Algorithm Selection and
        Parameter Optimization"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2021"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2021hyperplan"><h3>Abstract</h3><p>Over the years, many motion planning algorithms have been proposed. It is often unclear which algorithm might be best suited for a particular class of problems. The problem is compounded by the fact that algorithm performance can be highly dependent on parameter settings. This paper shows that hyperparameter optimization is an effective tool in both algorithm selection and parameter tuning over a given set of motion planning problems. We present different loss functions for optimization that capture different notions of optimality. The approach is evaluated on a broad range of scenes using two different manipulators, a Fetch and a Baxter. We show that optimized planning algorithm performance significantly improves upon baseline performance and generalizes broadly in the sense that performance improvements carry over to problems that are very different from the ones considered during optimization.</p></div></div></li><li> <span id="kingston2020informing-multi-modal-planning-with">Z. Kingston, A. M. Wells, <b>M. Moll</b>, J. M. Badger, and L. E. Kavraki, “Informing Multi-Modal Planning with Synergistic Discrete Leads,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2020.</span> <br><div id="acckingston2020informing-multi-modal-planning-with"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/kingston2020informing-multi-modal-planning-with.pdf');" href="/publications/kingston2020informing-multi-modal-planning-with.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA40945.2020.9197545');" href="https://dx.doi.org/10.1109/ICRA40945.2020.9197545"> publisher </a> <button class="btn" type="button" onclick="$('#bibkingston2020informing-multi-modal-planning-with').slideToggle(600);$('#abskingston2020informing-multi-modal-planning-with').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abskingston2020informing-multi-modal-planning-with').slideToggle(600);$('#bibkingston2020informing-multi-modal-planning-with').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibkingston2020informing-multi-modal-planning-with"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">kingston2020informing-multi-modal-planning-with</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Robotic manipulation problems are inherently continuous, but
        typically have underlying discrete structure, e.g., whether or not an
        object is grasped. This means many problems are multi-modal and in
        particular have a continuous infinity of modes. For example, in a
        pick-and-place manipulation domain, every grasp and placement of an
        object is a mode. Usually manipulation problems require the robot to
        transition into different modes, e.g., going from a mode with an object
        placed to another mode with the object grasped. To successfully find a
        manipulation plan, a planner must find a sequence of valid single-mode
        motions as well as valid transitions between these modes. Many
        manipulation planners have been proposed to solve tasks with
        multi-modal structure. However, these methods require mode-specific
        planners and fail to scale to very cluttered environments or to tasks
        that require long sequences of transitions. This paper presents a
        general layered planning approach to multi-modal planning that uses a
        discrete ``lead'' to bias search towards useful mode transitions. The
        difficulty of achieving specific mode transitions is captured online
        and used to bias search towards more promising sequences of modes. We
        demonstrate our planner on complex scenes and show that significant
        performance improvements are tied to both our discrete ``lead'' and our
        continuous representation."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Kingston, Zachary and Wells, Andrew M. and Moll, Mark and Badger, Julia M. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA40945.2020.9197545"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Informing Multi-Modal Planning with Synergistic Discrete Leads"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2020"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abskingston2020informing-multi-modal-planning-with"><h3>Abstract</h3><p>Robotic manipulation problems are inherently continuous, but typically have underlying discrete structure, e.g., whether or not an object is grasped. This means many problems are multi-modal and in particular have a continuous infinity of modes. For example, in a pick-and-place manipulation domain, every grasp and placement of an object is a mode. Usually manipulation problems require the robot to transition into different modes, e.g., going from a mode with an object placed to another mode with the object grasped. To successfully find a manipulation plan, a planner must find a sequence of valid single-mode motions as well as valid transitions between these modes. Many manipulation planners have been proposed to solve tasks with multi-modal structure. However, these methods require mode-specific planners and fail to scale to very cluttered environments or to tasks that require long sequences of transitions. This paper presents a general layered planning approach to multi-modal planning that uses a discrete “lead” to bias search towards useful mode transitions. The difficulty of achieving specific mode transitions is captured online and used to bias search towards more promising sequences of modes. We demonstrate our planner on complex scenes and show that significant performance improvements are tied to both our discrete “lead” and our continuous representation.</p></div></div></li><li> <span id="hernandez2019lazy-evaluation-of-goal-specifications">J. D. Hernández, <b>M. Moll</b>, and L. E. Kavraki, “Lazy Evaluation of Goal Specifications Guided by Motion Planning,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2019, pp. 944–950.</span> <br><div id="acchernandez2019lazy-evaluation-of-goal-specifications"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/hernandez2019lazy-evaluation-of-goal-specifications.pdf');" href="/publications/hernandez2019lazy-evaluation-of-goal-specifications.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA.2019.8793570');" href="https://dx.doi.org/10.1109/ICRA.2019.8793570"> publisher </a> <button class="btn" type="button" onclick="$('#bibhernandez2019lazy-evaluation-of-goal-specifications').slideToggle(600);$('#abshernandez2019lazy-evaluation-of-goal-specifications').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abshernandez2019lazy-evaluation-of-goal-specifications').slideToggle(600);$('#bibhernandez2019lazy-evaluation-of-goal-specifications').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibhernandez2019lazy-evaluation-of-goal-specifications"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">hernandez2019lazy-evaluation-of-goal-specifications</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Nowadays robotic systems are expected to share workspaces and
        collaborate with humans. In such collaborative environments, an
        important challenge is to ground or establish the correct semantic
        interpretation of a human request. Once such an interpretation is
        available, the request must be translated into robot motion commands in
        order to complete the desired task. It is not unusual that a human
        request cannot be grounded to a unique interpretation, thus leading to
        an ambiguous request. A simple example is to ask a robot to ``put a cup
        on the table,'' when there are multiple cups available. In order to
        deal with this kind of ambiguous request, we propose a delayed or lazy
        variable grounding. The focus of this paper is a motion planning
        algorithm that, given goal regions that represent different valid
        groundings, lazily finds a feasible path to any one valid grounding.
        This algorithm includes a rewardpenalty strategy, which attempts to
        prioritize those goal regions that seem more promising to provide a
        solution. We validate our approach by solving requests with multiple
        valid alternatives in both simulation and real-world experiments."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Hern{\'a}ndez, Juan David and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA.2019.8793570"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">may</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"944--950"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Lazy Evaluation of Goal Specifications Guided by Motion Planning"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2019"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abshernandez2019lazy-evaluation-of-goal-specifications"><h3>Abstract</h3><p>Nowadays robotic systems are expected to share workspaces and collaborate with humans. In such collaborative environments, an important challenge is to ground or establish the correct semantic interpretation of a human request. Once such an interpretation is available, the request must be translated into robot motion commands in order to complete the desired task. It is not unusual that a human request cannot be grounded to a unique interpretation, thus leading to an ambiguous request. A simple example is to ask a robot to “put a cup on the table,” when there are multiple cups available. In order to deal with this kind of ambiguous request, we propose a delayed or lazy variable grounding. The focus of this paper is a motion planning algorithm that, given goal regions that represent different valid groundings, lazily finds a feasible path to any one valid grounding. This algorithm includes a rewardpenalty strategy, which attempts to prioritize those goal regions that seem more promising to provide a solution. We validate our approach by solving requests with multiple valid alternatives in both simulation and real-world experiments.</p></div></div></li><li> <span id="vidal2019online-multilayered-motion-planning">E. Vidal Garcia, <b>M. Moll</b>, N. Palomeras, J. D. Hernández, M. Carreras, and L. Kavraki, “Online Multilayered Motion Planning with Dynamic Constraints for Autonomous Underwater Vehicles,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2019, pp. 8936–8942.</span> <br><div id="accvidal2019online-multilayered-motion-planning"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/vidal2019online-multilayered-motion-planning.pdf');" href="/publications/vidal2019online-multilayered-motion-planning.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA.2019.8794009');" href="https://dx.doi.org/10.1109/ICRA.2019.8794009"> publisher </a> <button class="btn" type="button" onclick="$('#bibvidal2019online-multilayered-motion-planning').slideToggle(600);$('#absvidal2019online-multilayered-motion-planning').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absvidal2019online-multilayered-motion-planning').slideToggle(600);$('#bibvidal2019online-multilayered-motion-planning').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibvidal2019online-multilayered-motion-planning"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">vidal2019online-multilayered-motion-planning</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Underwater robots are subject to complex hydrodynamic forces.
        These forces define how the vehicle moves, so it is important to
        consider them when planning trajectories. However, performing motion
        planning considering the dynamics on the robot's onboard computer is
        challenging due to the limited computational resources available. In
        this paper an efficient motion planning framework for autonomous
        underwater vehicles (AUVs) is presented. By introducing a loosely
        coupled multilayered planning design, our framework is able to generate
        dynamically feasible trajectories while keeping the planning time low
        enough for online planning. First, a fast path planner operating in a
        lower-dimensional projected space computes a lead path from the start
        to the goal configuration. Then, the lead path is used to bias the
        sampling of a second motion planner, which takes into account all the
        dynamic constraints. Furthermore, we propose a strategy for online
        planning that saves computational resources by generating the final
        trajectory only up to a finite horizon. By using the finite horizon
        strategy together with the multilayered approach, the sampling of the
        second planner focuses on regions where good quality solutions are more
        likely to be found, significantly reducing the planning time. To
        provide strong safety guarantees our framework also incorporates the
        conservative approximations of inevitable collision states (ICSs).
        Finally, we present simulations and experiments using a real underwater
        robot to demonstrate the capabilities of our framework."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Vidal Garcia, Eduard and Moll, Mark and Palomeras, N. and Hern{\'a}ndez, Juan David and Carreras, Marc and Kavraki, Lydia"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA.2019.8794009"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">may</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"8936--8942"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Online Multilayered Motion Planning with Dynamic Constraints for
        Autonomous Underwater Vehicles"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2019"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absvidal2019online-multilayered-motion-planning"><h3>Abstract</h3><p>Underwater robots are subject to complex hydrodynamic forces. These forces define how the vehicle moves, so it is important to consider them when planning trajectories. However, performing motion planning considering the dynamics on the robot’s onboard computer is challenging due to the limited computational resources available. In this paper an efficient motion planning framework for autonomous underwater vehicles (AUVs) is presented. By introducing a loosely coupled multilayered planning design, our framework is able to generate dynamically feasible trajectories while keeping the planning time low enough for online planning. First, a fast path planner operating in a lower-dimensional projected space computes a lead path from the start to the goal configuration. Then, the lead path is used to bias the sampling of a second motion planner, which takes into account all the dynamic constraints. Furthermore, we propose a strategy for online planning that saves computational resources by generating the final trajectory only up to a finite horizon. By using the finite horizon strategy together with the multilayered approach, the sampling of the second planner focuses on regions where good quality solutions are more likely to be found, significantly reducing the planning time. To provide strong safety guarantees our framework also incorporates the conservative approximations of inevitable collision states (ICSs). Finally, we present simulations and experiments using a real underwater robot to demonstrate the capabilities of our framework.</p></div></div></li><li> <span id="baker2017robonaut-2-and-you">W. Baker, Z. Kingston, <b>M. Moll</b>, J. Badger, and L. Kavraki, “Robonaut 2 and You: Specifying and Executing Complex Operations,” in <i>IEEE Workshop on Advanced Robotics and its Social Impacts (ARSO)</i>, Austin, TX, 2017.</span> <br><div id="accbaker2017robonaut-2-and-you"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/baker2017robonaut-2-and-you.pdf');" href="/publications/baker2017robonaut-2-and-you.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ARSO.2017.8025204');" href="https://dx.doi.org/10.1109/ARSO.2017.8025204"> publisher </a> <button class="btn" type="button" onclick="$('#bibbaker2017robonaut-2-and-you').slideToggle(600);$('#absbaker2017robonaut-2-and-you').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absbaker2017robonaut-2-and-you').slideToggle(600);$('#bibbaker2017robonaut-2-and-you').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibbaker2017robonaut-2-and-you"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">baker2017robonaut-2-and-you</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Crew time is a precious resource due to the expense of trained
        human operators in space. Efficient caretaker robots could lessen the
        manual labor load required by frequent vehicular and life support
        maintenance tasks, freeing astronaut time for scientific mission
        objectives. Humanoid robots can fluidly exist alongside human
        counterparts due to their form, but they are complex and
        high-dimensional platforms. This paper describes a system that human
        operators can use to maneuver Robonaut 2 (R2), a dexterous humanoid
        robot developed by NASA to research co-robotic applications. The system
        includes a specification of constraints used to describe operations,
        and the supporting planning framework that solves constrained problems
        on R2 at interactive speeds. The paper is developed in reference to an
        illustrative, typical example of an operation R2 performs to highlight
        the challenges inherent to the problems R2 must face. Finally, the
        interface and planner is validated through a case-study using the
        guiding example on the physical robot in a simulated microgravity
        environment. This work reveals the complexity of employing humanoid
        caretaker robots and suggest solutions that are broadly applicable."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Austin, TX"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Baker, William and Kingston, Zachary and Moll, Mark and Badger, Julia and Kavraki, Lydia"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"IEEE Workshop on Advanced Robotics and its Social Impacts
        (ARSO)"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ARSO.2017.8025204"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Robonaut 2 and You: Specifying and Executing Complex Operations"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2017"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absbaker2017robonaut-2-and-you"><h3>Abstract</h3><p>Crew time is a precious resource due to the expense of trained human operators in space. Efficient caretaker robots could lessen the manual labor load required by frequent vehicular and life support maintenance tasks, freeing astronaut time for scientific mission objectives. Humanoid robots can fluidly exist alongside human counterparts due to their form, but they are complex and high-dimensional platforms. This paper describes a system that human operators can use to maneuver Robonaut 2 (R2), a dexterous humanoid robot developed by NASA to research co-robotic applications. The system includes a specification of constraints used to describe operations, and the supporting planning framework that solves constrained problems on R2 at interactive speeds. The paper is developed in reference to an illustrative, typical example of an operation R2 performs to highlight the challenges inherent to the problems R2 must face. Finally, the interface and planner is validated through a case-study using the guiding example on the physical robot in a simulated microgravity environment. This work reveals the complexity of employing humanoid caretaker robots and suggest solutions that are broadly applicable.</p></div></div></li><li> <span id="hernandez2016planning-feasible-and-safe-paths">J. D. Hernández, <b>M. Moll</b>, E. Vidal Garcia, M. Carreras, and L. E. Kavraki, “Planning Feasible and Safe Paths Online for Autonomous Underwater Vehicles in Unknown Environments,” in <i>IEEE/RSJ Intl. Conf. on Intelligent Robots and Systems</i>, 2016, pp. 1313–1320.</span> <br><div id="acchernandez2016planning-feasible-and-safe-paths"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/hernandez2016planning-feasible-and-safe-paths.pdf');" href="/publications/hernandez2016planning-feasible-and-safe-paths.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/IROS.2016.7759217');" href="https://dx.doi.org/10.1109/IROS.2016.7759217"> publisher </a> <button class="btn" type="button" onclick="$('#bibhernandez2016planning-feasible-and-safe-paths').slideToggle(600);$('#abshernandez2016planning-feasible-and-safe-paths').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abshernandez2016planning-feasible-and-safe-paths').slideToggle(600);$('#bibhernandez2016planning-feasible-and-safe-paths').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibhernandez2016planning-feasible-and-safe-paths"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">hernandez2016planning-feasible-and-safe-paths</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a framework for planning collision-free and safe paths
        online for autonomous underwater vehicles (AUVs) in unknown
        environments. We build up on our previous work and propose an improved
        approach. While preserving its main modules (mapping, planning and
        mission handler), the framework now considers motion constraints to
        plan feasible paths, i.e., those that meet vehicle's motion
        capabilities. The new framework also incorporates a risk function to
        avoid navigating close to nearby obstacles, and reuses the last best
        known solution to eliminate time-consuming pruning routines. To
        evaluate this approach, we use the Sparus II AUV, a torpedo-shaped
        vehicle performing autonomous missions in a 2-dimensional workspace. We
        validate the framework's new features by solving tasks in both
        simulation and real-world in water trials and comparing results with
        our previous approach."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Hern{\'a}ndez, Juan David and Moll, Mark and Vidal Garcia, Eduard and Carreras, Marc and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE/RSJ} Intl.\ Conf.\ on Intelligent Robots and Systems"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/IROS.2016.7759217"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"1313-1320"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Planning Feasible and Safe Paths Online for Autonomous Underwater
        Vehicles in Unknown Environments"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2016"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abshernandez2016planning-feasible-and-safe-paths"><h3>Abstract</h3><p>We present a framework for planning collision-free and safe paths online for autonomous underwater vehicles (AUVs) in unknown environments. We build up on our previous work and propose an improved approach. While preserving its main modules (mapping, planning and mission handler), the framework now considers motion constraints to plan feasible paths, i.e., those that meet vehicle’s motion capabilities. The new framework also incorporates a risk function to avoid navigating close to nearby obstacles, and reuses the last best known solution to eliminate time-consuming pruning routines. To evaluate this approach, we use the Sparus II AUV, a torpedo-shaped vehicle performing autonomous missions in a 2-dimensional workspace. We validate the framework’s new features by solving tasks in both simulation and real-world in water trials and comparing results with our previous approach.</p></div></div></li><li> <span id="kim2016an-evaluation-of-different-clustering-methods">S. M. Kim, M. I. Peña, <b>M. Moll</b>, G. Giannakopoulos, G. N. Bennett, and L. E. Kavraki, “An Evaluation of Different Clustering Methods and Distance Measures Used for Grouping Metabolic Pathways,” in <i>Eighth International Conference on Bioinformatics and Computational Biology (BICoB)</i>, 2016.</span> <br><div id="acckim2016an-evaluation-of-different-clustering-methods"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/kim2016an-evaluation-of-different-clustering-methods.pdf');" href="/publications/kim2016an-evaluation-of-different-clustering-methods.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibkim2016an-evaluation-of-different-clustering-methods').slideToggle(600);$('#abskim2016an-evaluation-of-different-clustering-methods').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abskim2016an-evaluation-of-different-clustering-methods').slideToggle(600);$('#bibkim2016an-evaluation-of-different-clustering-methods').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibkim2016an-evaluation-of-different-clustering-methods"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">kim2016an-evaluation-of-different-clustering-methods</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Large-scale annotated metabolic databases, such as KEGG and
        MetaCyc, provide a wealth of information to researchers designing novel
        biosynthetic pathways. However, many metabolic pathfinding tools that
        assist in identifying possible solution pathways fail to facilitate the
        grouping and interpretation of these pathway results. Clustering
        possible solution pathways can help users of pathfinding tools quickly
        notice major patterns and unique pathways without having to sift
        through individual results one by one. In this paper, we assess the
        ability of three separate clustering methods (hierarchical, k-means,
        and k-medoids) along with three pair-wise distance measures
        (Levenshtein, Jaccard, and n-gram) to expertly group lysine,
        isoleucine, and 3-hydroxypropanoic acid (3-HP) biosynthesis pathways.
        The quality of the resulting clusters were quantitatively evaluated
        against expected pathway groupings taken from the literature.
        Hierarchical clustering and Levenshtein distance seemed to best match
        external pathway labels across the three biosynthesis pathways. The
        lysine biosynthesis pathways, which had the most distinct separation of
        pathways, had better quality clusters than isoleucine and 3-HP, showing
        that clustering methods may not be capable of grouping pathways with
        more complex underlying topographies."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Kim, Sarah M. and Pe{\~n}a, Matthew I. and Moll, Mark and Giannakopoulos, George and Bennett, George N. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Eighth International Conference on Bioinformatics and
        Computational Biology (BICoB)"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">apr</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"An Evaluation of Different Clustering Methods and Distance Measures
        Used for Grouping Metabolic Pathways"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2016"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abskim2016an-evaluation-of-different-clustering-methods"><h3>Abstract</h3><p>Large-scale annotated metabolic databases, such as KEGG and MetaCyc, provide a wealth of information to researchers designing novel biosynthetic pathways. However, many metabolic pathfinding tools that assist in identifying possible solution pathways fail to facilitate the grouping and interpretation of these pathway results. Clustering possible solution pathways can help users of pathfinding tools quickly notice major patterns and unique pathways without having to sift through individual results one by one. In this paper, we assess the ability of three separate clustering methods (hierarchical, k-means, and k-medoids) along with three pair-wise distance measures (Levenshtein, Jaccard, and n-gram) to expertly group lysine, isoleucine, and 3-hydroxypropanoic acid (3-HP) biosynthesis pathways. The quality of the resulting clusters were quantitatively evaluated against expected pathway groupings taken from the literature. Hierarchical clustering and Levenshtein distance seemed to best match external pathway labels across the three biosynthesis pathways. The lysine biosynthesis pathways, which had the most distinct separation of pathways, had better quality clusters than isoleucine and 3-HP, showing that clustering methods may not be capable of grouping pathways with more complex underlying topographies.</p></div></div></li><li> <span id="novinskaya2015improving-protein-conformational-sampling">A. Novinskaya, D. Devaurs, <b>M. Moll</b>, and L. E. Kavraki, “Improving Protein Conformational Sampling by Using Guiding Projections,” in <i>IEEE Intl. Conf. on Bioinformatics and Biomedicine Workshops (BIBMW)</i>, 2015, pp. 1272–1279.</span> <br><div id="accnovinskaya2015improving-protein-conformational-sampling"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/novinskaya2015improving-protein-conformational-sampling.pdf');" href="/publications/novinskaya2015improving-protein-conformational-sampling.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/BIBM.2015.7359863');" href="https://dx.doi.org/10.1109/BIBM.2015.7359863"> publisher </a> <button class="btn" type="button" onclick="$('#bibnovinskaya2015improving-protein-conformational-sampling').slideToggle(600);$('#absnovinskaya2015improving-protein-conformational-sampling').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absnovinskaya2015improving-protein-conformational-sampling').slideToggle(600);$('#bibnovinskaya2015improving-protein-conformational-sampling').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibnovinskaya2015improving-protein-conformational-sampling"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">novinskaya2015improving-protein-conformational-sampling</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Sampling-based motion planning algorithms from the field of
        robotics have been very successful in exploring the conformational
        space of proteins. However, studying the flexibility of large proteins
        with hundreds or thousands of Degrees of Freedom (DoFs) remains a big
        challenge. Large proteins are also highly-constrained systems, which
        makes them more challenging for standard robotic approaches. So-called
        ``expansive'' motion planning algorithms were specifically developed to
        address highly-dimensional and highly-constrained problems. Many such
        planners employ a low- dimensional projection to estimate exploration
        coverage and direct their search based on this information. We believe
        that such a projection plays an essential role in the success of these
        planners. This paper shows how the low-dimensional projection used by
        expansive planners can be tailored with respect to a given molecular
        system to enhance the process of conformational sampling. We introduce
        a methodology to generate an expert projection using any available
        information about a given protein. We evaluate this methodology on
        several conformational search problems involving proteins with hundreds
        of DoFs. Our experiments demonstrate that incorporating expert
        knowledge into the projection can significantly benefit the exploration
        process."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Novinskaya, Anastasia and Devaurs, Didier and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Bioinformatics and Biomedicine Workshops
        (BIBMW)"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/BIBM.2015.7359863"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">nov</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"1272-1279"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Improving Protein Conformational Sampling by Using Guiding
        Projections"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2015"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absnovinskaya2015improving-protein-conformational-sampling"><h3>Abstract</h3><p>Sampling-based motion planning algorithms from the field of robotics have been very successful in exploring the conformational space of proteins. However, studying the flexibility of large proteins with hundreds or thousands of Degrees of Freedom (DoFs) remains a big challenge. Large proteins are also highly-constrained systems, which makes them more challenging for standard robotic approaches. So-called “expansive” motion planning algorithms were specifically developed to address highly-dimensional and highly-constrained problems. Many such planners employ a low- dimensional projection to estimate exploration coverage and direct their search based on this information. We believe that such a projection plays an essential role in the success of these planners. This paper shows how the low-dimensional projection used by expansive planners can be tailored with respect to a given molecular system to enhance the process of conformational sampling. We introduce a methodology to generate an expert projection using any available information about a given protein. We evaluate this methodology on several conformational search problems involving proteins with hundreds of DoFs. Our experiments demonstrate that incorporating expert knowledge into the projection can significantly benefit the exploration process.</p></div></div></li><li> <span id="moll2015structure-guided-selection-of-specificity-determining"><b>M. Moll</b>, P. W. Finn, and L. E. Kavraki, “Structure-Guided Selection of Specificity Determining Positions in the Human Kinome,” in <i>IEEE International Conference on Bioinformatics and Biomedicine (BIBM)</i>, 2015, pp. 21–28.</span> <br><div id="accmoll2015structure-guided-selection-of-specificity-determining"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2015structure-guided-selection-of-specificity-determining.pdf');" href="/publications/moll2015structure-guided-selection-of-specificity-determining.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/BIBM.2015.7359650');" href="https://dx.doi.org/10.1109/BIBM.2015.7359650"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2015structure-guided-selection-of-specificity-determining').slideToggle(600);$('#absmoll2015structure-guided-selection-of-specificity-determining').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2015structure-guided-selection-of-specificity-determining').slideToggle(600);$('#bibmoll2015structure-guided-selection-of-specificity-determining').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2015structure-guided-selection-of-specificity-determining"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll2015structure-guided-selection-of-specificity-determining</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"It is well-known that inhibitors of protein kinases bind with
        very different selectivity profiles. This is also the case for
        inhibitors of many other protein families. A better understanding of
        binding selectivity would enhance the design of drugs that target only
        a subfamily, thereby minimizing possible side-effects. The increased
        availability of protein 3D structures has made it possible to study the
        structural variation within a given protein family. However, not every
        structural variation is related to binding specificity. We propose a
        greedy algorithm that computes a subset of residue positions in a
        multiple sequence alignment such that structural and chemical variation
        in those positions helps explain known binding affinities. By providing
        this information, the main purpose of the algorithm is to provide
        experimentalists with possible insights into how the selectivity
        profile of certain inhibitors is achieved, which is useful for lead
        optimization. In addition, the algorithm can also be used to predict
        binding affinities for structures whose affinity for a given inhibitor
        is unknown. The algorithm's performance is demonstrated using an
        extensive dataset for the human kinome, which includes a large and
        important set of drug targets. We show that the binding affinity of 38
        different kinase inhibitors can be explained with consistently high
        precision and accuracy using the variation of at most six residue
        positions in the kinome binding site. "</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Finn, Paul W. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"IEEE International Conference on Bioinformatics and Biomedicine
        (BIBM)"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/BIBM.2015.7359650"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">nov</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"21--28"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Structure-Guided Selection of Specificity Determining Positions in
        the Human Kinome"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2015"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2015structure-guided-selection-of-specificity-determining"><h3>Abstract</h3><p>It is well-known that inhibitors of protein kinases bind with very different selectivity profiles. This is also the case for inhibitors of many other protein families. A better understanding of binding selectivity would enhance the design of drugs that target only a subfamily, thereby minimizing possible side-effects. The increased availability of protein 3D structures has made it possible to study the structural variation within a given protein family. However, not every structural variation is related to binding specificity. We propose a greedy algorithm that computes a subset of residue positions in a multiple sequence alignment such that structural and chemical variation in those positions helps explain known binding affinities. By providing this information, the main purpose of the algorithm is to provide experimentalists with possible insights into how the selectivity profile of certain inhibitors is achieved, which is useful for lead optimization. In addition, the algorithm can also be used to predict binding affinities for structures whose affinity for a given inhibitor is unknown. The algorithm’s performance is demonstrated using an extensive dataset for the human kinome, which includes a large and important set of drug targets. We show that the binding affinity of 38 different kinase inhibitors can be explained with consistently high precision and accuracy using the variation of at most six residue positions in the kinome binding site.</p></div></div></li><li> <span id="voss2015a-heuristic-approach-to-finding-diverse">C. Voss, <b>M. Moll</b>, and L. E. Kavraki, “A Heuristic Approach to Finding Diverse Short Paths,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2015, pp. 4173–4179.</span> <br><div id="accvoss2015a-heuristic-approach-to-finding-diverse"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/voss2015a-heuristic-approach-to-finding-diverse.pdf');" href="/publications/voss2015a-heuristic-approach-to-finding-diverse.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA.2015.7139774');" href="https://dx.doi.org/10.1109/ICRA.2015.7139774"> publisher </a> <button class="btn" type="button" onclick="$('#bibvoss2015a-heuristic-approach-to-finding-diverse').slideToggle(600);$('#absvoss2015a-heuristic-approach-to-finding-diverse').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absvoss2015a-heuristic-approach-to-finding-diverse').slideToggle(600);$('#bibvoss2015a-heuristic-approach-to-finding-diverse').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibvoss2015a-heuristic-approach-to-finding-diverse"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">voss2015a-heuristic-approach-to-finding-diverse</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present an algorithm that seeks to find a set of diverse,
        short paths through a roadmap graph. The usefulness of a such a set is
        illustrated in robotic motion planning and routing applications wherein
        a precomputed roadmap of the environment is partially invalidated by
        some change, for example, relocation of obstacles or modification of
        the robot. Our algorithm employs the heuristic that configurations near
        each other are likely to be invalidated by the same change in the
        environment. To find short, diverse paths, the algorithm finds a detour
        that is the shortest path avoiding a selection of balls in the
        configuration space. Different collections of these balls, or simulated
        obstacles, yield different and diverse short paths. Paths may then be
        checked for validity as a cheap alternative to checking or
        reconstructing the entire roadmap. We describe a formal definition of
        path set diversity and several measures on which to evaluate our
        algorithm. We compare the speed and quality of our heuristic
        algorithm's results against an exact algorithm that computes the
        optimally shortest set of paths on the roadmap having a minimum
        diversity. We will show that, with a tolerable loss in path shortness,
        our algorithm produces equally diverse path sets orders of magnitude
        faster."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Voss, Caleb and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA.2015.7139774"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">may</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"4173--4179"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"A Heuristic Approach to Finding Diverse Short Paths"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2015"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absvoss2015a-heuristic-approach-to-finding-diverse"><h3>Abstract</h3><p>We present an algorithm that seeks to find a set of diverse, short paths through a roadmap graph. The usefulness of a such a set is illustrated in robotic motion planning and routing applications wherein a precomputed roadmap of the environment is partially invalidated by some change, for example, relocation of obstacles or modification of the robot. Our algorithm employs the heuristic that configurations near each other are likely to be invalidated by the same change in the environment. To find short, diverse paths, the algorithm finds a detour that is the shortest path avoiding a selection of balls in the configuration space. Different collections of these balls, or simulated obstacles, yield different and diverse short paths. Paths may then be checked for validity as a cheap alternative to checking or reconstructing the entire roadmap. We describe a formal definition of path set diversity and several measures on which to evaluate our algorithm. We compare the speed and quality of our heuristic algorithm’s results against an exact algorithm that computes the optimally shortest set of paths on the roadmap having a minimum diversity. We will show that, with a tolerable loss in path shortness, our algorithm produces equally diverse path sets orders of magnitude faster.</p></div></div></li><li> <span id="coleman2015experience-based-planning-with-sparse">D. Coleman, I. A. Şucan, <b>M. Moll</b>, K. Okada, and N. Correll, “Experience-Based Planning with Sparse Roadmap Spanners,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2015, pp. 900–905.</span> <br><div id="acccoleman2015experience-based-planning-with-sparse"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/coleman2015experience-based-planning-with-sparse.pdf');" href="/publications/coleman2015experience-based-planning-with-sparse.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA.2015.7139284');" href="https://dx.doi.org/10.1109/ICRA.2015.7139284"> publisher </a> <button class="btn" type="button" onclick="$('#bibcoleman2015experience-based-planning-with-sparse').slideToggle(600);$('#abscoleman2015experience-based-planning-with-sparse').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abscoleman2015experience-based-planning-with-sparse').slideToggle(600);$('#bibcoleman2015experience-based-planning-with-sparse').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibcoleman2015experience-based-planning-with-sparse"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">coleman2015experience-based-planning-with-sparse</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present an experienced-based planning framework called Thunder
        that learns to reduce computation time required to solve
        high-dimensional planning problems in varying environments. The
        approach is especially suited for large configuration spaces that
        include many invariant constraints, such as those found with whole body
        humanoid motion planning. Experiences are generated using probabilistic
        sampling and stored in a sparse roadmap spanner (SPARS), which provides
        asymptotically near-optimal coverage of the configuration space, making
        storing, retrieving, and repairing past experiences very efficient with
        respect to memory and time. The Thunder framework improves upon past
        experience- based planners by storing experiences in a graph rather
        than in individual paths, eliminating redundant information, providing
        more opportunities for path reuse, and providing a theoretical limit to
        the size of the experience graph. These properties also lead to
        improved handling of dynamically changing environments, reasoning about
        optimal paths, and reducing query resolution time. The approach is
        demonstrated on a 30 degrees of freedom humanoid robot and compared
        with the Lightning framework, an experience-based planner that uses
        individual paths to store past experiences. In environments with
        variable obstacles and stability constraints, experiments show that
        Thunder is on average an order of magnitude faster than Lightning and
        planning from scratch. Thunder also uses 98.8% less memory to store its
        experiences after 10,000 trials when compared to Lightning. Our
        framework is implemented and freely available in the Open Motion
        Planning Library."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Coleman, David and {\c{S}}ucan, Ioan A. and Moll, Mark and Okada, Kei and Correll, Nikolaus"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA.2015.7139284"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">may</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"900--905"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Experience-Based Planning with Sparse Roadmap Spanners"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2015"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abscoleman2015experience-based-planning-with-sparse"><h3>Abstract</h3><p>We present an experienced-based planning framework called Thunder that learns to reduce computation time required to solve high-dimensional planning problems in varying environments. The approach is especially suited for large configuration spaces that include many invariant constraints, such as those found with whole body humanoid motion planning. Experiences are generated using probabilistic sampling and stored in a sparse roadmap spanner (SPARS), which provides asymptotically near-optimal coverage of the configuration space, making storing, retrieving, and repairing past experiences very efficient with respect to memory and time. The Thunder framework improves upon past experience- based planners by storing experiences in a graph rather than in individual paths, eliminating redundant information, providing more opportunities for path reuse, and providing a theoretical limit to the size of the experience graph. These properties also lead to improved handling of dynamically changing environments, reasoning about optimal paths, and reducing query resolution time. The approach is demonstrated on a 30 degrees of freedom humanoid robot and compared with the Lightning framework, an experience-based planner that uses individual paths to store past experiences. In environments with variable obstacles and stability constraints, experiments show that Thunder is on average an order of magnitude faster than Lightning and planning from scratch. Thunder also uses 98.8% less memory to store its experiences after 10,000 trials when compared to Lightning. Our framework is implemented and freely available in the Open Motion Planning Library.</p></div></div></li><li> <span id="luna2014optimal-and-efficient-stochastic-motion">R. Luna, M. Lahijanian, <b>M. Moll</b>, and L. E. Kavraki, “Optimal and Efficient Stochastic Motion Planning in Partially-Known Environments,” in <i>28th AAAI Conference on Artificial Intelligence (AAAI-14)</i>, Québec City, Canada, 2014, pp. 2549–2555.</span> <br><div id="accluna2014optimal-and-efficient-stochastic-motion"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/luna2014optimal-and-efficient-stochastic-motion.pdf');" href="/publications/luna2014optimal-and-efficient-stochastic-motion.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://www.aaai.org/ocs/index.php/AAAI/AAAI14/paper/view/8457');" href="https://www.aaai.org/ocs/index.php/AAAI/AAAI14/paper/view/8457"> publisher </a> <button class="btn" type="button" onclick="$('#bibluna2014optimal-and-efficient-stochastic-motion').slideToggle(600);$('#absluna2014optimal-and-efficient-stochastic-motion').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absluna2014optimal-and-efficient-stochastic-motion').slideToggle(600);$('#bibluna2014optimal-and-efficient-stochastic-motion').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibluna2014optimal-and-efficient-stochastic-motion"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">luna2014optimal-and-efficient-stochastic-motion</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"A framework capable of computing optimal control policies for a
        continuous system in the presence of both action and environment
        uncertainty is presented in this work. The framework decomposes the
        planning problem into two stages: an offline phase that reasons only
        over action uncertainty and an online phase that quickly reacts to the
        uncertain environment. Offline, a bounded-parameter Markov decision
        process (BMDP) is employed to model the evolution of the stochastic
        system over a discretization of the environment. Online, an optimal
        control policy over the BMDP is computed. Upon the discovery of an
        unknown environment feature during policy execution, the BMDP is
        updated and the optimal control policy is efficiently recomputed.
        Depending on the desired quality of the control policy, a suite of
        methods is presented to incorporate new information into the BMDP with
        varying degrees of detail online. Experiments confirm that the
        framework recomputes high-quality policies in seconds and is orders of
        magnitude faster than existing methods."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Qu\'ebec City, Canada"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Luna, Ryan and Lahijanian, Morteza and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"28th AAAI Conference on Artificial Intelligence (AAAI-14)"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jul</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"2549--2555"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Optimal and Efficient Stochastic Motion Planning in Partially-Known
        Environments"</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">"https://www.aaai.org/ocs/index.php/AAAI/AAAI14/paper/view/8457"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2014"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absluna2014optimal-and-efficient-stochastic-motion"><h3>Abstract</h3><p>A framework capable of computing optimal control policies for a continuous system in the presence of both action and environment uncertainty is presented in this work. The framework decomposes the planning problem into two stages: an offline phase that reasons only over action uncertainty and an online phase that quickly reacts to the uncertain environment. Offline, a bounded-parameter Markov decision process (BMDP) is employed to model the evolution of the stochastic system over a discretization of the environment. Online, an optimal control policy over the BMDP is computed. Upon the discovery of an unknown environment feature during policy execution, the BMDP is updated and the optimal control policy is efficiently recomputed. Depending on the desired quality of the control policy, a suite of methods is presented to incorporate new information into the BMDP with varying degrees of detail online. Experiments confirm that the framework recomputes high-quality policies in seconds and is orders of magnitude faster than existing methods.</p></div></div></li><li> <span id="wang2014active-planning-sensing-and-recognition">Z. Wang, Z. Wang, <b>M. Moll</b>, P.-S. Huang, D. Grady, N. Nasrabadi, T. Huang, L. Kavraki, and M. Hasegawa-Johnson, “Active Planning, Sensing and Recognition Using a Resource-Constrained Discriminant POMDP,” in <i>IEEE/ISPRS Workshop on Multi-Sensor Fusion for Outdoor Dynamic Scene Understanding at CVPR</i>, 2014.</span> <br><div id="accwang2014active-planning-sensing-and-recognition"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/wang2014active-planning-sensing-and-recognition.pdf');" href="/publications/wang2014active-planning-sensing-and-recognition.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/CVPRW.2014.116');" href="https://dx.doi.org/10.1109/CVPRW.2014.116"> publisher </a> <button class="btn" type="button" onclick="$('#bibwang2014active-planning-sensing-and-recognition').slideToggle(600);$('#abswang2014active-planning-sensing-and-recognition').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abswang2014active-planning-sensing-and-recognition').slideToggle(600);$('#bibwang2014active-planning-sensing-and-recognition').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibwang2014active-planning-sensing-and-recognition"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">wang2014active-planning-sensing-and-recognition</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"In this paper, we address the problem of object class recognition
        via observations from actively selected views/modalities/features under
        limited resource budgets. A Partially Observable Markov Decision
        Process (POMDP) is employed to find optimal sensing and recognition
        actions with the goal of long-term classification accuracy. Hetero-
        geneous resource constraints -- such as motion, number of measurements
        and bandwidth -- are explicitly modeled in the state variable, and a
        prohibitively high penalty is used to prevent the violation of any
        resource constraint. To improve recognition performance, we further
        incorporate discriminative classification models with POMDP, and
        customize the reward function and observation model correspondingly.
        The proposed model is validated on several data sets for multi-view,
        multi-modal vehicle classification and multi-view face recognition, and
        demonstrates improvement in both recognition and resource management
        over greedy methods and previous POMDP formulations."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Wang, Zhaowen and Wang, Zhangyang and Moll, Mark and Huang, Po-Sen and Grady, Devin and Nasrabadi, Nasser and Huang, Thomas and Kavraki, Lydia and Hasegawa-Johnson, Mark"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"IEEE/ISPRS Workshop on Multi-Sensor Fusion for Outdoor Dynamic
        Scene Understanding at CVPR"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/CVPRW.2014.116"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Active Planning, Sensing and Recognition Using a
        Resource-Constrained Discriminant {POMDP}"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2014"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abswang2014active-planning-sensing-and-recognition"><h3>Abstract</h3><p>In this paper, we address the problem of object class recognition via observations from actively selected views/modalities/features under limited resource budgets. A Partially Observable Markov Decision Process (POMDP) is employed to find optimal sensing and recognition actions with the goal of long-term classification accuracy. Hetero- geneous resource constraints – such as motion, number of measurements and bandwidth – are explicitly modeled in the state variable, and a prohibitively high penalty is used to prevent the violation of any resource constraint. To improve recognition performance, we further incorporate discriminative classification models with POMDP, and customize the reward function and observation model correspondingly. The proposed model is validated on several data sets for multi-view, multi-modal vehicle classification and multi-view face recognition, and demonstrates improvement in both recognition and resource management over greedy methods and previous POMDP formulations.</p></div></div></li><li> <span id="nedunuri2014smt-based-synthesis-of-integrated-task">S. Nedunuri, S. Prabhu, <b>M. Moll</b>, S. Chaudhuri, and L. E. Kavraki, “SMT-Based Synthesis of Integrated Task and Motion Plans from Plan Outlines,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2014, pp. 655–662.</span> <br><div id="accnedunuri2014smt-based-synthesis-of-integrated-task"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/nedunuri2014smt-based-synthesis-of-integrated-task.pdf');" href="/publications/nedunuri2014smt-based-synthesis-of-integrated-task.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA.2014.6906924');" href="https://dx.doi.org/10.1109/ICRA.2014.6906924"> publisher </a> <button class="btn" type="button" onclick="$('#bibnedunuri2014smt-based-synthesis-of-integrated-task').slideToggle(600);$('#absnedunuri2014smt-based-synthesis-of-integrated-task').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absnedunuri2014smt-based-synthesis-of-integrated-task').slideToggle(600);$('#bibnedunuri2014smt-based-synthesis-of-integrated-task').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibnedunuri2014smt-based-synthesis-of-integrated-task"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">nedunuri2014smt-based-synthesis-of-integrated-task</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a new approach to integrated task and motion planning
        (ITMP) for robots performing mobile manipulation. In our approach, the
        user writes a high-level specification that captures partial knowledge
        about a mobile manipulation setting. In particular, this specification
        includes a plan outline that syntactically defines a space of plausible
        integrated plans, a set of logical requirements that the generated plan
        must satisfy, and a description of the physical space that the robot
        manipulates. A synthesis algorithm is now used to search for an
        integrated plan that falls within the space defined by the plan
        outline, and also satisfies all requirements. Our synthesis algorithm
        complements continuous motion planning algorithms with calls to a
        Satisfiability Modulo Theories (SMT) solver. From the scene
        description, a motion planning algorithm is used to construct a
        placement graph, an abstraction of a manipulation graph whose paths
        represent feasible, low-level motion plans. An SMT-solver is now used
        to symbolically explore the space of all integrated plans that
        correspond to paths in the placement graph, and also satisfy the
        constraints demanded by the plan outline and the requirements. Our
        approach is implemented in a system called ROBOSYNTH. We have evaluated
        ROBOSYNTH on a generalization of an ITMP problem investigated in prior
        work. The experiments demonstrate that our method is capable of
        generating integrated plans for a number of interesting variations on
        the problem."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Nedunuri, Srinivas and Prabhu, Sailesh and Moll, Mark and Chaudhuri, Swarat and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA.2014.6906924"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">may</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"655-662"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"{SMT}-Based Synthesis of Integrated Task and Motion Plans from Plan
        Outlines"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2014"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absnedunuri2014smt-based-synthesis-of-integrated-task"><h3>Abstract</h3><p>We present a new approach to integrated task and motion planning (ITMP) for robots performing mobile manipulation. In our approach, the user writes a high-level specification that captures partial knowledge about a mobile manipulation setting. In particular, this specification includes a plan outline that syntactically defines a space of plausible integrated plans, a set of logical requirements that the generated plan must satisfy, and a description of the physical space that the robot manipulates. A synthesis algorithm is now used to search for an integrated plan that falls within the space defined by the plan outline, and also satisfies all requirements. Our synthesis algorithm complements continuous motion planning algorithms with calls to a Satisfiability Modulo Theories (SMT) solver. From the scene description, a motion planning algorithm is used to construct a placement graph, an abstraction of a manipulation graph whose paths represent feasible, low-level motion plans. An SMT-solver is now used to symbolically explore the space of all integrated plans that correspond to paths in the placement graph, and also satisfy the constraints demanded by the plan outline and the requirements. Our approach is implemented in a system called ROBOSYNTH. We have evaluated ROBOSYNTH on a generalization of an ITMP problem investigated in prior work. The experiments demonstrate that our method is capable of generating integrated plans for a number of interesting variations on the problem.</p></div></div></li><li> <span id="luna2014fast-stochastic-motion-planning">R. Luna, M. Lahijanian, <b>M. Moll</b>, and L. E. Kavraki, “Fast Stochastic Motion Planning with Optimality Guarantees using Local Policy Reconfiguration,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2014, pp. 3013–3019.</span> <br><div id="accluna2014fast-stochastic-motion-planning"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/luna2014fast-stochastic-motion-planning.pdf');" href="/publications/luna2014fast-stochastic-motion-planning.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA.2014.6907293');" href="https://dx.doi.org/10.1109/ICRA.2014.6907293"> publisher </a> <button class="btn" type="button" onclick="$('#bibluna2014fast-stochastic-motion-planning').slideToggle(600);$('#absluna2014fast-stochastic-motion-planning').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absluna2014fast-stochastic-motion-planning').slideToggle(600);$('#bibluna2014fast-stochastic-motion-planning').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibluna2014fast-stochastic-motion-planning"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">luna2014fast-stochastic-motion-planning</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"This work presents a framework for fast reconfiguration of local
        control policies for a stochastic system to satisfy a high-level task
        specification. The motion of the system is abstracted to a class of
        uncertain Markov models known as bounded-parameter Markov decision
        processes (BMDPs). During the abstraction, an efficient sampling-based
        method for stochastic optimal control is used to construct several
        policies within a discrete region of the state space in order for the
        system to transit between neighboring regions. A BMDP is then used to
        find an optimal strategy over the local policies by maximizing a
        continuous reward function; a new policy can be computed quickly if the
        reward function changes. The efficacy of the framework is demonstrated
        using a sequence of online tasks, showing that highly desirable
        policies can be obtained by reconfiguring existing local policies in
        just a few seconds."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Luna, Ryan and Lahijanian, Morteza and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA.2014.6907293"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"3013-3019"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Fast Stochastic Motion Planning with Optimality Guarantees using
        Local Policy Reconfiguration"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2014"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absluna2014fast-stochastic-motion-planning"><h3>Abstract</h3><p>This work presents a framework for fast reconfiguration of local control policies for a stochastic system to satisfy a high-level task specification. The motion of the system is abstracted to a class of uncertain Markov models known as bounded-parameter Markov decision processes (BMDPs). During the abstraction, an efficient sampling-based method for stochastic optimal control is used to construct several policies within a discrete region of the state space in order for the system to transit between neighboring regions. A BMDP is then used to find an optimal strategy over the local policies by maximizing a continuous reward function; a new policy can be computed quickly if the reward function changes. The efficacy of the framework is demonstrated using a sequence of online tasks, showing that highly desirable policies can be obtained by reconfiguring existing local policies in just a few seconds.</p></div></div></li><li> <span id="grady2013combining-a-pomdp-abstraction-with-replanning">D. K. Grady, <b>M. Moll</b>, and L. E. Kavraki, “Combining a POMDP Abstraction with Replanning to Solve Complex, Position-Dependent Sensing Tasks,” in <i>AAAI Fall Symposium</i>, Arlington, VA, 2013.</span> <br><div id="accgrady2013combining-a-pomdp-abstraction-with-replanning"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/grady2013combining-a-pomdp-abstraction-with-replanning.pdf');" href="/publications/grady2013combining-a-pomdp-abstraction-with-replanning.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://www.aaai.org/ocs/index.php/FSS/FSS13/paper/view/7578');" href="https://www.aaai.org/ocs/index.php/FSS/FSS13/paper/view/7578"> publisher </a> <button class="btn" type="button" onclick="$('#bibgrady2013combining-a-pomdp-abstraction-with-replanning').slideToggle(600);$('#absgrady2013combining-a-pomdp-abstraction-with-replanning').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absgrady2013combining-a-pomdp-abstraction-with-replanning').slideToggle(600);$('#bibgrady2013combining-a-pomdp-abstraction-with-replanning').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibgrady2013combining-a-pomdp-abstraction-with-replanning"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">grady2013combining-a-pomdp-abstraction-with-replanning</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The Partially-Observable Markov Decision Process (POMDP) is a
        general framework to determine reward-maximizing action policies under
        noisy action and sensing. However, determining an optimal policy for
        POMDPs is often intractable for robotic tasks due to the
        PSPACE-complete nature of the computation required. Several recent
        solvers have been introduced that expand the size of problems that can
        be considered. Although these POMDP solvers can respect complex motion
        constraints in theory, we show that the computational cost does not
        pro- vide a benefit in the eventual online execution, compared to our
        alternative approach that relies on a policy that ignores some of the
        motion constraints. We advocate using the POMDP framework where it is
        critical -- to find a policy that provides the optimal action given all
        past noisy sensor observations, while abstracting some of the motion
        constraints to reduce solution time. However, the actions of an
        abstract robot are generally not executable under its true motion
        constraints. The problem is addressed offline with a less-constrained
        POMDP, and navigation under the full system constraints is handled
        online with replanning. It is empirically demonstrated that the policy
        generated using this abstracted motion model is faster to compute and
        achieves similar or higher reward than addressing the motion
        constraints for a car-like robot as used in our experiments directly in
        the POMDP."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Arlington, VA"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Grady, Devin K. and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"AAAI Fall Symposium"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">nov</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Combining a {POMDP} Abstraction with Replanning to Solve Complex,
        Position-Dependent Sensing Tasks"</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">"https://www.aaai.org/ocs/index.php/FSS/FSS13/paper/view/7578"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2013"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absgrady2013combining-a-pomdp-abstraction-with-replanning"><h3>Abstract</h3><p>The Partially-Observable Markov Decision Process (POMDP) is a general framework to determine reward-maximizing action policies under noisy action and sensing. However, determining an optimal policy for POMDPs is often intractable for robotic tasks due to the PSPACE-complete nature of the computation required. Several recent solvers have been introduced that expand the size of problems that can be considered. Although these POMDP solvers can respect complex motion constraints in theory, we show that the computational cost does not pro- vide a benefit in the eventual online execution, compared to our alternative approach that relies on a policy that ignores some of the motion constraints. We advocate using the POMDP framework where it is critical – to find a policy that provides the optimal action given all past noisy sensor observations, while abstracting some of the motion constraints to reduce solution time. However, the actions of an abstract robot are generally not executable under its true motion constraints. The problem is addressed offline with a less-constrained POMDP, and navigation under the full system constraints is handled online with replanning. It is empirically demonstrated that the policy generated using this abstracted motion model is faster to compute and achieves similar or higher reward than addressing the motion constraints for a car-like robot as used in our experiments directly in the POMDP.</p></div></div></li><li> <span id="chyan2013improving-the-prediction-of-kinase-binding">J. Chyan, <b>M. Moll</b>, and L. E. Kavraki, “Improving the Prediction of Kinase Binding Affinity Using Homology Models,” in <i>Computational Structural Bioinformatics Workshop at the ACM Conf. on Bioinf., Comp. Bio. and Biomedical Informatics</i>, Washington, DC, 2013, pp. 741–748.</span> <br><div id="accchyan2013improving-the-prediction-of-kinase-binding"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/chyan2013improving-the-prediction-of-kinase-binding.pdf');" href="/publications/chyan2013improving-the-prediction-of-kinase-binding.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1145/2506583.2506704');" href="https://dx.doi.org/10.1145/2506583.2506704"> publisher </a> <button class="btn" type="button" onclick="$('#bibchyan2013improving-the-prediction-of-kinase-binding').slideToggle(600);$('#abschyan2013improving-the-prediction-of-kinase-binding').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abschyan2013improving-the-prediction-of-kinase-binding').slideToggle(600);$('#bibchyan2013improving-the-prediction-of-kinase-binding').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibchyan2013improving-the-prediction-of-kinase-binding"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">chyan2013improving-the-prediction-of-kinase-binding</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Kinases are a class of proteins very important to drug design;
        they play a pivotal role in many of the cell signaling pathways in the
        human body. Thus, many drug design studies involve finding inhibitors
        for kinases in the human kinome. However, identifying inhibitors of
        high selectivity is a difficult task. As a result, computational
        prediction methods have been developed to aid in this drug design
        problem.
        
        The recently published CCORPS method [3] is a semi-supervised learning
        method that identifies structural features in protein kinases that
        correlate with kinase binding affinity to inhibitors. However, CCORPS
        is dependent on the amount of available structural data. The amount of
        known structural data for proteins is extremely small compared to the
        amount of known protein sequences. To paint a clearer picture of how
        kinase structure relates to binding affinity, we propose extending the
        CCORPS method by integrating homology models for predicting kinase
        binding affinity. Our results show that using homology models
        significantly improves the prediction performance for some drugs while
        maintaining comparable performance for other drugs."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Washington, DC"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Chyan, Jeffrey and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Computational Structural Bioinformatics Workshop at the ACM
        Conf.\ on Bioinf., Comp.\ Bio.\ and Biomedical Informatics"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1145/2506583.2506704"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">sep</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"741-748"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Improving the Prediction of Kinase Binding Affinity Using Homology
        Models"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2013"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abschyan2013improving-the-prediction-of-kinase-binding"><h3>Abstract</h3><p>Kinases are a class of proteins very important to drug design; they play a pivotal role in many of the cell signaling pathways in the human body. Thus, many drug design studies involve finding inhibitors for kinases in the human kinome. However, identifying inhibitors of high selectivity is a difficult task. As a result, computational prediction methods have been developed to aid in this drug design problem. The recently published CCORPS method [3] is a semi-supervised learning method that identifies structural features in protein kinases that correlate with kinase binding affinity to inhibitors. However, CCORPS is dependent on the amount of available structural data. The amount of known structural data for proteins is extremely small compared to the amount of known protein sequences. To paint a clearer picture of how kinase structure relates to binding affinity, we propose extending the CCORPS method by integrating homology models for predicting kinase binding affinity. Our results show that using homology models significantly improves the prediction performance for some drugs while maintaining comparable performance for other drugs.</p></div></div></li><li> <span id="gipson2013resolution-independent-density-estimation">B. Gipson, <b>M. Moll</b>, and L. E. Kavraki, “Resolution Independent Density Estimation for Motion Planning in High-Dimensional Spaces,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2013, pp. 2429–2435.</span> <br><div id="accgipson2013resolution-independent-density-estimation"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/gipson2013resolution-independent-density-estimation.pdf');" href="/publications/gipson2013resolution-independent-density-estimation.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA.2013.6630908');" href="https://dx.doi.org/10.1109/ICRA.2013.6630908"> publisher </a> <button class="btn" type="button" onclick="$('#bibgipson2013resolution-independent-density-estimation').slideToggle(600);$('#absgipson2013resolution-independent-density-estimation').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absgipson2013resolution-independent-density-estimation').slideToggle(600);$('#bibgipson2013resolution-independent-density-estimation').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibgipson2013resolution-independent-density-estimation"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">gipson2013resolution-independent-density-estimation</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"This paper presents a new motion planner, Search Tree with
        Resolution Independent Density Estimation (STRIDE), designed for rapid
        exploration and path planning in high-dimensional systems (greater than
        10). A Geometric Near- neighbor Access Tree (GNAT) is maintained to
        estimate the sampling density of the configuration space, allowing an
        implicit, resolution-independent, Voronoi partitioning to provide
        sampling density estimates, naturally guiding the planner towards
        unexplored regions of the configuration space. This planner is capable
        of rapid exploration in the full dimension of the configuration space
        and, given that a GNAT requires only a valid distance metric, STRIDE is
        largely parameter-free. Extensive experimental results demonstrate
        significant dimension- dependent performance improvements over
        alternative state-of-the-art planners. In particular, high-dimensional
        systems where the free space is mostly defined by narrow passages were
        found to yield the greatest performance improvements. Experimental
        results are shown for both a classical 6-dimensional problem and those
        for which the dimension incrementally varies from 3 to 27."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Gipson, Bryant and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA.2013.6630908"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"2429--2435"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Resolution Independent Density Estimation for Motion Planning in
        High-Dimensional Spaces"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2013"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absgipson2013resolution-independent-density-estimation"><h3>Abstract</h3><p>This paper presents a new motion planner, Search Tree with Resolution Independent Density Estimation (STRIDE), designed for rapid exploration and path planning in high-dimensional systems (greater than 10). A Geometric Near- neighbor Access Tree (GNAT) is maintained to estimate the sampling density of the configuration space, allowing an implicit, resolution-independent, Voronoi partitioning to provide sampling density estimates, naturally guiding the planner towards unexplored regions of the configuration space. This planner is capable of rapid exploration in the full dimension of the configuration space and, given that a GNAT requires only a valid distance metric, STRIDE is largely parameter-free. Extensive experimental results demonstrate significant dimension- dependent performance improvements over alternative state-of-the-art planners. In particular, high-dimensional systems where the free space is mostly defined by narrow passages were found to yield the greatest performance improvements. Experimental results are shown for both a classical 6-dimensional problem and those for which the dimension incrementally varies from 3 to 27.</p></div></div></li><li> <span id="grady2013automated-model-approximation">D. K. Grady, <b>M. Moll</b>, and L. E. Kavraki, “Automated Model Approximation for Robotic Navigation with POMDPs,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2013, pp. 78–84.</span> <br><div id="accgrady2013automated-model-approximation"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/grady2013automated-model-approximation.pdf');" href="/publications/grady2013automated-model-approximation.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA.2013.6630559');" href="https://dx.doi.org/10.1109/ICRA.2013.6630559"> publisher </a> <button class="btn" type="button" onclick="$('#bibgrady2013automated-model-approximation').slideToggle(600);$('#absgrady2013automated-model-approximation').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absgrady2013automated-model-approximation').slideToggle(600);$('#bibgrady2013automated-model-approximation').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibgrady2013automated-model-approximation"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">grady2013automated-model-approximation</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Partially-Observable Markov Decision Processes (POMDPs) are a
        problem class with significant applicability to robotics when
        considering the uncertainty present in the real world, however, they
        quickly become intractable for large state and action spaces. A method
        to create a less complex but accurate action model approximation is
        proposed and evaluated using a state-of-the-art POMDP solver. We apply
        this general and powerful formulation to a robotic navigation task
        under state and sensing uncertainty. Results show that this method can
        provide a useful action model that yields a policy with similar overall
        expected reward compared to the true action model, often with
        significant computational savings. In some cases, our reduced
        complexity model can solve problems where the true model is too complex
        to find a policy that accomplishes the task. We conclude that this
        technique of building problem-dependent approximations can provide
        significant computational advantages and can help expand the complexity
        of problems that can be considered using current POMDP techniques."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Grady, Devin K. and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA.2013.6630559"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"78--84"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Automated Model Approximation for Robotic Navigation with {POMDP}s"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2013"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absgrady2013automated-model-approximation"><h3>Abstract</h3><p>Partially-Observable Markov Decision Processes (POMDPs) are a problem class with significant applicability to robotics when considering the uncertainty present in the real world, however, they quickly become intractable for large state and action spaces. A method to create a less complex but accurate action model approximation is proposed and evaluated using a state-of-the-art POMDP solver. We apply this general and powerful formulation to a robotic navigation task under state and sensing uncertainty. Results show that this method can provide a useful action model that yields a policy with similar overall expected reward compared to the true action model, often with significant computational savings. In some cases, our reduced complexity model can solve problems where the true model is too complex to find a policy that accomplishes the task. We conclude that this technique of building problem-dependent approximations can provide significant computational advantages and can help expand the complexity of problems that can be considered using current POMDP techniques.</p></div></div></li><li> <span id="luna2013anytime-solution-optimization">R. Luna, I. A. Şucan, <b>M. Moll</b>, and L. E. Kavraki, “Anytime Solution Optimization for Sampling-Based Motion Planning,” in <i>IEEE Intl. Conf. on Robotics and Automation</i>, 2013, pp. 5053–5059.</span> <br><div id="accluna2013anytime-solution-optimization"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/luna2013anytime-solution-optimization.pdf');" href="/publications/luna2013anytime-solution-optimization.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ICRA.2013.6631301');" href="https://dx.doi.org/10.1109/ICRA.2013.6631301"> publisher </a> <button class="btn" type="button" onclick="$('#bibluna2013anytime-solution-optimization').slideToggle(600);$('#absluna2013anytime-solution-optimization').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absluna2013anytime-solution-optimization').slideToggle(600);$('#bibluna2013anytime-solution-optimization').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibluna2013anytime-solution-optimization"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">luna2013anytime-solution-optimization</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Recent work in sampling-based motion planning has yielded several
        different approaches for computing good quality paths in high degree of
        freedom systems: path shortcutting methods that attempt to shorten a
        single solution path by connecting non-consecutive configurations, a
        path hybridization technique that combines portions of two or more
        solutions to form a shorter path, and asymptotically optimal algorithms
        that converge to the shortest path over time. This paper presents an
        extensible meta-algorithm that incorporates a traditional
        sampling-based planning algorithm with offline path shorten- ing
        techniques to form an anytime algorithm which exhibits competitive
        solution lengths to the best known methods and optimizers. A series of
        experiments involving rigid motion and complex manipulation are
        performed as well as a comparison with asymptotically optimal methods
        which show the efficacy of the proposed scheme, particularly in
        high-dimensional spaces."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Luna, Ryan and {\c{S}}ucan, Ioan A. and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ICRA.2013.6631301"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"5053--5059"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Anytime Solution Optimization for Sampling-Based Motion Planning"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2013"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absluna2013anytime-solution-optimization"><h3>Abstract</h3><p>Recent work in sampling-based motion planning has yielded several different approaches for computing good quality paths in high degree of freedom systems: path shortcutting methods that attempt to shorten a single solution path by connecting non-consecutive configurations, a path hybridization technique that combines portions of two or more solutions to form a shorter path, and asymptotically optimal algorithms that converge to the shortest path over time. This paper presents an extensible meta-algorithm that incorporates a traditional sampling-based planning algorithm with offline path shorten- ing techniques to form an anytime algorithm which exhibits competitive solution lengths to the best known methods and optimizers. A series of experiments involving rigid motion and complex manipulation are performed as well as a comparison with asymptotically optimal methods which show the efficacy of the proposed scheme, particularly in high-dimensional spaces.</p></div></div></li><li> <span id="grady2012multi-objective-sensor-based-replanning-for-a-car-like">D. K. Grady, <b>M. Moll</b>, C. Hegde, A. C. Sankaranarayanan, R. G. Baraniuk, and L. E. Kavraki, “Multi-Objective Sensor-Based Replanning for a Car-Like Robot,” in <i>IEEE Intl. Symp. on Safety, Security, and Rescue Robotics</i>, 2012.</span> <br><div id="accgrady2012multi-objective-sensor-based-replanning-for-a-car-like"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/grady2012multi-objective-sensor-based-replanning-for-a-car-like.pdf');" href="/publications/grady2012multi-objective-sensor-based-replanning-for-a-car-like.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/SSRR.2012.6523898');" href="https://dx.doi.org/10.1109/SSRR.2012.6523898"> publisher </a> <button class="btn" type="button" onclick="$('#bibgrady2012multi-objective-sensor-based-replanning-for-a-car-like').slideToggle(600);$('#absgrady2012multi-objective-sensor-based-replanning-for-a-car-like').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absgrady2012multi-objective-sensor-based-replanning-for-a-car-like').slideToggle(600);$('#bibgrady2012multi-objective-sensor-based-replanning-for-a-car-like').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibgrady2012multi-objective-sensor-based-replanning-for-a-car-like"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">grady2012multi-objective-sensor-based-replanning-for-a-car-like</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">" In search and rescue applications it is important that mobile
        ground robots can verify whether a potential target/victim is indeed a
        target of interest. This paper describes a novel approach to
        multi-robot target verification of multiple static objects. Suppose a
        team of multiple mobile ground robots are operating in a partially
        known environment with knowledge of possible target locations and
        obstacles. The ground robots' goal is to (a) collectively classify the
        targets (or build models of them) by identifying good viewpoints to
        sense the targets, while (b) coordinating their actions to execute the
        mission and always be safe by avoiding obstacles and each other. As
        opposed to a traditional next-best-view (NBV) algorithm that generates
        a single good view, we characterize the informativeness of all
        potential views. We propose a measure for the informativeness of a view
        that exploits the geometric structure of the pose manifold. This
        information is encoded in a cost map that guides a multi-robot motion
        planning algorithm towards views that are both reachable and
        informative. Finally, we account for differential constraints in the
        robots' motion that prevent unrealistic scenarios such as the robots
        stopping or turning instantaneously. A range of simulations indicates
        that our approach outperforms current approaches and demonstrates the
        advantages of predictive sensing and accounting for reachability
        constraints."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Grady, Devin K. and Moll, Mark and Hegde, Chinmay and Sankaranarayanan, Aswin C. and Baraniuk, Richard G. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"IEEE Intl.\ Symp.\ on Safety, Security, and Rescue Robotics"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/SSRR.2012.6523898"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Multi-Objective Sensor-Based Replanning for a Car-Like Robot"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2012"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absgrady2012multi-objective-sensor-based-replanning-for-a-car-like"><h3>Abstract</h3><p> In search and rescue applications it is important that mobile ground robots can verify whether a potential target/victim is indeed a target of interest. This paper describes a novel approach to multi-robot target verification of multiple static objects. Suppose a team of multiple mobile ground robots are operating in a partially known environment with knowledge of possible target locations and obstacles. The ground robots’ goal is to (a) collectively classify the targets (or build models of them) by identifying good viewpoints to sense the targets, while (b) coordinating their actions to execute the mission and always be safe by avoiding obstacles and each other. As opposed to a traditional next-best-view (NBV) algorithm that generates a single good view, we characterize the informativeness of all potential views. We propose a measure for the informativeness of a view that exploits the geometric structure of the pose manifold. This information is encoded in a cost map that guides a multi-robot motion planning algorithm towards views that are both reachable and informative. Finally, we account for differential constraints in the robots’ motion that prevent unrealistic scenarios such as the robots stopping or turning instantaneously. A range of simulations indicates that our approach outperforms current approaches and demonstrates the advantages of predictive sensing and accounting for reachability constraints.</p></div></div></li><li> <span id="grady2012multi-robot-target-verification">D. K. Grady, <b>M. Moll</b>, C. Hegde, A. C. Sankaranarayanan, R. G. Baraniuk, and L. E. Kavraki, “Multi-Robot Target Verification with Reachability Constraints,” in <i>IEEE Intl. Symp. on Safety, Security, and Rescue Robotics</i>, 2012.</span> <br><div id="accgrady2012multi-robot-target-verification"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/grady2012multi-robot-target-verification.pdf');" href="/publications/grady2012multi-robot-target-verification.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/SSRR.2012.6523873');" href="https://dx.doi.org/10.1109/SSRR.2012.6523873"> publisher </a> <button class="btn" type="button" onclick="$('#bibgrady2012multi-robot-target-verification').slideToggle(600);$('#absgrady2012multi-robot-target-verification').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absgrady2012multi-robot-target-verification').slideToggle(600);$('#bibgrady2012multi-robot-target-verification').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibgrady2012multi-robot-target-verification"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">grady2012multi-robot-target-verification</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"This paper studies a core problem in multi-objective mission
        planning for robots governed by differential constraints. The problem
        considered is the following. A car-like robot computes a plan to move
        from a start configuration to a goal region. The robot is equipped with
        a sensor that can alert it if an anomaly appears within some range
        while the robot is moving. In that case, the robot tries to deviate
        from its computed path and gather more information about the target
        without incurring considerable delays in fulfilling its primary
        mission, which is to move to its final destination. This problem is
        important in, e.g., surveillance, where inspection of possible threats
        needs to be balanced with completing a nominal route. The paper
        presents a simple and intuitive framework to study the trade-offs
        present in the above problem. Our work utilizes a state-of-the-art
        sampling-based planner, which employs both a high-level discrete guide
        and low-level planning. We show that modifications to the distance
        function used by the planner and to the weights that the planner
        employs to compute the high-level guide can help the robot react online
        to new secondary objectives that were unknown at the outset of the
        mission. The modifications are computed using information obtained from
        a conventional camera model. We find that for small percentage
        increases in path length, the robot can achieve significant gains in
        information about an unexpected target."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Grady, Devin K. and Moll, Mark and Hegde, Chinmay and Sankaranarayanan, Aswin C. and Baraniuk, Richard G. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"IEEE Intl.\ Symp.\ on Safety, Security, and Rescue Robotics"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/SSRR.2012.6523873"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Multi-Robot Target Verification with Reachability Constraints"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2012"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absgrady2012multi-robot-target-verification"><h3>Abstract</h3><p>This paper studies a core problem in multi-objective mission planning for robots governed by differential constraints. The problem considered is the following. A car-like robot computes a plan to move from a start configuration to a goal region. The robot is equipped with a sensor that can alert it if an anomaly appears within some range while the robot is moving. In that case, the robot tries to deviate from its computed path and gather more information about the target without incurring considerable delays in fulfilling its primary mission, which is to move to its final destination. This problem is important in, e.g., surveillance, where inspection of possible threats needs to be balanced with completing a nominal route. The paper presents a simple and intuitive framework to study the trade-offs present in the above problem. Our work utilizes a state-of-the-art sampling-based planner, which employs both a high-level discrete guide and low-level planning. We show that modifications to the distance function used by the planner and to the weights that the planner employs to compute the high-level guide can help the robot react online to new secondary objectives that were unknown at the outset of the mission. The modifications are computed using information obtained from a conventional camera model. We find that for small percentage increases in path length, the robot can achieve significant gains in information about an unexpected target.</p></div></div></li><li> <span id="moll2011teaching-motion-planning-concepts"><b>M. Moll</b>, I. A. Şucan, J. Bordeaux, and L. E. Kavraki, “Teaching Motion Planning Concepts to Undergraduate Students,” in <i>IEEE Workshop on Advanced Robotics and its Social Impacts (ARSO)</i>, 2011.</span> <br><div id="accmoll2011teaching-motion-planning-concepts"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2011teaching-motion-planning-concepts.pdf');" href="/publications/moll2011teaching-motion-planning-concepts.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ARSO.2011.6301976');" href="https://dx.doi.org/10.1109/ARSO.2011.6301976"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2011teaching-motion-planning-concepts').slideToggle(600);$('#absmoll2011teaching-motion-planning-concepts').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2011teaching-motion-planning-concepts').slideToggle(600);$('#bibmoll2011teaching-motion-planning-concepts').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2011teaching-motion-planning-concepts"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll2011teaching-motion-planning-concepts</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Motion planning is a central problem in robotics. Although it is
        an engaging topic for undergraduate students, it is difficult to teach,
        and as a result, the material is often only covered at an abstract
        level. Deep learning could be achieved by having students implement and
        test different algorithms. However, there is usually no time within a
        single class to have students completely implement several motion
        planning algorithms as they require the development of many lower-level
        data structures. We present an ongoing project to develop a teaching
        module for robotic motion planning centered around an integrated
        software environment. The module can be taught early in the
        undergraduate curriculum, after students have taken an introductory
        programming class."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and {\c{S}}ucan, Ioan A. and Bordeaux, Janice and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"IEEE Workshop on Advanced Robotics and its Social Impacts
        (ARSO)"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ARSO.2011.6301976"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Teaching Motion Planning Concepts to Undergraduate Students"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2011"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2011teaching-motion-planning-concepts"><h3>Abstract</h3><p>Motion planning is a central problem in robotics. Although it is an engaging topic for undergraduate students, it is difficult to teach, and as a result, the material is often only covered at an abstract level. Deep learning could be achieved by having students implement and test different algorithms. However, there is usually no time within a single class to have students completely implement several motion planning algorithms as they require the development of many lower-level data structures. We present an ongoing project to develop a teaching module for robotic motion planning centered around an integrated software environment. The module can be taught early in the undergraduate curriculum, after students have taken an introductory programming class.</p></div></div></li><li> <span id="grady2011look-before-you-leap">D. K. Grady, <b>M. Moll</b>, C. Hegde, A. C. Sankaranarayanan, R. G. Baraniuk, and L. E. Kavraki, “Look Before You Leap: Predictive Sensing and Opportunistic Navigation,” in <i>Workshop on Progress and Open Problems in Motion Planning at the IEEE/RSJ Conf. on Intelligent Robots and Systems</i>, 2011.</span> <br><div id="accgrady2011look-before-you-leap"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/grady2011look-before-you-leap.pdf');" href="/publications/grady2011look-before-you-leap.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibgrady2011look-before-you-leap').slideToggle(600);$('#absgrady2011look-before-you-leap').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absgrady2011look-before-you-leap').slideToggle(600);$('#bibgrady2011look-before-you-leap').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibgrady2011look-before-you-leap"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">grady2011look-before-you-leap</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"This paper describes a novel method for identifying multiple
        targets with multiple robots in a partially known environment. Two main
        issues are addressed. The first relates to the use of motion planning
        algorithms to determine whether robots can reach ``good'' positions
        that offer the most informative measurements. The second concerns the
        use of predictive sensing to decide where sensor measurements should be
        taken. The problem is formulated similar to a next-best-view problem
        with differential constraints on the robots' motion, with additional
        layers of complexity due to visual occlusions as well as navigational
        obstacles. We propose a new distributed sensing strategy that exploits
        the structure of image manifolds to predict the utility of the
        measurements at a given position. This information is encoded in a cost
        map that guides a motion planning algorithm. Coordination among robots
        is achieved by incorporating additional information in each robot's
        cost map. A range of simulations indicates that our approach
        outperforms current approaches and demonstrates the advantages of
        predictive sensing and accounting for reachability constraints."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Grady, Devin K. and Moll, Mark and Hegde, Chinmay and Sankaranarayanan, Aswin C. and Baraniuk, Richard G. and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Workshop on Progress and Open Problems in Motion Planning at the
        {IEEE}/{RSJ} Conf.\ on Intelligent Robots and Systems"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Look Before You Leap: Predictive Sensing and Opportunistic
        Navigation"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2011"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absgrady2011look-before-you-leap"><h3>Abstract</h3><p>This paper describes a novel method for identifying multiple targets with multiple robots in a partially known environment. Two main issues are addressed. The first relates to the use of motion planning algorithms to determine whether robots can reach “good” positions that offer the most informative measurements. The second concerns the use of predictive sensing to decide where sensor measurements should be taken. The problem is formulated similar to a next-best-view problem with differential constraints on the robots’ motion, with additional layers of complexity due to visual occlusions as well as navigational obstacles. We propose a new distributed sensing strategy that exploits the structure of image manifolds to predict the utility of the measurements at a given position. This information is encoded in a cost map that guides a motion planning algorithm. Coordination among robots is achieved by incorporating additional information in each robot’s cost map. A range of simulations indicates that our approach outperforms current approaches and demonstrates the advantages of predictive sensing and accounting for reachability constraints.</p></div></div></li><li> <span id="haspel2009tracing-conformational-changes-in-proteins">N. Haspel, <b>M. Moll</b>, M. L. Baker, W. Chiu, and L. E. Kavraki, “Tracing Conformational Changes in Proteins,” in <i>IEEE Intl. Conf. on Bioinformatics and Biomedicine Workshops (BIBMW)</i>, Washington, DC, 2009, pp. 120–127.</span> <br><div id="acchaspel2009tracing-conformational-changes-in-proteins"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/haspel2009tracing-conformational-changes-in-proteins.pdf');" href="/publications/haspel2009tracing-conformational-changes-in-proteins.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/BIBMW.2009.5332115');" href="https://dx.doi.org/10.1109/BIBMW.2009.5332115"> publisher </a> <button class="btn" type="button" onclick="$('#bibhaspel2009tracing-conformational-changes-in-proteins').slideToggle(600);$('#abshaspel2009tracing-conformational-changes-in-proteins').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abshaspel2009tracing-conformational-changes-in-proteins').slideToggle(600);$('#bibhaspel2009tracing-conformational-changes-in-proteins').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibhaspel2009tracing-conformational-changes-in-proteins"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">haspel2009tracing-conformational-changes-in-proteins</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Many proteins undergo extensive conformational changes as part of
        their functionality. Tracing these changes is important for
        understanding the way these proteins function. Traditional
        biophysics-based conformational search methods require a large number
        of calculations and are hard to apply to large-scale conformational
        motions. In this work we investigate the application of a
        robotics-inspired method, using backbone and limited side chain
        representation and a coarse grained energy function to trace
        large-scale conformational motions. We tested the algorithm on three
        well known medium to large proteins and we show that even with
        relatively little information we are able to trace low-energy
        conformational pathways efficiently. The conformational pathways
        produced by our methods can be further filtered and refined to produce
        more useful information on the way proteins function under
        physiological conditions."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Washington, DC"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Haspel, Nurit and Moll, Mark and Baker, Matthew L. and Chiu, Wah and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Bioinformatics and Biomedicine Workshops
        (BIBMW)"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/BIBMW.2009.5332115"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">nov</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"120--127"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Tracing Conformational Changes in Proteins"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2009"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abshaspel2009tracing-conformational-changes-in-proteins"><h3>Abstract</h3><p>Many proteins undergo extensive conformational changes as part of their functionality. Tracing these changes is important for understanding the way these proteins function. Traditional biophysics-based conformational search methods require a large number of calculations and are hard to apply to large-scale conformational motions. In this work we investigate the application of a robotics-inspired method, using backbone and limited side chain representation and a coarse grained energy function to trace large-scale conformational motions. We tested the algorithm on three well known medium to large proteins and we show that even with relatively little information we are able to trace low-energy conformational pathways efficiently. The conformational pathways produced by our methods can be further filtered and refined to produce more useful information on the way proteins function under physiological conditions.</p></div></div></li><li> <span id="moll2008labelhash-a-flexible-and-extensible-method"><b>M. Moll</b> and L. E. Kavraki, “LabelHash: A Flexible and Extensible Method for Matching Structural Motifs,” in <i>Automated Function Prediction / BioSapiens Meeting (AFP-BioSapiens)</i>, Toronto, Canada, 2008.</span> Available from Nature Precedings <br><div id="accmoll2008labelhash-a-flexible-and-extensible-method"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2008labelhash-a-flexible-and-extensible-method.pdf');" href="/publications/moll2008labelhash-a-flexible-and-extensible-method.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1038/npre.2008.2199.1');" href="https://dx.doi.org/10.1038/npre.2008.2199.1"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2008labelhash-a-flexible-and-extensible-method').slideToggle(600);$('#absmoll2008labelhash-a-flexible-and-extensible-method').slideUp(600);"> bibtex </button> <button class="btn disabled" type="button">abstract</button></div><div class="card collapse" id="bibmoll2008labelhash-a-flexible-and-extensible-method"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll2008labelhash-a-flexible-and-extensible-method</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Toronto, Canada"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Automated Function Prediction / BioSapiens Meeting
        (AFP-BioSapiens)"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1038/npre.2008.2199.1"</span><span class="p">,</span>
  <span class="na">note</span> <span class="p">=</span> <span class="s">"Available from Nature Precedings"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"{LabelHash}: A Flexible and Extensible Method for Matching
        Structural Motifs"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2008"</span>
<span class="p">}</span></code></pre></figure></div></div></li><li> <span id="moll2008matching-of-structural-motifs"><b>M. Moll</b> and L. E. Kavraki, “Matching of Structural Motifs Using Hashing on Residue Labels and Geometric Filtering for Protein Function Prediction,” in <i>The Seventh Annual International Conference on Computational Systems Bioinformatics (CSB2008)</i>, 2008, pp. 157–168.</span> <br><div id="accmoll2008matching-of-structural-motifs"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2008matching-of-structural-motifs.pdf');" href="/publications/moll2008matching-of-structural-motifs.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1142/9781848162648_0014');" href="https://dx.doi.org/10.1142/9781848162648_0014"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2008matching-of-structural-motifs').slideToggle(600);$('#absmoll2008matching-of-structural-motifs').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2008matching-of-structural-motifs').slideToggle(600);$('#bibmoll2008matching-of-structural-motifs').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2008matching-of-structural-motifs"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll2008matching-of-structural-motifs</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"There is an increasing number of proteins with known structure
        but unknown function. Determining their function would have a
        significant impact on understanding diseases and designing new
        therapeutics. However, experimental protein function determination is
        expensive and very time-consuming. Computational methods can facilitate
        function determination by identifying proteins that have high
        structural and chemical similarity. Our focus is on methods that
        determine binding site similarity. Although several such methods exist,
        it still remains a challenging problem to quickly find all
        functionally-related matches for structural motifs in large data sets
        with high specificity. In this context, a structural motif is a set of
        3D points annotated with physicochemical information that characterize
        a molecular function. We propose a new method called LabelHash that
        creates hash tables of $n$-tuples of residues for a set of targets.
        Using these hash tables, we can quickly look up partial matches to a
        motif and expand those matches to complete matches. We show that by
        applying only very mild geometric constraints we can find statistically
        significant matches with extremely high specificity in very large data
        sets and for very general structural motifs. We demonstrate that our
        method requires a reasonable amount of storage when employing a simple
        geometric filter and further improves on the specificity of our
        previous work while maintaining very high sensitivity. Our algorithm is
        evaluated on 20 homolog classes and a non-redundant version of the
        Protein Data Bank as our background data set. We use cluster analysis
        to analyze why certain classes of homologs are more difficult to
        classify than others. The LabelHash algorithm is implemented on a web
        server at http://kavrakilab.org/labelhash/."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"The Seventh Annual International Conference on Computational
        Systems Bioinformatics (CSB2008)"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1142/9781848162648_0014"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"157-168"</span><span class="p">,</span>
  <span class="na">pmid</span> <span class="p">=</span> <span class="s">"19642277"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Matching of Structural Motifs Using Hashing on Residue Labels and
        Geometric Filtering for Protein Function Prediction"</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">"http://csb2008.org/csb2008papers/077Moll.pdf"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2008"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2008matching-of-structural-motifs"><h3>Abstract</h3><p>There is an increasing number of proteins with known structure but unknown function. Determining their function would have a significant impact on understanding diseases and designing new therapeutics. However, experimental protein function determination is expensive and very time-consuming. Computational methods can facilitate function determination by identifying proteins that have high structural and chemical similarity. Our focus is on methods that determine binding site similarity. Although several such methods exist, it still remains a challenging problem to quickly find all functionally-related matches for structural motifs in large data sets with high specificity. In this context, a structural motif is a set of 3D points annotated with physicochemical information that characterize a molecular function. We propose a new method called LabelHash that creates hash tables of n-tuples of residues for a set of targets. Using these hash tables, we can quickly look up partial matches to a motif and expand those matches to complete matches. We show that by applying only very mild geometric constraints we can find statistically significant matches with extremely high specificity in very large data sets and for very general structural motifs. We demonstrate that our method requires a reasonable amount of storage when employing a simple geometric filter and further improves on the specificity of our previous work while maintaining very high sensitivity. Our algorithm is evaluated on 20 homolog classes and a non-redundant version of the Protein Data Bank as our background data set. We use cluster analysis to analyze why certain classes of homologs are more difficult to classify than others. The LabelHash algorithm is implemented on a web server at http://kavrakilab.org/labelhash/.</p></div></div></li><li> <span id="fofanov2008a-statistical-model-to-correct-systematic">V. Y. Fofanov, B. Y. Chen, D. H. Bryant, <b>M. Moll</b>, O. Lichtarge, L. E. Kavraki, and M. Kimmel, “A statistical model to correct systematic bias introduced by algorithmic thresholds in protein structural comparison algorithms,” in <i>IEEE Intl. Conf. on Bioinformatics and Biomedicine Workshops (BIBMW)</i>, 2008, pp. 1–8.</span> <br><div id="accfofanov2008a-statistical-model-to-correct-systematic"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/fofanov2008a-statistical-model-to-correct-systematic.pdf');" href="/publications/fofanov2008a-statistical-model-to-correct-systematic.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/BIBMW.2008.4686202');" href="https://dx.doi.org/10.1109/BIBMW.2008.4686202"> publisher </a> <button class="btn" type="button" onclick="$('#bibfofanov2008a-statistical-model-to-correct-systematic').slideToggle(600);$('#absfofanov2008a-statistical-model-to-correct-systematic').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absfofanov2008a-statistical-model-to-correct-systematic').slideToggle(600);$('#bibfofanov2008a-statistical-model-to-correct-systematic').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibfofanov2008a-statistical-model-to-correct-systematic"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">fofanov2008a-statistical-model-to-correct-systematic</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The identification of protein function is crucial to
        understanding cellular processes and selecting novel proteins as drug
        targets. However, experimental methods for determining protein function
        can be expensive and time-consuming. Protein partial structure
        comparison methods seek to guide and accelerate the process of function
        determination by matching characterized functional site
        representations, motifs, to substructures within uncharacterized
        proteins, matches. One common difficulty of all protein structural
        comparison techniques is the computational cost of obtaining a match.
        In an effort to maintain practical efficiency, some algorithms employ
        efficient geometric threshold-based searches to eliminate biologically
        irrelevant matches. Thresholds refine and accelerate the method by
        limiting the number of potential matches that need to be considered.
        However, because statistical models rely on the output of the geometric
        matching method to accurately measure statistical significance,
        geometric thresholds can also artificially distort the basis of
        statistical models, making statistical scores dependant on geometric
        thresholds and potentially causing significant reductions in accuracy
        of the functional annotation method. This paper proposes a point-weight
        based correction approach to quantify and model the dependence of
        statistical scores to account for the systematic bias introduced by
        heuristics. Using a benchmark dataset of 20 structural motifs, we show
        that the point-weight correction procedure accurately models the
        information lost during the geometric comparison phase, removing
        systematic bias and greatly reducing misclassification rates of
        functionally related proteins, while maintaining specificity. "</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Fofanov, Viacheslav Y. and Chen, Brian Y. and Bryant, Drew H. and Moll, Mark and Lichtarge, Olivier and Kavraki, Lydia E. and Kimmel, Marek"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"{IEEE} Intl.\ Conf.\ on Bioinformatics and Biomedicine Workshops
        (BIBMW)"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/BIBMW.2008.4686202"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"1--8"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"A statistical model to correct systematic bias introduced by
        algorithmic thresholds in protein structural comparison algorithms"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2008"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absfofanov2008a-statistical-model-to-correct-systematic"><h3>Abstract</h3><p>The identification of protein function is crucial to understanding cellular processes and selecting novel proteins as drug targets. However, experimental methods for determining protein function can be expensive and time-consuming. Protein partial structure comparison methods seek to guide and accelerate the process of function determination by matching characterized functional site representations, motifs, to substructures within uncharacterized proteins, matches. One common difficulty of all protein structural comparison techniques is the computational cost of obtaining a match. In an effort to maintain practical efficiency, some algorithms employ efficient geometric threshold-based searches to eliminate biologically irrelevant matches. Thresholds refine and accelerate the method by limiting the number of potential matches that need to be considered. However, because statistical models rely on the output of the geometric matching method to accurately measure statistical significance, geometric thresholds can also artificially distort the basis of statistical models, making statistical scores dependant on geometric thresholds and potentially causing significant reductions in accuracy of the functional annotation method. This paper proposes a point-weight based correction approach to quantify and model the dependence of statistical scores to account for the systematic bias introduced by heuristics. Using a benchmark dataset of 20 structural motifs, we show that the point-weight correction procedure accurately models the information lost during the geometric comparison phase, removing systematic bias and greatly reducing misclassification rates of functionally related proteins, while maintaining specificity.</p></div></div></li><li> <span id="chen2007representations-of-structural-motifs-for-protein">B. Y. Chen, D. H. Bryant, J. H. Bylund, A. E. Cruess, D. M. Kristensen, V. Y. Fofanov, <b>M. Moll</b>, M. Kimmel, O. Lichtarge, and L. E. Kavraki, “Representations of Structural Motifs for Protein Function Prediction,” in <i>15th Annual Intl. Conf. on Intelligent Systems for Molecular Biology (ISMB) &amp; 6th European Conf. on Comp. Bio. (ECCB)</i>, Vienna, Austria, 2007.</span> <br><div id="accchen2007representations-of-structural-motifs-for-protein"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/chen2007representations-of-structural-motifs-for-protein.pdf');" href="/publications/chen2007representations-of-structural-motifs-for-protein.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibchen2007representations-of-structural-motifs-for-protein').slideToggle(600);$('#abschen2007representations-of-structural-motifs-for-protein').slideUp(600);"> bibtex </button> <button class="btn disabled" type="button">abstract</button></div><div class="card collapse" id="bibchen2007representations-of-structural-motifs-for-protein"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">chen2007representations-of-structural-motifs-for-protein</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Vienna, Austria"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Chen, Brian Y. and Bryant, Drew H. and Bylund, Joseph H. and Cruess, Amanda E. and Kristensen, David M. and Fofanov, Viacheslav Y. and Moll, Mark and Kimmel, Marek and Lichtarge, Olivier and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"15th Annual Intl.\ Conf.\ on Intelligent Systems for Molecular
        Biology (ISMB) \&amp; 6th European Conf.\ on Comp.\ Bio.\ (ECCB)"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Representations of Structural Motifs for Protein Function
        Prediction"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2007"</span>
<span class="p">}</span></code></pre></figure></div></div></li><li> <span id="shen2007multifunctional-and-reconfigurable-superbot-modules">W.-M. Shen, B. Salemi, <b>M. Moll</b>, C. H. Chiu, J. Everist, F. Hou, N. Ranasinghe, and M. Rubenstein, “Multifunctional Behaviors of Reconfigurable SuperBot Modules,” in <i>Proc. 2007 IEEE/RSJ Intl. Conf. on Intelligent Robots and Systems</i>, 2007.</span> <a href="http://www.isi.edu/robots/superbot/movies/SuperBot.mov">Video</a> <br><div id="accshen2007multifunctional-and-reconfigurable-superbot-modules"><div> <button class="btn disabled">pdf</button> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/IROS.2007.4399001');" href="https://dx.doi.org/10.1109/IROS.2007.4399001"> publisher </a> <button class="btn" type="button" onclick="$('#bibshen2007multifunctional-and-reconfigurable-superbot-modules').slideToggle(600);$('#absshen2007multifunctional-and-reconfigurable-superbot-modules').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absshen2007multifunctional-and-reconfigurable-superbot-modules').slideToggle(600);$('#bibshen2007multifunctional-and-reconfigurable-superbot-modules').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibshen2007multifunctional-and-reconfigurable-superbot-modules"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">shen2007multifunctional-and-reconfigurable-superbot-modules</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Superbot consists of Lego-like but autonomous robotic modules
        that can reconfigure into different systems for different tasks.
        Examples of configurable systems include rolling tracks or wheels (for
        efficient travel), spiders or centipedes (for climbing), snakes (for
        burrowing in ground), and climbers (for inspection and repair in
        space). This video shows several configurations and behaviors that are
        new for modular and reconfigurable robots. Each SuperBot module is a
        complete robotic system and has a power supply, micro- controllers,
        sensors, communication, three degrees of freedom, and six connecting
        faces (front, back, left, right, up and down) to dynamically connect to
        other modules. This design allows flexible bending, docking, and
        continuous rotation. A single module can move forward, back, left,
        right, flip-over, and rotate as a wheel. Modules can communication with
        each other for totally distributed control and can support arbitrary
        module reshuffling during their operation. The modules have both
        internal and external sensors for monitoring self-status and
        environmental parameters. They can form arbitrary configurations
        (graphs) and can control these configurations for different
        functionality such as locomotion, manipulation, and self-repair. This
        video shows the latest status the SuperBot modules and all these
        behaviors were made in just one week. The fact that SuperBot can
        achieve so much in so short a time demonstrates the unique value of
        modular, multifunctional and self-reconfigurable robots."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Shen, Wei-Min and Salemi, Behnam and Moll, Mark and Chiu, Chi Ho and Everist, Jacob and Hou, Feili and Ranasinghe, Nadeesha and Rubenstein, Michael"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 2007 {IEEE/RSJ} Intl.\ Conf.\ on Intelligent Robots and
        Systems"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/IROS.2007.4399001"</span><span class="p">,</span>
  <span class="na">note</span> <span class="p">=</span> <span class="s">"\href{http://www.isi.edu/robots/superbot/movies/SuperBot.mov}{Video}"</span><span class="p">,</span>
  <span class="na">pubtype</span> <span class="p">=</span> <span class="s">"Other"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Multifunctional Behaviors of Reconfigurable {SuperBot} Modules"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2007"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absshen2007multifunctional-and-reconfigurable-superbot-modules"><h3>Abstract</h3><p>Superbot consists of Lego-like but autonomous robotic modules that can reconfigure into different systems for different tasks. Examples of configurable systems include rolling tracks or wheels (for efficient travel), spiders or centipedes (for climbing), snakes (for burrowing in ground), and climbers (for inspection and repair in space). This video shows several configurations and behaviors that are new for modular and reconfigurable robots. Each SuperBot module is a complete robotic system and has a power supply, micro- controllers, sensors, communication, three degrees of freedom, and six connecting faces (front, back, left, right, up and down) to dynamically connect to other modules. This design allows flexible bending, docking, and continuous rotation. A single module can move forward, back, left, right, flip-over, and rotate as a wheel. Modules can communication with each other for totally distributed control and can support arbitrary module reshuffling during their operation. The modules have both internal and external sensors for monitoring self-status and environmental parameters. They can form arbitrary configurations (graphs) and can control these configurations for different functionality such as locomotion, manipulation, and self-repair. This video shows the latest status the SuperBot modules and all these behaviors were made in just one week. The fact that SuperBot can achieve so much in so short a time demonstrates the unique value of modular, multifunctional and self-reconfigurable robots.</p></div></div></li><li> <span id="chen2007geometry-inspired-optimization-methods-for-structural">B. Y. Chen, D. H. Bryant, V. Y. Fofanov, D. M. Kristensen, <b>M. Moll</b>, M. Kimmel, O. Lichtarge, and L. E. Kavraki, “Geometry-inspired Optimization Methods for Structural Motifs for Protein Function Prediction,” in <i>Automated Function Prediction Meeting (AFP)</i>, Vienna, Austria, 2007.</span> <br><div id="accchen2007geometry-inspired-optimization-methods-for-structural"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/chen2007geometry-inspired-optimization-methods-for-structural.pdf');" href="/publications/chen2007geometry-inspired-optimization-methods-for-structural.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibchen2007geometry-inspired-optimization-methods-for-structural').slideToggle(600);$('#abschen2007geometry-inspired-optimization-methods-for-structural').slideUp(600);"> bibtex </button> <button class="btn disabled" type="button">abstract</button></div><div class="card collapse" id="bibchen2007geometry-inspired-optimization-methods-for-structural"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">chen2007geometry-inspired-optimization-methods-for-structural</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Vienna, Austria"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Chen, Brian Y. and Bryant, Drew H. and Fofanov, Viacheslav Y. and Kristensen, David M. and Moll, Mark and Kimmel, Marek and Lichtarge, Olivier and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Automated Function Prediction Meeting (AFP)"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Geometry-inspired Optimization Methods for Structural Motifs for
        Protein Function Prediction"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2007"</span>
<span class="p">}</span></code></pre></figure></div></div></li><li> <span id="moll2006distributed-control-of-the-center-of-mass"><b>M. Moll</b>, P. Will, M. Krivokon, and W.-M. Shen, “Distributed Control of the Center of Mass of a Modular Robot,” in <i>Proc. 2006 IEEE/RSJ Intl. Conf. on Intelligent Robots and Systems</i>, Beijing, China, 2006, pp. 4710–4715.</span> <br><div id="accmoll2006distributed-control-of-the-center-of-mass"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2006distributed-control-of-the-center-of-mass.pdf');" href="/publications/moll2006distributed-control-of-the-center-of-mass.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/IROS.2006.282261');" href="https://dx.doi.org/10.1109/IROS.2006.282261"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2006distributed-control-of-the-center-of-mass').slideToggle(600);$('#absmoll2006distributed-control-of-the-center-of-mass').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2006distributed-control-of-the-center-of-mass').slideToggle(600);$('#bibmoll2006distributed-control-of-the-center-of-mass').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2006distributed-control-of-the-center-of-mass"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll2006distributed-control-of-the-center-of-mass</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a distributed controller for the center of mass of a
        modular robot. This is useful for locomotion of a modular robot over
        uneven and unknown terrain. By controlling the center of mass, a robot
        can prevent itself from falling over. We present a distributed and
        decentralized algorithm that computes the mass properties of the robot.
        Additionally, each module also computes the mass properties of the
        modules that are directly or indirectly connected to each of its
        connectors. With this information, each module can independently steer
        the center of mass towards a desired position by adjusting its joint
        positions. We present simulation results that show the feasibility of
        the approach."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Beijing, China"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Will, Peter and Krivokon, Maks and Shen, Wei-Min"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 2006 {IEEE/RSJ} Intl.\ Conf.\ on Intelligent Robots and
        Systems"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/IROS.2006.282261"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"4710--4715"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Distributed Control of the Center of Mass of a Modular Robot"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2006"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2006distributed-control-of-the-center-of-mass"><h3>Abstract</h3><p>We present a distributed controller for the center of mass of a modular robot. This is useful for locomotion of a modular robot over uneven and unknown terrain. By controlling the center of mass, a robot can prevent itself from falling over. We present a distributed and decentralized algorithm that computes the mass properties of the robot. Additionally, each module also computes the mass properties of the modules that are directly or indirectly connected to each of its connectors. With this information, each module can independently steer the center of mass towards a desired position by adjusting its joint positions. We present simulation results that show the feasibility of the approach.</p></div></div></li><li> <span id="salemi2006superbot">B. Salemi, <b>M. Moll</b>, and W.-M. Shen, “SUPERBOT: A Deployable, Multi-Functional, and Modular Self-Reconfigurable Robotic System,” in <i>Proc. 2006 IEEE/RSJ Intl. Conf. on Intelligent Robots and Systems</i>, Beijing, China, 2006, pp. 3636–3641.</span> <br><div id="accsalemi2006superbot"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/salemi2006superbot.pdf');" href="/publications/salemi2006superbot.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/IROS.2006.281719');" href="https://dx.doi.org/10.1109/IROS.2006.281719"> publisher </a> <button class="btn" type="button" onclick="$('#bibsalemi2006superbot').slideToggle(600);$('#abssalemi2006superbot').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abssalemi2006superbot').slideToggle(600);$('#bibsalemi2006superbot').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibsalemi2006superbot"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">salemi2006superbot</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Self-reconfigurable robots are modular robots that can
        autonomously change their shape and size to meet specific operational
        demands. Recently, there has been a great interest in using
        self-reconfigurable robots in applications such as reconnaissance,
        rescue missions, and space applications. Designing and controlling
        self-reconfigurable robots is a difficult task. Hence, the research has
        primarily been focused on developing systems that can function in a
        controlled environment. This paper presents a novel self-reconfigurable
        robotic system called SuperBot, which addresses the challenges of
        building and controlling deployable self-reconfigurable robots. Six
        prototype modules have been built and preliminary experimental results
        demonstrate that SuperBot is a flexible and powerful system that can be
        used in challenging real-world applications."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Beijing, China"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Salemi, Behnam and Moll, Mark and Shen, Wei-Min"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 2006 {IEEE/RSJ} Intl.\ Conf.\ on Intelligent Robots and
        Systems"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/IROS.2006.281719"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"3636--3641"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"{SUPERBOT}: A Deployable, Multi-Functional, and Modular
        Self-Reconfigurable Robotic System"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2006"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abssalemi2006superbot"><h3>Abstract</h3><p>Self-reconfigurable robots are modular robots that can autonomously change their shape and size to meet specific operational demands. Recently, there has been a great interest in using self-reconfigurable robots in applications such as reconnaissance, rescue missions, and space applications. Designing and controlling self-reconfigurable robots is a difficult task. Hence, the research has primarily been focused on developing systems that can function in a controlled environment. This paper presents a novel self-reconfigurable robotic system called SuperBot, which addresses the challenges of building and controlling deployable self-reconfigurable robots. Six prototype modules have been built and preliminary experimental results demonstrate that SuperBot is a flexible and powerful system that can be used in challenging real-world applications.</p></div></div></li><li> <span id="moll-kavraki2005path-plann-variab-resol"><b>M. Moll</b> and L. E. Kavraki, “Path Planning for Variable Resolution Minimal-Energy Curves of Constant Length,” in <i>Proc. 2005 IEEE Intl. Conf. on Robotics and Automation</i>, 2005, pp. 2142–2147.</span> <br><div id="accmoll-kavraki2005path-plann-variab-resol"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll-kavraki2005path-plann-variab-resol.pdf');" href="/publications/moll-kavraki2005path-plann-variab-resol.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ROBOT.2005.1570428');" href="https://dx.doi.org/10.1109/ROBOT.2005.1570428"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll-kavraki2005path-plann-variab-resol').slideToggle(600);$('#absmoll-kavraki2005path-plann-variab-resol').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll-kavraki2005path-plann-variab-resol').slideToggle(600);$('#bibmoll-kavraki2005path-plann-variab-resol').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll-kavraki2005path-plann-variab-resol"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll-kavraki2005path-plann-variab-resol</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a new approach to path planning for flexible wires. We
        introduce a method for computing stable configurations of a wire
        subject to manipulation constraints. These configurations correspond to
        minimal-energy curves. The representation is adaptive in the sense that
        the number of parameters automatically varies with the complexity of
        the underlying curve. We introduce a planner that computes paths from
        one minimal-energy curve to another such that all intermediate curves
        are also minimal-energy curves. Using a simplified model for obstacles,
        we can find minimal-energy curves of fixed length that pass through
        specified tangents at given control points. Our work has applications
        in motion planning for surgical suturing and snake-like robots."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 2005 {IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ROBOT.2005.1570428"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"path planning, minimal-energy curves, subdivision"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"2142--2147"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Path Planning for Variable Resolution Minimal-Energy Curves of
        Constant Length"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2005"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll-kavraki2005path-plann-variab-resol"><h3>Abstract</h3><p>We present a new approach to path planning for flexible wires. We introduce a method for computing stable configurations of a wire subject to manipulation constraints. These configurations correspond to minimal-energy curves. The representation is adaptive in the sense that the number of parameters automatically varies with the complexity of the underlying curve. We introduce a planner that computes paths from one minimal-energy curve to another such that all intermediate curves are also minimal-energy curves. Using a simplified model for obstacles, we can find minimal-energy curves of fixed length that pass through specified tangents at given control points. Our work has applications in motion planning for surgical suturing and snake-like robots.</p></div></div></li><li> <span id="moll-kavraki2004path-plann-minim-energ"><b>M. Moll</b> and L. E. Kavraki, “Path Planning for Minimal Energy Curves of Constant Length,” in <i>Proc. 2004 IEEE Intl. Conf. on Robotics and Automation</i>, 2004, pp. 2826–2831.</span> <br><div id="accmoll-kavraki2004path-plann-minim-energ"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll-kavraki2004path-plann-minim-energ.pdf');" href="/publications/moll-kavraki2004path-plann-minim-energ.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ROBOT.2004.1307489');" href="https://dx.doi.org/10.1109/ROBOT.2004.1307489"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll-kavraki2004path-plann-minim-energ').slideToggle(600);$('#absmoll-kavraki2004path-plann-minim-energ').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll-kavraki2004path-plann-minim-energ').slideToggle(600);$('#bibmoll-kavraki2004path-plann-minim-energ').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll-kavraki2004path-plann-minim-energ"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll-kavraki2004path-plann-minim-energ</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"In this paper we present a new path planning technique for a
        flexible wire. We first introduce a new parametrization designed to
        represent low-energy configurations. Based on this parametrization we
        can find curves that satisfy endpoint constraints. Next, we present
        three different techniques for minimizing energy within the self-motion
        manifold of the curve. We introduce a local planner to find smooth
        minimal energy deformations for these curves that can be used by a
        general path planning algorithm. Using a simplified model for
        obstacles, we can find minimal energy curves of fixed length that pass
        through specified tangents at given control points. Finally, we show
        that the parametrization introduced in this paper is a good
        approximation of true minimal energy curves. Our work has applications
        in surgical suturing and snake-like robots."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 2004 {IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ROBOT.2004.1307489"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"path planning, curve parametrization, minimal energy curves"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"2826--2831"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Path Planning for Minimal Energy Curves of Constant Length"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2004"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll-kavraki2004path-plann-minim-energ"><h3>Abstract</h3><p>In this paper we present a new path planning technique for a flexible wire. We first introduce a new parametrization designed to represent low-energy configurations. Based on this parametrization we can find curves that satisfy endpoint constraints. Next, we present three different techniques for minimizing energy within the self-motion manifold of the curve. We introduce a local planner to find smooth minimal energy deformations for these curves that can be used by a general path planning algorithm. Using a simplified model for obstacles, we can find minimal energy curves of fixed length that pass through specified tangents at given control points. Finally, we show that the parametrization introduced in this paper is a good approximation of true minimal energy curves. Our work has applications in surgical suturing and snake-like robots.</p></div></div></li><li> <span id="moll-erdmann2002dynam-shape-recon-using"><b>M. Moll</b> and M. A. Erdmann, “Dynamic Shape Reconstruction Using Tactile Sensors,” in <i>Proc. 2002 IEEE Intl. Conf. on Robotics and Automation</i>, 2002, pp. 1636–1641.</span> <br><div id="accmoll-erdmann2002dynam-shape-recon-using"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll-erdmann2002dynam-shape-recon-using.pdf');" href="/publications/moll-erdmann2002dynam-shape-recon-using.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ROBOT.2002.1014777');" href="https://dx.doi.org/10.1109/ROBOT.2002.1014777"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll-erdmann2002dynam-shape-recon-using').slideToggle(600);$('#absmoll-erdmann2002dynam-shape-recon-using').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll-erdmann2002dynam-shape-recon-using').slideToggle(600);$('#bibmoll-erdmann2002dynam-shape-recon-using').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll-erdmann2002dynam-shape-recon-using"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll-erdmann2002dynam-shape-recon-using</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present new results on reconstruction of the shape and motion
        of an unknown object using tactile sensors without requiring object
        immobilization. A robot manipulates the object with two flat palms
        covered with tactile sensors. We model the full dynamics and prove
        local observability of the shape, motion and center of mass of the
        object based on the motion of the contact points as measured by the
        tactile sensors."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Erdmann, Michael A."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 2002 {IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ROBOT.2002.1014777"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"shape reconstruction, tactile sensing, observability"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"1636--1641"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Dynamic Shape Reconstruction Using Tactile Sensors"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2002"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll-erdmann2002dynam-shape-recon-using"><h3>Abstract</h3><p>We present new results on reconstruction of the shape and motion of an unknown object using tactile sensors without requiring object immobilization. A robot manipulates the object with two flat palms covered with tactile sensors. We model the full dynamics and prove local observability of the shape, motion and center of mass of the object based on the motion of the contact points as measured by the tactile sensors.</p></div></div></li><li> <span id="moll2002orien-micro-scale-parts"><b>M. Moll</b>, K. Goldberg, M. A. Erdmann, and R. Fearing, “Orienting Micro-Scale Parts with Squeeze and Roll Primitives,” in <i>Proc. 2002 IEEE Intl. Conf. on Robotics and Automation</i>, 2002, pp. 1931–1936.</span> <br><div id="accmoll2002orien-micro-scale-parts"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2002orien-micro-scale-parts.pdf');" href="/publications/moll2002orien-micro-scale-parts.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ROBOT.2002.1014823');" href="https://dx.doi.org/10.1109/ROBOT.2002.1014823"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll2002orien-micro-scale-parts').slideToggle(600);$('#absmoll2002orien-micro-scale-parts').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2002orien-micro-scale-parts').slideToggle(600);$('#bibmoll2002orien-micro-scale-parts').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2002orien-micro-scale-parts"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll2002orien-micro-scale-parts</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Orienting parts that measure only a few micrometers in diameter
        introduces several challenges that need not be considered at the
        macro-scale. First, there are several kinds of sticking effects due to
        Van der Waals forces and static electricity which complicate hand-off
        motions and release of a part. Second, the degrees of freedom of
        micro-manipulators are limited. This paper proposes a pair of
        manipulation primitives and a complete algorithm that addresses these
        challenges. We will show that a sequence of these two manipulation
        primitives can uniquely orient any asymmetric part while maintaining
        contact without sensing. This allows us to apply the same plan to many
        (identical) parts simultaneously. For asymmetric parts we can find a
        plan of length O(n) in O(n) time that orients the part, where n is the
        number of vertices."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Goldberg, Ken and Erdmann, Michael A. and Fearing, Ron"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 2002 {IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ROBOT.2002.1014823"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"micromanipulation, parts orienting, rolling"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"1931--1936"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Orienting Micro-Scale Parts with Squeeze and Roll Primitives"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2002"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2002orien-micro-scale-parts"><h3>Abstract</h3><p>Orienting parts that measure only a few micrometers in diameter introduces several challenges that need not be considered at the macro-scale. First, there are several kinds of sticking effects due to Van der Waals forces and static electricity which complicate hand-off motions and release of a part. Second, the degrees of freedom of micro-manipulators are limited. This paper proposes a pair of manipulation primitives and a complete algorithm that addresses these challenges. We will show that a sequence of these two manipulation primitives can uniquely orient any asymmetric part while maintaining contact without sensing. This allows us to apply the same plan to many (identical) parts simultaneously. For asymmetric parts we can find a plan of length O(n) in O(n) time that orients the part, where n is the number of vertices.</p></div></div></li><li> <span id="moll-erdmann2001recon-shape-motion-using"><b>M. Moll</b> and M. A. Erdmann, “Reconstructing Shape from Motion Using Tactile Sensors,” in <i>Proc. 2001 IEEE/RSJ Intl. Conf. on Intelligent Robots and Systems</i>, Maui, HI, 2001, pp. 691–700.</span> <br><div id="accmoll-erdmann2001recon-shape-motion-using"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll-erdmann2001recon-shape-motion-using.pdf');" href="/publications/moll-erdmann2001recon-shape-motion-using.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/IROS.2001.976250');" href="https://dx.doi.org/10.1109/IROS.2001.976250"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll-erdmann2001recon-shape-motion-using').slideToggle(600);$('#absmoll-erdmann2001recon-shape-motion-using').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll-erdmann2001recon-shape-motion-using').slideToggle(600);$('#bibmoll-erdmann2001recon-shape-motion-using').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll-erdmann2001recon-shape-motion-using"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll-erdmann2001recon-shape-motion-using</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a new method to reconstruct the shape of an unknown
        object using tactile sensors without requiring object immobilization.
        Instead, the robot manipulates the object without prehension. The robot
        infers the shape, motion and center of mass of the object based on the
        motion of the contact points as measured by tactile sensors. Our
        analysis is supported by simulation and experimental results."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Maui, HI"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Erdmann, Michael A."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 2001 {IEEE/RSJ} Intl.\ Conf.\ on Intelligent Robots and
        Systems"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/IROS.2001.976250"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"tactile sensing, shape reconstruction, contact kinematics"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"691--700"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Reconstructing Shape from Motion Using Tactile Sensors"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2001"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll-erdmann2001recon-shape-motion-using"><h3>Abstract</h3><p>We present a new method to reconstruct the shape of an unknown object using tactile sensors without requiring object immobilization. Instead, the robot manipulates the object without prehension. The robot infers the shape, motion and center of mass of the object based on the motion of the contact points as measured by tactile sensors. Our analysis is supported by simulation and experimental results.</p></div></div></li><li> <span id="moll-erdmann2000uncer-reduc-using-dynam"><b>M. Moll</b> and M. A. Erdmann, “Uncertainty Reduction Using Dynamics,” in <i>Proc. 2000 IEEE Intl. Conf. on Robotics and Automation</i>, San Francisco, California, 2000, pp. 3673–3680.</span> <br><div id="accmoll-erdmann2000uncer-reduc-using-dynam"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll-erdmann2000uncer-reduc-using-dynam.pdf');" href="/publications/moll-erdmann2000uncer-reduc-using-dynam.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.1109/ROBOT.2000.845304');" href="https://dx.doi.org/10.1109/ROBOT.2000.845304"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll-erdmann2000uncer-reduc-using-dynam').slideToggle(600);$('#absmoll-erdmann2000uncer-reduc-using-dynam').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll-erdmann2000uncer-reduc-using-dynam').slideToggle(600);$('#bibmoll-erdmann2000uncer-reduc-using-dynam').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll-erdmann2000uncer-reduc-using-dynam"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll-erdmann2000uncer-reduc-using-dynam</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"For assembly tasks parts often have to be oriented before they
        can be put in an assembly. The results presented in this paper are a
        component of the automated design of parts orienting devices. The focus
        is on orienting parts with minimal sensing and manipulation. We present
        a new approach to parts orienting through the manipulation of pose
        distributions. Through dynamic simulation we can determine the pose
        distribution for an object being dropped from an arbitrary height on an
        arbitrary surface. By varying the drop height and the shape of the
        support surface we can find the initial conditions that will result in
        a pose distribution with minimal entropy. We are trying to uniquely
        orient a part with high probability just by varying the initial
        conditions. We will derive a condition on the pose and velocity of an
        object in contact with a sloped surface that will allow us to quickly
        determine the final resting configuration of the object. This condition
        can then be used to quickly compute the pose distribution. We also show
        simulation and experimental results that confirm that our dynamic
        simulator can be used to find the true pose distribution of an
        object."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"San Francisco, California"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Erdmann, Michael A."</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 2000 {IEEE} Intl.\ Conf.\ on Robotics and Automation"</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.1109/ROBOT.2000.845304"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"pose distributions, parts orienting, dynamic simulation"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"3673--3680"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Uncertainty Reduction Using Dynamics"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2000"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll-erdmann2000uncer-reduc-using-dynam"><h3>Abstract</h3><p>For assembly tasks parts often have to be oriented before they can be put in an assembly. The results presented in this paper are a component of the automated design of parts orienting devices. The focus is on orienting parts with minimal sensing and manipulation. We present a new approach to parts orienting through the manipulation of pose distributions. Through dynamic simulation we can determine the pose distribution for an object being dropped from an arbitrary height on an arbitrary surface. By varying the drop height and the shape of the support surface we can find the initial conditions that will result in a pose distribution with minimal entropy. We are trying to uniquely orient a part with high probability just by varying the initial conditions. We will derive a condition on the pose and velocity of an object in contact with a sloped surface that will allow us to quickly determine the final resting configuration of the object. This condition can then be used to quickly compute the pose distribution. We also show simulation and experimental results that confirm that our dynamic simulator can be used to find the true pose distribution of an object.</p></div></div></li><li> <span id="ter1996languag-engin-dialog-system">H. ter Doest, <b>M. Moll</b>, R. Bos, S. Van de Burgt, and A. Nijholt, “Language Engineering in Dialogue Systems,” in <i>Computers in Engineering Symposium</i>, Houston, TX, 1996, pp. 68–79.</span> <br><div id="accter1996languag-engin-dialog-system"><div> <button class="btn disabled">pdf</button> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibter1996languag-engin-dialog-system').slideToggle(600);$('#abster1996languag-engin-dialog-system').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abster1996languag-engin-dialog-system').slideToggle(600);$('#bibter1996languag-engin-dialog-system').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibter1996languag-engin-dialog-system"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">ter1996languag-engin-dialog-system</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The analysis of natural language in the context of
        keyboard-driven dialogue systems is the central issue addressed in this
        paper. A module that corrects typing errors, performs domain-specific
        morphological analysis is developed. A parser for typed unification
        grammars has been designed and implemented in C++; for description of
        the lexicon and the grammar a suitable specification language has been
        developed. It is argued that typed unification grammars and especially
        the newly developed specification language are convenient formalisms
        for describing natural language use in dialogue systems. Finally we
        present a dialogue manager that is based on a finite state automaton;
        transitions in the automaton depend upon availability of information in
        utterances of the user. In order to keep track of the history of the
        dialogue, a context stack is constructed during the dialogue. The
        manager is implemented in Prolog. "</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Houston, TX"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"{ter} Doest, Hugo and Moll, Mark and Bos, Ren{\'e} and Van de Burgt, Stan and Nijholt, Anton"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Computers in Engineering Symposium"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"68--79"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Language Engineering in Dialogue Systems"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"1996"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abster1996languag-engin-dialog-system"><h3>Abstract</h3><p>The analysis of natural language in the context of keyboard-driven dialogue systems is the central issue addressed in this paper. A module that corrects typing errors, performs domain-specific morphological analysis is developed. A parser for typed unification grammars has been designed and implemented in C++; for description of the lexicon and the grammar a suitable specification language has been developed. It is argued that typed unification grammars and especially the newly developed specification language are convenient formalisms for describing natural language use in dialogue systems. Finally we present a dialogue manager that is based on a finite state automaton; transitions in the automaton depend upon availability of information in utterances of the user. In order to keep track of the history of the dialogue, a context stack is constructed during the dialogue. The manager is implemented in Prolog.</p></div></div></li><li> <span id="op1995parsin-dialog-system-typed">R. op den Akker, H. ter Doest, <b>M. Moll</b>, and A. Nijholt, “Parsing in Dialogue Systems using Typed Feature Structures,” in <i>Proceedings of the International Workshop on Parsing Technologies</i>, Prague/Karlovy Vary, Czech Republic, 1995.</span> <br><div id="accop1995parsin-dialog-system-typed"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/op1995parsin-dialog-system-typed.pdf');" href="/publications/op1995parsin-dialog-system-typed.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibop1995parsin-dialog-system-typed').slideToggle(600);$('#absop1995parsin-dialog-system-typed').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absop1995parsin-dialog-system-typed').slideToggle(600);$('#bibop1995parsin-dialog-system-typed').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibop1995parsin-dialog-system-typed"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">op1995parsin-dialog-system-typed</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The analysis of natural language in the context of
        keyboard-driven dialogue systems is the central issue addressed in this
        paper. A module that corrects typing errors, performs domain-specific
        morphological analysis is developed. A parser for typed unification
        grammars is designed and implemented in C++; for description of the
        lexicon and the grammer a specialised specification language is
        developed. It is argued that typed unification grammars and especially
        the newly developed specification language are convenient formalisms
        for describing natural language use in dialogue systems. Research on
        these issues is carried out in the context of the Schisma project, a
        research project in linguistic engineering; participants in Schisma are
        KPN Research and the University of Twente."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Prague/Karlovy Vary, Czech Republic"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"{op} den Akker, Rieks and {ter} Doest, Hugo and Moll, Mark and Nijholt, Anton"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proceedings of the International Workshop on Parsing
        Technologies"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Parsing in Dialogue Systems using Typed Feature Structures"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"1995"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absop1995parsin-dialog-system-typed"><h3>Abstract</h3><p>The analysis of natural language in the context of keyboard-driven dialogue systems is the central issue addressed in this paper. A module that corrects typing errors, performs domain-specific morphological analysis is developed. A parser for typed unification grammars is designed and implemented in C++; for description of the lexicon and the grammer a specialised specification language is developed. It is argued that typed unification grammars and especially the newly developed specification language are convenient formalisms for describing natural language use in dialogue systems. Research on these issues is carried out in the context of the Schisma project, a research project in linguistic engineering; participants in Schisma are KPN Research and the University of Twente.</p></div></div></li><li> <span id="moll1994capac-conver-zone-episod"><b>M. Moll</b>, R. Miikkulainen, and J. Abbey, “The Capacity of Convergence-Zone Episodic Memory,” in <i>Proc. 12th Natl. Conf. on Artificial Intelligence (AAAI-94)</i>, Cambridge, MA, 1994, pp. 68–73.</span> <br><div id="accmoll1994capac-conver-zone-episod"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll1994capac-conver-zone-episod.pdf');" href="/publications/moll1994capac-conver-zone-episod.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://www.aaai.org/Papers/AAAI/1994/AAAI94-011.pdf');" href="https://www.aaai.org/Papers/AAAI/1994/AAAI94-011.pdf"> publisher </a> <button class="btn" type="button" onclick="$('#bibmoll1994capac-conver-zone-episod').slideToggle(600);$('#absmoll1994capac-conver-zone-episod').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll1994capac-conver-zone-episod').slideToggle(600);$('#bibmoll1994capac-conver-zone-episod').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll1994capac-conver-zone-episod"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">moll1994capac-conver-zone-episod</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Human episodic memory provides a seemingly unlimited storage for
        everyday experiences, and a retrieval system that allows us to access
        the experiences with partial activation of their components. This paper
        presents a neural network model of episodic memory inspired by
        Damasio's idea of Convergence Zones. The model consists of a layer of
        perceptual feature maps and a binding layer. A perceptual feature
        pattern is coarse coded in the binding layer, and stored on the weights
        between layers. A partial activation of the stored features activates
        the binding pattern which in turn reactivates the entire stored
        pattern. A worst-case analysis shows that with realistic-size layers,
        the memory capacity of the model is several times larger than the
        number of units in the model, and could account for the large capacity
        of human episodic memory."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Cambridge, MA"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Miikkulainen, Risto and Abbey, Jonathan"</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">"Proc.\ 12th Natl. Conf.\ on Artificial Intelligence (AAAI-94)"</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">"68--73"</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">"MIT Press"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"The Capacity of Convergence-Zone Episodic Memory"</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">"https://www.aaai.org/Papers/AAAI/1994/AAAI94-011.pdf"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"1994"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll1994capac-conver-zone-episod"><h3>Abstract</h3><p>Human episodic memory provides a seemingly unlimited storage for everyday experiences, and a retrieval system that allows us to access the experiences with partial activation of their components. This paper presents a neural network model of episodic memory inspired by Damasio’s idea of Convergence Zones. The model consists of a layer of perceptual feature maps and a binding layer. A perceptual feature pattern is coarse coded in the binding layer, and stored on the weights between layers. A partial activation of the stored features activates the binding pattern which in turn reactivates the entire stored pattern. A worst-case analysis shows that with realistic-size layers, the memory capacity of the model is several times larger than the number of units in the model, and could account for the large capacity of human episodic memory.</p></div></div></li></ol><h2 class="bibliography" id="sectheses">Theses</h2><ol class="bibliography"><li> <span id="moll2002shape-recon-using-activ"><b>M. Moll</b>, “Shape Reconstruction Using Active Tactile Sensors,” PhD thesis, Computer Science Department, Carnegie Mellon University, Pittsburgh, PA, 2002.</span> <br><div id="accmoll2002shape-recon-using-activ"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2002shape-recon-using-activ.pdf');" href="/publications/moll2002shape-recon-using-activ.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibmoll2002shape-recon-using-activ').slideToggle(600);$('#absmoll2002shape-recon-using-activ').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2002shape-recon-using-activ').slideToggle(600);$('#bibmoll2002shape-recon-using-activ').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2002shape-recon-using-activ"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@phdthesis</span><span class="p">{</span><span class="nl">moll2002shape-recon-using-activ</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a new method to reconstruct the shape of an unknown
        object using tactile sensors, without requiring object immobilization.
        Instead, sensing and nonprehensile manipulation occur simultaneously.
        The robot infers the shape, motion and center of mass of the object
        based on the motion of the contact points as measured by the tactile
        sensors. This allows for a natural, continuous interaction between
        manipulation and sensing. We analyze the planar case first by assuming
        quasistatic dynamics, and present simulation results and experimental
        results obtained using this analysis. We extend this analysis to the
        full dynamics and prove observability of the nonlinear system
        describing the shape and motion of the object being manipulated. In our
        simulations, a simple observer based on Newton's method for root
        finding performs really well. Using the same framework we can also
        describe the shape and dynamics of three-dimensional objects. However,
        there are some fundamental differences between the planar and
        three-dimensional case, due to increased tangent dimensionality. Also,
        perfect global shape reconstruction is impossible in the 3D case, but
        it is almost trivial to obtain upper and lower bounds on the shape. The
        3D shape reconstruction method has also been implemented and we present
        some simulation results."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Pittsburgh, PA"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"tactile sensing, shape reconstruction, nonprehensile
        manipulation"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jul</span><span class="p">,</span>
  <span class="na">pubtype</span> <span class="p">=</span> <span class="s">"Other"</span><span class="p">,</span>
  <span class="na">school</span> <span class="p">=</span> <span class="s">"Computer Science Department, Carnegie Mellon University"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Shape Reconstruction Using Active Tactile Sensors"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2002"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2002shape-recon-using-activ"><h3>Abstract</h3><p>We present a new method to reconstruct the shape of an unknown object using tactile sensors, without requiring object immobilization. Instead, sensing and nonprehensile manipulation occur simultaneously. The robot infers the shape, motion and center of mass of the object based on the motion of the contact points as measured by the tactile sensors. This allows for a natural, continuous interaction between manipulation and sensing. We analyze the planar case first by assuming quasistatic dynamics, and present simulation results and experimental results obtained using this analysis. We extend this analysis to the full dynamics and prove observability of the nonlinear system describing the shape and motion of the object being manipulated. In our simulations, a simple observer based on Newton’s method for root finding performs really well. Using the same framework we can also describe the shape and dynamics of three-dimensional objects. However, there are some fundamental differences between the planar and three-dimensional case, due to increased tangent dimensionality. Also, perfect global shape reconstruction is impossible in the 3D case, but it is almost trivial to obtain upper and lower bounds on the shape. The 3D shape reconstruction method has also been implemented and we present some simulation results.</p></div></div></li><li> <span id="moll1995head-parsin-typed-featur"><b>M. Moll</b>, “Head-corner Parsing Using Typed Feature Structures,” Master's thesis, Department of Computer Science, University of Twente, 1995.</span> <br><div id="accmoll1995head-parsin-typed-featur"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll1995head-parsin-typed-featur.pdf');" href="/publications/moll1995head-parsin-typed-featur.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibmoll1995head-parsin-typed-featur').slideToggle(600);$('#absmoll1995head-parsin-typed-featur').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll1995head-parsin-typed-featur').slideToggle(600);$('#bibmoll1995head-parsin-typed-featur').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll1995head-parsin-typed-featur"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@mastersthesis</span><span class="p">{</span><span class="nl">moll1995head-parsin-typed-featur</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"In this report a description will be given of how typed feature
        structures can be specified. A specification language will be presented
        for the specification of types, words and grammar rules. An unification
        algorithm for typed feature structures as well as an algorithm to
        compute the least upper bound relation for a type lattice will be
        given. Finally, a head-corner parsing schema for typed feature
        structures will be presented."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark"</span><span class="p">,</span>
  <span class="na">pubtype</span> <span class="p">=</span> <span class="s">"Other"</span><span class="p">,</span>
  <span class="na">school</span> <span class="p">=</span> <span class="s">"Department of Computer Science, University of Twente"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Head-corner Parsing Using Typed Feature Structures"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"1995"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll1995head-parsin-typed-featur"><h3>Abstract</h3><p>In this report a description will be given of how typed feature structures can be specified. A specification language will be presented for the specification of types, words and grammar rules. An unification algorithm for typed feature structures as well as an algorithm to compute the least upper bound relation for a type lattice will be given. Finally, a head-corner parsing schema for typed feature structures will be presented.</p></div></div></li></ol><h2 class="bibliography" id="sectechnical-reports">Technical Reports</h2><ol class="bibliography"><li> <span id="lewis2019how-much-do-unstated-problem-constraints">W. C. Lewis II, <b>M. Moll</b>, and L. E. Kavraki, “How Much Do Unstated Problem Constraints Limit Deep Robotic Reinforcement Learning?,” Rice University, Department of Computer Science, Houston, Texas, TR19-01, 2019.</span> <br><div id="acclewis2019how-much-do-unstated-problem-constraints"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/lewis2019how-much-do-unstated-problem-constraints.pdf');" href="/publications/lewis2019how-much-do-unstated-problem-constraints.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://dx.doi.org/10.25611/az5z-xt37');" href="https://dx.doi.org/10.25611/az5z-xt37"> publisher </a> <button class="btn" type="button" onclick="$('#biblewis2019how-much-do-unstated-problem-constraints').slideToggle(600);$('#abslewis2019how-much-do-unstated-problem-constraints').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abslewis2019how-much-do-unstated-problem-constraints').slideToggle(600);$('#biblewis2019how-much-do-unstated-problem-constraints').slideUp(600);"> abstract </button></div><div class="card collapse" id="biblewis2019how-much-do-unstated-problem-constraints"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@techreport</span><span class="p">{</span><span class="nl">lewis2019how-much-do-unstated-problem-constraints</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"Deep Reinforcement Learning is a promising paradigm for robotic
        control which has been shown to be capable of learning policies for
        high-dimensional, continuous control of unmodeled systems. However,
        Robotic Reinforcement Learning currently lacks clearly defined
        benchmark tasks, which makes it difficult for researchers to reproduce
        and compare against prior work. ``Reacher'' tasks, which are
        fundamental to robotic manipulation, are commonly used as benchmarks,
        but the lack of a formal specification elides details that are crucial
        to replication. In this paper we present a novel empirical analysis
        which shows that the unstated spatial constraints in commonly used
        implementations of Reacher tasks make it dramatically easier to learn a
        successful control policy with Deep Deterministic Policy Gradients
        (DDPG), a state-of-the-art Deep RL algorithm. Our analysis suggests
        that less constrained Reacher tasks are significantly more difficult to
        learn, and hence that existing de facto benchmarks are not
        representative of the difficulty of general robotic manipulation."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Houston, Texas"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Lewis~II, W. Cannon and Moll, Mark and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">"10.25611/az5z-xt37"</span><span class="p">,</span>
  <span class="na">institution</span> <span class="p">=</span> <span class="s">"Rice University, Department of Computer Science"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"TR19-01"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"How Much Do Unstated Problem Constraints Limit Deep Robotic
        Reinforcement Learning?"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2019"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abslewis2019how-much-do-unstated-problem-constraints"><h3>Abstract</h3><p>Deep Reinforcement Learning is a promising paradigm for robotic control which has been shown to be capable of learning policies for high-dimensional, continuous control of unmodeled systems. However, Robotic Reinforcement Learning currently lacks clearly defined benchmark tasks, which makes it difficult for researchers to reproduce and compare against prior work. “Reacher” tasks, which are fundamental to robotic manipulation, are commonly used as benchmarks, but the lack of a formal specification elides details that are crucial to replication. In this paper we present a novel empirical analysis which shows that the unstated spatial constraints in commonly used implementations of Reacher tasks make it dramatically easier to learn a successful control policy with Deep Deterministic Policy Gradients (DDPG), a state-of-the-art Deep RL algorithm. Our analysis suggests that less constrained Reacher tasks are significantly more difficult to learn, and hence that existing de facto benchmarks are not representative of the difficulty of general robotic manipulation.</p></div></div></li><li> <span id="moll2004dockin-flexib-ligan"><b>M. Moll</b>, D. Schwarz, A. Heath, and L. E. Kavraki, “On Flexible Docking Using Expansive Search,” Rice University, Houston, TX, 04-443, 2004.</span> <br><div id="accmoll2004dockin-flexib-ligan"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll2004dockin-flexib-ligan.pdf');" href="/publications/moll2004dockin-flexib-ligan.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibmoll2004dockin-flexib-ligan').slideToggle(600);$('#absmoll2004dockin-flexib-ligan').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll2004dockin-flexib-ligan').slideToggle(600);$('#bibmoll2004dockin-flexib-ligan').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll2004dockin-flexib-ligan"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@techreport</span><span class="p">{</span><span class="nl">moll2004dockin-flexib-ligan</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The activity of most drugs is regulated by the binding of one
        molecule (the ligand) to a pocket of another, usually larger, molecule,
        which is commonly a protein. This report describes a new approach to
        creating low-energy structures of flexible proteins to which ligands
        can be docked. The flexibility of molecules is encoded with thousands
        of parameters making the search for valid complexes a formidable
        problem. Our method takes into account the flexibility of the protein
        as this can be encoded by its major modes of motion. The output of the
        program consists of low-energy protein conformations that can then be
        docked with a ligand using a traditional docking program. We employ a
        robotics-based approach for exploring the conformational space of the
        protein. Our long term goal is to develop an efficient, accurate, and
        automated algorithm that will be used to screen large databases of
        molecules for novel therapeutics."</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Houston, TX"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Schwarz, David and Heath, Allison and Kavraki, Lydia E."</span><span class="p">,</span>
  <span class="na">institution</span> <span class="p">=</span> <span class="s">"Rice University"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"04-443"</span><span class="p">,</span>
  <span class="na">pubtype</span> <span class="p">=</span> <span class="s">"Other"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"On Flexible Docking Using Expansive Search"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2004"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll2004dockin-flexib-ligan"><h3>Abstract</h3><p>The activity of most drugs is regulated by the binding of one molecule (the ligand) to a pocket of another, usually larger, molecule, which is commonly a protein. This report describes a new approach to creating low-energy structures of flexible proteins to which ligands can be docked. The flexibility of molecules is encoded with thousands of parameters making the search for valid complexes a formidable problem. Our method takes into account the flexibility of the protein as this can be encoded by its major modes of motion. The output of the program consists of low-energy protein conformations that can then be docked with a ligand using a traditional docking program. We employ a robotics-based approach for exploring the conformational space of the protein. Our long term goal is to develop an efficient, accurate, and automated algorithm that will be used to screen large databases of molecules for novel therapeutics.</p></div></div></li><li> <span id="moll-erdmann2001shape-recon-planar-dynam"><b>M. Moll</b> and M. A. Erdmann, “Shape Reconstruction in a Planar Dynamic Environment,” Dept. of Computer Science, Carnegie Mellon University, CMU-CS-01-107, 2001.</span> <br><div id="accmoll-erdmann2001shape-recon-planar-dynam"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/moll-erdmann2001shape-recon-planar-dynam.pdf');" href="/publications/moll-erdmann2001shape-recon-planar-dynam.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibmoll-erdmann2001shape-recon-planar-dynam').slideToggle(600);$('#absmoll-erdmann2001shape-recon-planar-dynam').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absmoll-erdmann2001shape-recon-planar-dynam').slideToggle(600);$('#bibmoll-erdmann2001shape-recon-planar-dynam').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibmoll-erdmann2001shape-recon-planar-dynam"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@techreport</span><span class="p">{</span><span class="nl">moll-erdmann2001shape-recon-planar-dynam</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"We present a a new method to reconstruct the shape of an unknown
        object using tactile sensors, without requiring object immobilization.
        Instead, sensing and nonprehensile manipulation occur simultaneously.
        The robot infers the shape, motion and center of mass of the object
        based on the motion of the contact points as measured by the tactile
        sensors. We present analytic results and simulation results assuming
        quasistatic dynamics. We prove that the shape and motion are observable
        in both the quasistatic and the fully dynamic case."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark and Erdmann, Michael A."</span><span class="p">,</span>
  <span class="na">institution</span> <span class="p">=</span> <span class="s">"Dept. of Computer Science, Carnegie Mellon University"</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">"tactile reconstruction, contact kinematics, nonlinear observer
        theory, tactile sensing"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"CMU-CS-01-107"</span><span class="p">,</span>
  <span class="na">pubtype</span> <span class="p">=</span> <span class="s">"Other"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Shape Reconstruction in a Planar Dynamic Environment"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"2001"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absmoll-erdmann2001shape-recon-planar-dynam"><h3>Abstract</h3><p>We present a a new method to reconstruct the shape of an unknown object using tactile sensors, without requiring object immobilization. Instead, sensing and nonprehensile manipulation occur simultaneously. The robot infers the shape, motion and center of mass of the object based on the motion of the contact points as measured by the tactile sensors. We present analytic results and simulation results assuming quasistatic dynamics. We prove that the shape and motion are observable in both the quasistatic and the fully dynamic case.</p></div></div></li><li> <span id="moll1996mappin-scien"><b>M. Moll</b>, “Mapping Science: Methods and Tools for the Automatic Creation of Semantic Maps of Large Corpora,” Centre for Science and Technology Studies (CWTS), Leiden, the Netherlands, Report CWTS 96-06, Aug. 1996.</span> Research report to the Netherlands Organization for Scientific Research (NWO), Foundation for Economic and Socio-Cultural Sciences (ESR) <br><div id="accmoll1996mappin-scien"><div> <button class="btn disabled">pdf</button> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibmoll1996mappin-scien').slideToggle(600);$('#absmoll1996mappin-scien').slideUp(600);"> bibtex </button> <button class="btn disabled" type="button">abstract</button></div><div class="card collapse" id="bibmoll1996mappin-scien"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@techreport</span><span class="p">{</span><span class="nl">moll1996mappin-scien</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">"Leiden, the Netherlands"</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"Moll, Mark"</span><span class="p">,</span>
  <span class="na">institution</span> <span class="p">=</span> <span class="s">"Centre for Science and Technology Studies (CWTS)"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">aug</span><span class="p">,</span>
  <span class="na">note</span> <span class="p">=</span> <span class="s">"Research report to the Netherlands Organization for Scientific
        Research (NWO), Foundation for Economic and Socio-Cultural Sciences
        (ESR)"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"96-06"</span><span class="p">,</span>
  <span class="na">pubtype</span> <span class="p">=</span> <span class="s">"Other"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Mapping Science: Methods and Tools for the Automatic Creation of
        Semantic Maps of Large Corpora"</span><span class="p">,</span>
  <span class="na">type</span> <span class="p">=</span> <span class="s">"Report CWTS"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"1996"</span>
<span class="p">}</span></code></pre></figure></div></div></li><li> <span id="ter-doest1996language-engineering-in-dialogue-systems">H. ter Doest, <b>M. Moll</b>, R. Bos, S. Van de Burgt, and A. Nijholt, “Language Engineering in Dialogue Systems,” Department of Computer Science, University of Twente, Memoranda Informatica 96-2, Jan. 1996.</span> <br><div id="accter-doest1996language-engineering-in-dialogue-systems"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/ter-doest1996language-engineering-in-dialogue-systems.pdf');" href="/publications/ter-doest1996language-engineering-in-dialogue-systems.pdf"> pdf </a> <button class="btn disabled">publisher</button> <button class="btn" type="button" onclick="$('#bibter-doest1996language-engineering-in-dialogue-systems').slideToggle(600);$('#abster-doest1996language-engineering-in-dialogue-systems').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#abster-doest1996language-engineering-in-dialogue-systems').slideToggle(600);$('#bibter-doest1996language-engineering-in-dialogue-systems').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibter-doest1996language-engineering-in-dialogue-systems"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@techreport</span><span class="p">{</span><span class="nl">ter-doest1996language-engineering-in-dialogue-systems</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The analysis of natural language in the context of
        keyboard-driven dialogue systems is the central issue addressed in this
        paper. A module that corrects typing errors, performs domain-specific
        morphological analysis is developed. A parser for typed unification
        grammars has been designed and implemented in C++; for description of
        the lexicon and the grammar a suitable specification language has been
        developed. It is argued that typed unification grammars and especially
        the newly developed specification language are convenient formalisms
        for describing natural language use in dialogue systems. Finally we
        present a dialogue manager that is based on a finite state automaton;
        transitions in the automaton depend upon availability of information in
        utterances of the user. In order to keep track of the history of the
        dialogue, a context stack is constructed during the dialogue. The
        manager is implemented in Prolog. "</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"{ter} Doest, Hugo and Moll, Mark and Bos, Ren{\'e} and Van de Burgt, Stan and Nijholt, Anton"</span><span class="p">,</span>
  <span class="na">institution</span> <span class="p">=</span> <span class="s">"Department of Computer Science, University of Twente"</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jan</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"96-2"</span><span class="p">,</span>
  <span class="na">pubtype</span> <span class="p">=</span> <span class="s">"Other"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Language Engineering in Dialogue Systems"</span><span class="p">,</span>
  <span class="na">type</span> <span class="p">=</span> <span class="s">"Memoranda Informatica"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"1996"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="abster-doest1996language-engineering-in-dialogue-systems"><h3>Abstract</h3><p>The analysis of natural language in the context of keyboard-driven dialogue systems is the central issue addressed in this paper. A module that corrects typing errors, performs domain-specific morphological analysis is developed. A parser for typed unification grammars has been designed and implemented in C++; for description of the lexicon and the grammar a suitable specification language has been developed. It is argued that typed unification grammars and especially the newly developed specification language are convenient formalisms for describing natural language use in dialogue systems. Finally we present a dialogue manager that is based on a finite state automaton; transitions in the automaton depend upon availability of information in utterances of the user. In order to keep track of the history of the dialogue, a context stack is constructed during the dialogue. The manager is implemented in Prolog.</p></div></div></li><li> <span id="op1995parsin-dialog-system-typed-tr">R. op den Akker, H. ter Doest, <b>M. Moll</b>, and A. Nijholt, “Parsing in Dialogue Systems using Typed Feature Structures,” Department of Computer Science, University of Twente, Memoranda Informatica 95-25, 1995.</span> <br><div id="accop1995parsin-dialog-system-typed-tr"><div> <a class="btn" onclick="pageTracker._trackPageview('/publications/op1995parsin-dialog-system-typed-tr.pdf');" href="/publications/op1995parsin-dialog-system-typed-tr.pdf"> pdf </a> <a class="btn" onclick="pageTracker._trackPageview('https://research.utwente.nl/en/publications/parsing-in-dialogue-systems-using-typed-feature-structures-2');" href="https://research.utwente.nl/en/publications/parsing-in-dialogue-systems-using-typed-feature-structures-2"> publisher </a> <button class="btn" type="button" onclick="$('#bibop1995parsin-dialog-system-typed-tr').slideToggle(600);$('#absop1995parsin-dialog-system-typed-tr').slideUp(600);"> bibtex </button> <button class="btn" type="button" onclick="$('#absop1995parsin-dialog-system-typed-tr').slideToggle(600);$('#bibop1995parsin-dialog-system-typed-tr').slideUp(600);"> abstract </button></div><div class="card collapse" id="bibop1995parsin-dialog-system-typed-tr"><h3>BibTeX</h3><figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@techreport</span><span class="p">{</span><span class="nl">op1995parsin-dialog-system-typed-tr</span><span class="p">,</span>
  <span class="na">abstract</span> <span class="p">=</span> <span class="s">"The analysis of natural language in the context of
        keyboard-driven dialogue systems is the central issue addressed in this
        paper. A module that corrects typing errors and performs
        domain-specific morphological analysis has been developed. A parser for
        typed unification grammars is designed and implemented in C++; for
        description of the lexicon and the grammer a specialised specification
        language has been developed. It is argued that typed unification
        grammars and especially the newly developed specification language are
        convenient formalisms for describing natural language use in dialogue
        systems. Research on these issues is carried out in the context of the
        Schisma project, a research project of the Parlevink group in
        linguistic engineering; participants in Schisma are KPN Research and
        the University of Twente. The aims of the Schisma project are twofold:
        both the accumulation of knowledge in the field of computational
        linguistics and the development of a natural language interfaced
        theatre information and booking system is envisaged. The Schisma
        project serves as a testbed for the development of the various language
        analysis modules necessary for dialogue systems."</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">"{op} den Akker, Rieks and {ter} Doest, Hugo and Moll, Mark and Nijholt, Anton"</span><span class="p">,</span>
  <span class="na">institution</span> <span class="p">=</span> <span class="s">"Department of Computer Science, University of Twente"</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">"95-25"</span><span class="p">,</span>
  <span class="na">pubtype</span> <span class="p">=</span> <span class="s">"Other"</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">"Parsing in Dialogue Systems using Typed Feature Structures"</span><span class="p">,</span>
  <span class="na">type</span> <span class="p">=</span> <span class="s">"Memoranda Informatica"</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">"https://research.utwente.nl/en/publications/parsing-in-dialogue-systems-using-typed-feature-structures-2"</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">"1995"</span>
<span class="p">}</span></code></pre></figure></div><div class="card collapse" id="absop1995parsin-dialog-system-typed-tr"><h3>Abstract</h3><p>The analysis of natural language in the context of keyboard-driven dialogue systems is the central issue addressed in this paper. A module that corrects typing errors and performs domain-specific morphological analysis has been developed. A parser for typed unification grammars is designed and implemented in C++; for description of the lexicon and the grammer a specialised specification language has been developed. It is argued that typed unification grammars and especially the newly developed specification language are convenient formalisms for describing natural language use in dialogue systems. Research on these issues is carried out in the context of the Schisma project, a research project of the Parlevink group in linguistic engineering; participants in Schisma are KPN Research and the University of Twente. The aims of the Schisma project are twofold: both the accumulation of knowledge in the field of computational linguistics and the development of a natural language interfaced theatre information and booking system is envisaged. The Schisma project serves as a testbed for the development of the various language analysis modules necessary for dialogue systems.</p></div></div></li></ol></section><footer class="page__meta"></footer></div></article></div></div><div id="footer" class="page__footer"><footer> <!-- start custom footer snippets --> <!-- end custom footer snippets --><div class="page__footer-follow"><ul class="social-icons"><li><a href="mailto:mark@moll.ai" rel="nofollow noopener noreferrer"><i class="fas fa-fw fa-envelope-square" aria-hidden="true"></i> Email</a></li><li><a href="https://www.linkedin.com/in/mmoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-linkedin" aria-hidden="true"></i> LinkedIn</a></li><li><a href="https://scholar.google.com/citations?user=sLXafasAAAAJ&hl=en" rel="nofollow noopener noreferrer"><i class="ai ai-fw ai-google-scholar" aria-hidden="true"></i> Google Scholar</a></li><li><a href="http://orcid.org/0000-0002-0451-2797" rel="nofollow noopener noreferrer"><i class="ai ai-fw ai-orcid" aria-hidden="true"></i> ORCID</a></li><li><a href="https://github.com/mamoll" rel="nofollow noopener noreferrer"><i class="fab fa-fw fa-github" aria-hidden="true"></i> GitHub</a></li></ul></div><div class="page__footer-copyright">&copy; 2021 Mark Moll.</div></footer></div><script src="/assets/js/main.min.js"></script> <!-- Global site tag (gtag.js) - Google Analytics --> <script async src="https://www.googletagmanager.com/gtag/js?id=G-NV77NNFC6G"></script> <script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-NV77NNFC6G', { 'anonymize_ip': false}); </script> <script src="/assets/js/footer-script.js"></script></body></html>
